// grid-layout-plus/dist/grid-layout-plus.mjs
(function(){"use strict";try{if(typeof document<"u"){var e=document.createElement("style");e.appendChild(document.createTextNode('.vgl-layout{--vgl-placeholder-bg: red;--vgl-placeholder-opacity: 20%;--vgl-placeholder-z-index: 2;--vgl-item-resizing-z-index: 3;--vgl-item-resizing-opacity: 60%;--vgl-item-dragging-z-index: 3;--vgl-item-dragging-opacity: 100%;--vgl-resizer-size: 10px;--vgl-resizer-border-color: #444;--vgl-resizer-border-width: 2px;position:relative;box-sizing:border-box;transition:height .2s ease}.vgl-item{position:absolute;box-sizing:border-box;transition:.2s ease;transition-property:left,top,right}.vgl-item--placeholder{z-index:var(--vgl-placeholder-z-index, 2);-webkit-user-select:none;-moz-user-select:none;user-select:none;background-color:var(--vgl-placeholder-bg, red);opacity:var(--vgl-placeholder-opacity, 20%);transition-duration:.1s}.vgl-item--no-touch{touch-action:none}.vgl-item--transform{right:auto;left:0;transition-property:transform}.vgl-item--transform.vgl-item--rtl{right:0;left:auto}.vgl-item--resizing{z-index:var(--vgl-item-resizing-z-index, 3);-webkit-user-select:none;-moz-user-select:none;user-select:none;opacity:var(--vgl-item-resizing-opacity, 60%)}.vgl-item--dragging{z-index:var(--vgl-item-dragging-z-index, 3);-webkit-user-select:none;-moz-user-select:none;user-select:none;opacity:var(--vgl-item-dragging-opacity, 100%);transition:none}.vgl-item__resizer{position:absolute;right:0;bottom:0;box-sizing:border-box;width:var(--vgl-resizer-size);height:var(--vgl-resizer-size);cursor:se-resize}.vgl-item__resizer:before{position:absolute;inset:0 3px 3px 0;content:"";border:0 solid var(--vgl-resizer-border-color);border-right-width:var(--vgl-resizer-border-width);border-bottom-width:var(--vgl-resizer-border-width)}.vgl-item__resizer--rtl{right:auto;left:0;cursor:sw-resize}.vgl-item__resizer--rtl:before{inset:0 0 3px 3px;border-right-width:0;border-bottom-width:var(--vgl-resizer-border-width);border-left-width:var(--vgl-resizer-border-width)}')),document.head.appendChild(e)}}catch(r){console.error("vite-plugin-css-injected-by-js",r)}})();
import { defineComponent as Ir, inject as ar, ref as Ot, reactive as Pt, toRef as ho, onBeforeMount as Cr, onMounted as Ar, watchEffect as go, onBeforeUnmount as Br, computed as Ue, watch as ee, openBlock as St, createElementBlock as Yt, normalizeClass as sr, unref as pn, normalizeStyle as Fr, renderSlot as yn, createCommentVNode as vo, nextTick as qe, provide as cr, toRefs as lr, Fragment as mo, renderList as yo, createBlock as bo, mergeProps as wo, withCtx as xo, withDirectives as Eo, createVNode as ko, vShow as So } from "vue";
const To = typeof window < "u";
var ur;
To && ((ur = window == null ? void 0 : window.navigator) != null && ur.userAgent) && /iP(ad|hone|od)/.test(window.navigator.userAgent);
function we(c) {
  return c == null;
}
function jr() {
}
const zo = Object.freeze({
  aliceblue: "f0f8ff",
  antiquewhite: "faebd7",
  aqua: "0ff",
  aquamarine: "7fffd4",
  azure: "f0ffff",
  beige: "f5f5dc",
  bisque: "ffe4c4",
  black: "000",
  blanchedalmond: "ffebcd",
  blue: "00f",
  blueviolet: "8a2be2",
  brown: "a52a2a",
  burlywood: "deb887",
  burntsienna: "ea7e5d",
  cadetblue: "5f9ea0",
  chartreuse: "7fff00",
  chocolate: "d2691e",
  coral: "ff7f50",
  cornflowerblue: "6495ed",
  cornsilk: "fff8dc",
  crimson: "dc143c",
  cyan: "0ff",
  darkblue: "00008b",
  darkcyan: "008b8b",
  darkgoldenrod: "b8860b",
  darkgray: "a9a9a9",
  darkgreen: "006400",
  darkgrey: "a9a9a9",
  darkkhaki: "bdb76b",
  darkmagenta: "8b008b",
  darkolivegreen: "556b2f",
  darkorange: "ff8c00",
  darkorchid: "9932cc",
  darkred: "8b0000",
  darksalmon: "e9967a",
  darkseagreen: "8fbc8f",
  darkslateblue: "483d8b",
  darkslategray: "2f4f4f",
  darkslategrey: "2f4f4f",
  darkturquoise: "00ced1",
  darkviolet: "9400d3",
  deeppink: "ff1493",
  deepskyblue: "00bfff",
  dimgray: "696969",
  dimgrey: "696969",
  dodgerblue: "1e90ff",
  firebrick: "b22222",
  floralwhite: "fffaf0",
  forestgreen: "228b22",
  fuchsia: "f0f",
  gainsboro: "dcdcdc",
  ghostwhite: "f8f8ff",
  gold: "ffd700",
  goldenrod: "daa520",
  gray: "808080",
  green: "008000",
  greenyellow: "adff2f",
  grey: "808080",
  honeydew: "f0fff0",
  hotpink: "ff69b4",
  indianred: "cd5c5c",
  indigo: "4b0082",
  ivory: "fffff0",
  khaki: "f0e68c",
  lavender: "e6e6fa",
  lavenderblush: "fff0f5",
  lawngreen: "7cfc00",
  lemonchiffon: "fffacd",
  lightblue: "add8e6",
  lightcoral: "f08080",
  lightcyan: "e0ffff",
  lightgoldenrodyellow: "fafad2",
  lightgray: "d3d3d3",
  lightgreen: "90ee90",
  lightgrey: "d3d3d3",
  lightpink: "ffb6c1",
  lightsalmon: "ffa07a",
  lightseagreen: "20b2aa",
  lightskyblue: "87cefa",
  lightslategray: "789",
  lightslategrey: "789",
  lightsteelblue: "b0c4de",
  lightyellow: "ffffe0",
  lime: "0f0",
  limegreen: "32cd32",
  linen: "faf0e6",
  magenta: "f0f",
  maroon: "800000",
  mediumaquamarine: "66cdaa",
  mediumblue: "0000cd",
  mediumorchid: "ba55d3",
  mediumpurple: "9370db",
  mediumseagreen: "3cb371",
  mediumslateblue: "7b68ee",
  mediumspringgreen: "00fa9a",
  mediumturquoise: "48d1cc",
  mediumvioletred: "c71585",
  midnightblue: "191970",
  mintcream: "f5fffa",
  mistyrose: "ffe4e1",
  moccasin: "ffe4b5",
  navajowhite: "ffdead",
  navy: "000080",
  oldlace: "fdf5e6",
  olive: "808000",
  olivedrab: "6b8e23",
  orange: "ffa500",
  orangered: "ff4500",
  orchid: "da70d6",
  palegoldenrod: "eee8aa",
  palegreen: "98fb98",
  paleturquoise: "afeeee",
  palevioletred: "db7093",
  papayawhip: "ffefd5",
  peachpuff: "ffdab9",
  peru: "cd853f",
  pink: "ffc0cb",
  plum: "dda0dd",
  powderblue: "b0e0e6",
  purple: "800080",
  rebeccapurple: "663399",
  red: "f00",
  rosybrown: "bc8f8f",
  royalblue: "4169e1",
  saddlebrown: "8b4513",
  salmon: "fa8072",
  sandybrown: "f4a460",
  seagreen: "2e8b57",
  seashell: "fff5ee",
  sienna: "a0522d",
  silver: "c0c0c0",
  skyblue: "87ceeb",
  slateblue: "6a5acd",
  slategray: "708090",
  slategrey: "708090",
  snow: "fffafa",
  springgreen: "00ff7f",
  steelblue: "4682b4",
  tan: "d2b48c",
  teal: "008080",
  thistle: "d8bfd8",
  tomato: "ff6347",
  turquoise: "40e0d0",
  violet: "ee82ee",
  wheat: "f5deb3",
  white: "fff",
  whitesmoke: "f5f5f5",
  yellow: "ff0",
  yellowgreen: "9acd32"
});
Object.freeze(new Set(Object.keys(zo)));
function _o() {
  const c = /* @__PURE__ */ new Map();
  return {
    on(d, p) {
      const f = c.get(d);
      f != null && f.add(p) || c.set(d, /* @__PURE__ */ new Set([p]));
    },
    off(d, p) {
      const f = c.get(d);
      f && f.delete(p);
    },
    clear(d) {
      const p = c.get(d);
      p && p.clear();
    },
    clearAll() {
      c.clear();
    },
    emit(d, ...p) {
      const f = c.get(d);
      f && f.forEach((v) => {
        v(...p);
      });
    }
  };
}
function fr(c, d = 16) {
  if (typeof c != "function")
    return jr;
  const p = (...w) => {
    c(...w);
  };
  if (d <= 0)
    return Lr(p);
  let f = 0, v;
  return function(...w) {
    const k = Date.now(), g = k - f;
    clearTimeout(v), g >= d ? (f = k, p(...w)) : v = setTimeout(
      () => {
        f = Date.now(), p(...w);
      },
      Math.max(0, d - g)
    );
  };
}
function Oo(c, d = 100) {
  if (typeof c != "function")
    return jr;
  const p = (...v) => {
    c(...v);
  };
  if (d <= 0)
    return Lr(p);
  let f;
  return function(...v) {
    clearTimeout(f), f = setTimeout(() => {
      p(...v);
    }, d);
  };
}
function Lr(c) {
  if (typeof c != "function")
    return c;
  let d = !1, p, f;
  return function(...v) {
    return p = v, d || (d = !0, f = Promise.resolve().then(() => (d = !1, f = void 0, c(...p)))), f;
  };
}
const Tt = /* @__PURE__ */ new Set(), Wr = /* @__PURE__ */ new WeakMap();
function Po() {
  Tt.forEach((c) => {
    c(...Wr.get(c));
  }), Tt.clear();
}
function he(c, ...d) {
  if (typeof c != "function")
    return c;
  Wr.set(c, d), !Tt.has(c) && (Tt.add(c), Tt.size === 1 && Promise.resolve().then(Po));
}
const Hr = Symbol("LAYOUT_KEY"), Nr = Symbol("EMITTER_KEY");
function Mo(c) {
  let d = 0, p;
  for (let f = 0, v = c.length; f < v; f++)
    p = c[f].y + c[f].h, p > d && (d = p);
  return d;
}
function bn(c) {
  const d = Array(c.length);
  for (let p = 0, f = c.length; p < f; p++)
    d[p] = Do(c[p]);
  return d;
}
function Do(c) {
  return { ...c };
}
function qr(c, d) {
  return !(c === d || c.x + c.w <= d.x || c.x >= d.x + d.w || c.y + c.h <= d.y || c.y >= d.y + d.h);
}
function ut(c, d, p) {
  const f = Yr(c), v = $r(c), w = Array(c.length);
  for (let k = 0, g = v.length; k < g; k++) {
    let K = v[k];
    K.static || (K = Ro(f, K, d, p), f.push(K)), w[c.indexOf(K)] = K, K.moved = !1;
  }
  return w;
}
function Ro(c, d, p, f) {
  if (p)
    for (; d.y > 0 && !zt(c, d); )
      d.y--;
  else if (f) {
    const w = f[d.i].y;
    for (; d.y > w && !zt(c, d); )
      d.y--;
  }
  let v;
  for (; v = zt(c, d); )
    d.y = v.y + v.h;
  return d;
}
function Io(c, d) {
  const p = Yr(c);
  for (let f = 0, v = c.length; f < v; f++) {
    const w = c[f];
    if (w.x + w.w > d.cols && (w.x = d.cols - w.w), w.x < 0 && (w.x = 0, w.w = d.cols), !w.static)
      p.push(w);
    else
      for (; zt(p, w); )
        w.y++;
  }
  return c;
}
function dr(c, d) {
  for (let p = 0, f = c.length; p < f; p++)
    if (c[p].i === d)
      return c[p];
}
function zt(c, d) {
  for (let p = 0, f = c.length; p < f; p++)
    if (qr(c[p], d))
      return c[p];
}
function Xr(c, d) {
  return c.filter((p) => qr(p, d));
}
function Yr(c) {
  return c.filter((d) => d.static);
}
function wn(c, d, p, f, v = !1, w = !1) {
  if (d.static)
    return c;
  const k = d.x, g = d.y, K = f && d.y > f;
  typeof p == "number" && (d.x = p), typeof f == "number" && (d.y = f), d.moved = !0;
  let J = $r(c);
  K && (J = J.reverse());
  const F = Xr(J, d);
  if (w && F.length)
    return d.x = k, d.y = g, d.moved = !1, c;
  for (let ae = 0, ce = F.length; ae < ce; ae++) {
    const ne = F[ae];
    ne.moved || d.y > ne.y && d.y - ne.y > ne.h / 4 || (ne.static ? c = pr(c, ne, d, v) : c = pr(c, d, ne, v));
  }
  return c;
}
function pr(c, d, p, f) {
  if (f) {
    const w = {
      x: p.x,
      y: p.y,
      w: p.w,
      h: p.h,
      i: "-1"
    };
    if (w.y = Math.max(d.y - p.h, 0), !zt(c, w))
      return wn(c, p, void 0, w.y, !1);
  }
  return wn(c, p, void 0, p.y + 1, !1);
}
function Co(c, d, p, f) {
  const v = "translate3d(" + d + "px," + c + "px, 0)";
  return {
    transform: v,
    WebkitTransform: v,
    MozTransform: v,
    msTransform: v,
    OTransform: v,
    width: p + "px",
    height: f + "px",
    position: "absolute"
  };
}
function Ao(c, d, p, f) {
  const v = "translate3d(" + d * -1 + "px," + c + "px, 0)";
  return {
    transform: v,
    WebkitTransform: v,
    MozTransform: v,
    msTransform: v,
    OTransform: v,
    width: p + "px",
    height: f + "px",
    position: "absolute"
  };
}
function Bo(c, d, p, f) {
  return {
    top: c + "px",
    left: d + "px",
    width: p + "px",
    height: f + "px",
    position: "absolute"
  };
}
function Fo(c, d, p, f) {
  return {
    top: c + "px",
    right: d + "px",
    width: p + "px",
    height: f + "px",
    position: "absolute"
  };
}
function $r(c) {
  return Array.from(c).sort(function(d, p) {
    return d.y === p.y && d.x === p.x ? 0 : d.y > p.y || d.y === p.y && d.x > p.x ? 1 : -1;
  });
}
function jo(c, d) {
  d = d || "Layout";
  const p = ["x", "y", "w", "h"], f = [];
  if (!Array.isArray(c))
    throw new Error(d + " must be an array!");
  for (let v = 0, w = c.length; v < w; v++) {
    const k = c[v];
    for (let g = 0; g < p.length; g++)
      if (typeof k[p[g]] != "number")
        throw new Error(
          "VueGridLayout: " + d + "[" + v + "]." + p[g] + " must be a number!"
        );
    if (k.i === void 0 || k.i === null)
      throw new Error("VueGridLayout: " + d + "[" + v + "].i cannot be null!");
    if (typeof k.i != "number" && typeof k.i != "string")
      throw new Error("VueGridLayout: " + d + "[" + v + "].i must be a string or number!");
    if (f.indexOf(k.i) >= 0)
      throw new Error("VueGridLayout: " + d + "[" + v + "].i must be unique!");
    if (f.push(k.i), k.static !== void 0 && typeof k.static != "boolean")
      throw new Error("VueGridLayout: " + d + "[" + v + "].static must be a boolean!");
  }
}
function Lo(c, d = "vgl") {
  const p = () => `${d}-${c}`;
  return {
    b: p,
    be: (k) => `${p()}__${k}`,
    bm: (k) => `${p()}--${k}`,
    bem: (k, g) => `${p()}__${k}--${g}`
  };
}
function hr(c) {
  return Wo(c);
}
function Wo(c) {
  var w;
  const d = ((w = c.target) == null ? void 0 : w.offsetParent) || document.body, p = c.offsetParent === document.body ? { left: 0, top: 0 } : d.getBoundingClientRect(), f = c.clientX + d.scrollLeft - p.left, v = c.clientY + d.scrollTop - p.top;
  return { x: f, y: v };
}
function gr(c, d, p, f) {
  return Ho(c) ? {
    deltaX: p - c,
    deltaY: f - d,
    lastX: c,
    lastY: d,
    x: p,
    y: f
  } : {
    deltaX: 0,
    deltaY: 0,
    lastX: p,
    lastY: f,
    x: p,
    y: f
  };
}
function Ho(c) {
  return typeof c == "number" && !Number.isNaN(c);
}
function No(c, d) {
  const p = Gr(c);
  let f = p[0];
  for (let v = 1, w = p.length; v < w; v++) {
    const k = p[v];
    d > c[k] && (f = k);
  }
  return f;
}
function xn(c, d) {
  if (!d[c])
    throw new Error("ResponsiveGridLayout: `cols` entry for breakpoint " + c + " is missing!");
  return d[c];
}
function qo(c, d, p, f, v, w, k) {
  if (d[f])
    return bn(d[f]);
  let g = c;
  const K = Gr(p), J = K.slice(K.indexOf(f));
  for (let F = 0, ae = J.length; F < ae; F++) {
    const ce = J[F];
    if (d[ce]) {
      g = d[ce];
      break;
    }
  }
  return g = bn(g || []), ut(Io(g, { cols: w }), k);
}
function Gr(c) {
  return Object.keys(c).sort((p, f) => c[p] - c[f]);
}
let Xo = "auto";
function Yo() {
  return typeof document < "u";
}
function vr() {
  return Yo() ? typeof document.dir < "u" ? document.dir : document.getElementsByTagName("html")[0].getAttribute("dir") : Xo;
}
var $o = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, En = {}, Go = {
  get exports() {
    return En;
  },
  set exports(c) {
    En = c;
  }
};
(function(c, d) {
  (function(p, f) {
    c.exports = f();
  })($o, function() {
    function p(t, e) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
        var r = Object.getOwnPropertySymbols(t);
        e && (r = r.filter(function(i) {
          return Object.getOwnPropertyDescriptor(t, i).enumerable;
        })), n.push.apply(n, r);
      }
      return n;
    }
    function f(t) {
      for (var e = 1; e < arguments.length; e++) {
        var n = arguments[e] != null ? arguments[e] : {};
        e % 2 ? p(Object(n), !0).forEach(function(r) {
          K(t, r, n[r]);
        }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : p(Object(n)).forEach(function(r) {
          Object.defineProperty(t, r, Object.getOwnPropertyDescriptor(n, r));
        });
      }
      return t;
    }
    function v(t) {
      return v = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
        return typeof e;
      } : function(e) {
        return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
      }, v(t);
    }
    function w(t, e) {
      if (!(t instanceof e))
        throw new TypeError("Cannot call a class as a function");
    }
    function k(t, e) {
      for (var n = 0; n < e.length; n++) {
        var r = e[n];
        r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), Object.defineProperty(t, De(r.key), r);
      }
    }
    function g(t, e, n) {
      return e && k(t.prototype, e), n && k(t, n), Object.defineProperty(t, "prototype", { writable: !1 }), t;
    }
    function K(t, e, n) {
      return (e = De(e)) in t ? Object.defineProperty(t, e, { value: n, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = n, t;
    }
    function J(t, e) {
      if (typeof e != "function" && e !== null)
        throw new TypeError("Super expression must either be null or a function");
      t.prototype = Object.create(e && e.prototype, { constructor: { value: t, writable: !0, configurable: !0 } }), Object.defineProperty(t, "prototype", { writable: !1 }), e && ae(t, e);
    }
    function F(t) {
      return F = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(e) {
        return e.__proto__ || Object.getPrototypeOf(e);
      }, F(t);
    }
    function ae(t, e) {
      return ae = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(n, r) {
        return n.__proto__ = r, n;
      }, ae(t, e);
    }
    function ce(t) {
      if (t === void 0)
        throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
      return t;
    }
    function ne(t) {
      var e = function() {
        if (typeof Reflect > "u" || !Reflect.construct || Reflect.construct.sham)
          return !1;
        if (typeof Proxy == "function")
          return !0;
        try {
          return Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function() {
          })), !0;
        } catch {
          return !1;
        }
      }();
      return function() {
        var n, r = F(t);
        if (e) {
          var i = F(this).constructor;
          n = Reflect.construct(r, arguments, i);
        } else
          n = r.apply(this, arguments);
        return function(o, a) {
          if (a && (typeof a == "object" || typeof a == "function"))
            return a;
          if (a !== void 0)
            throw new TypeError("Derived constructors may only return object or undefined");
          return ce(o);
        }(this, n);
      };
    }
    function Ee() {
      return Ee = typeof Reflect < "u" && Reflect.get ? Reflect.get.bind() : function(t, e, n) {
        var r = function(o, a) {
          for (; !Object.prototype.hasOwnProperty.call(o, a) && (o = F(o)) !== null; )
            ;
          return o;
        }(t, e);
        if (r) {
          var i = Object.getOwnPropertyDescriptor(r, e);
          return i.get ? i.get.call(arguments.length < 3 ? t : n) : i.value;
        }
      }, Ee.apply(this, arguments);
    }
    function De(t) {
      var e = function(n, r) {
        if (typeof n != "object" || n === null)
          return n;
        var i = n[Symbol.toPrimitive];
        if (i !== void 0) {
          var o = i.call(n, r || "default");
          if (typeof o != "object")
            return o;
          throw new TypeError("@@toPrimitive must return a primitive value.");
        }
        return (r === "string" ? String : Number)(n);
      }(t, "string");
      return typeof e == "symbol" ? e : e + "";
    }
    var Te = function(t) {
      return !(!t || !t.Window) && t instanceof t.Window;
    }, ue = void 0, V = void 0;
    function oe(t) {
      ue = t;
      var e = t.document.createTextNode("");
      e.ownerDocument !== t.document && typeof t.wrap == "function" && t.wrap(e) === e && (t = t.wrap(t)), V = t;
    }
    function Y(t) {
      return Te(t) ? t : (t.ownerDocument || t).defaultView || V.window;
    }
    typeof window < "u" && window && oe(window);
    var re = function(t) {
      return !!t && v(t) === "object";
    }, ve = function(t) {
      return typeof t == "function";
    }, b = { window: function(t) {
      return t === V || Te(t);
    }, docFrag: function(t) {
      return re(t) && t.nodeType === 11;
    }, object: re, func: ve, number: function(t) {
      return typeof t == "number";
    }, bool: function(t) {
      return typeof t == "boolean";
    }, string: function(t) {
      return typeof t == "string";
    }, element: function(t) {
      if (!t || v(t) !== "object")
        return !1;
      var e = Y(t) || V;
      return /object|function/.test(typeof Element > "u" ? "undefined" : v(Element)) ? t instanceof Element || t instanceof e.Element : t.nodeType === 1 && typeof t.nodeName == "string";
    }, plainObject: function(t) {
      return re(t) && !!t.constructor && /function Object\b/.test(t.constructor.toString());
    }, array: function(t) {
      return re(t) && t.length !== void 0 && ve(t.splice);
    } };
    function ge(t) {
      var e = t.interaction;
      if (e.prepared.name === "drag") {
        var n = e.prepared.axis;
        n === "x" ? (e.coords.cur.page.y = e.coords.start.page.y, e.coords.cur.client.y = e.coords.start.client.y, e.coords.velocity.client.y = 0, e.coords.velocity.page.y = 0) : n === "y" && (e.coords.cur.page.x = e.coords.start.page.x, e.coords.cur.client.x = e.coords.start.client.x, e.coords.velocity.client.x = 0, e.coords.velocity.page.x = 0);
      }
    }
    function xe(t) {
      var e = t.iEvent, n = t.interaction;
      if (n.prepared.name === "drag") {
        var r = n.prepared.axis;
        if (r === "x" || r === "y") {
          var i = r === "x" ? "y" : "x";
          e.page[i] = n.coords.start.page[i], e.client[i] = n.coords.start.client[i], e.delta[i] = 0;
        }
      }
    }
    var _ = { id: "actions/drag", install: function(t) {
      var e = t.actions, n = t.Interactable, r = t.defaults;
      n.prototype.draggable = _.draggable, e.map.drag = _, e.methodDict.drag = "draggable", r.actions.drag = _.defaults;
    }, listeners: { "interactions:before-action-move": ge, "interactions:action-resume": ge, "interactions:action-move": xe, "auto-start:check": function(t) {
      var e = t.interaction, n = t.interactable, r = t.buttons, i = n.options.drag;
      if (i && i.enabled && (!e.pointerIsDown || !/mouse|pointer/.test(e.pointerType) || r & n.options.drag.mouseButtons))
        return t.action = { name: "drag", axis: i.lockAxis === "start" ? i.startAxis : i.lockAxis }, !1;
    } }, draggable: function(t) {
      return b.object(t) ? (this.options.drag.enabled = t.enabled !== !1, this.setPerAction("drag", t), this.setOnEvents("drag", t), /^(xy|x|y|start)$/.test(t.lockAxis) && (this.options.drag.lockAxis = t.lockAxis), /^(xy|x|y)$/.test(t.startAxis) && (this.options.drag.startAxis = t.startAxis), this) : b.bool(t) ? (this.options.drag.enabled = t, this) : this.options.drag;
    }, beforeMove: ge, move: xe, defaults: { startAxis: "xy", lockAxis: "xy" }, getCursor: function() {
      return "move";
    }, filterEventType: function(t) {
      return t.search("drag") === 0;
    } }, $ = _, L = { init: function(t) {
      var e = t;
      L.document = e.document, L.DocumentFragment = e.DocumentFragment || ie, L.SVGElement = e.SVGElement || ie, L.SVGSVGElement = e.SVGSVGElement || ie, L.SVGElementInstance = e.SVGElementInstance || ie, L.Element = e.Element || ie, L.HTMLElement = e.HTMLElement || L.Element, L.Event = e.Event, L.Touch = e.Touch || ie, L.PointerEvent = e.PointerEvent || e.MSPointerEvent;
    }, document: null, DocumentFragment: null, SVGElement: null, SVGSVGElement: null, SVGElementInstance: null, Element: null, HTMLElement: null, Event: null, Touch: null, PointerEvent: null };
    function ie() {
    }
    var q = L, U = { init: function(t) {
      var e = q.Element, n = t.navigator || {};
      U.supportsTouch = "ontouchstart" in t || b.func(t.DocumentTouch) && q.document instanceof t.DocumentTouch, U.supportsPointerEvent = n.pointerEnabled !== !1 && !!q.PointerEvent, U.isIOS = /iP(hone|od|ad)/.test(n.platform), U.isIOS7 = /iP(hone|od|ad)/.test(n.platform) && /OS 7[^\d]/.test(n.appVersion), U.isIe9 = /MSIE 9/.test(n.userAgent), U.isOperaMobile = n.appName === "Opera" && U.supportsTouch && /Presto/.test(n.userAgent), U.prefixedMatchesSelector = "matches" in e.prototype ? "matches" : "webkitMatchesSelector" in e.prototype ? "webkitMatchesSelector" : "mozMatchesSelector" in e.prototype ? "mozMatchesSelector" : "oMatchesSelector" in e.prototype ? "oMatchesSelector" : "msMatchesSelector", U.pEventTypes = U.supportsPointerEvent ? q.PointerEvent === t.MSPointerEvent ? { up: "MSPointerUp", down: "MSPointerDown", over: "mouseover", out: "mouseout", move: "MSPointerMove", cancel: "MSPointerCancel" } : { up: "pointerup", down: "pointerdown", over: "pointerover", out: "pointerout", move: "pointermove", cancel: "pointercancel" } : null, U.wheelEvent = q.document && "onmousewheel" in q.document ? "mousewheel" : "wheel";
    }, supportsTouch: null, supportsPointerEvent: null, isIOS7: null, isIOS: null, isIe9: null, isOperaMobile: null, prefixedMatchesSelector: null, pEventTypes: null, wheelEvent: null }, C = U;
    function le(t, e) {
      if (t.contains)
        return t.contains(e);
      for (; e; ) {
        if (e === t)
          return !0;
        e = e.parentNode;
      }
      return !1;
    }
    function Re(t, e) {
      for (; b.element(t); ) {
        if (G(t, e))
          return t;
        t = te(t);
      }
      return null;
    }
    function te(t) {
      var e = t.parentNode;
      if (b.docFrag(e)) {
        for (; (e = e.host) && b.docFrag(e); )
          ;
        return e;
      }
      return e;
    }
    function G(t, e) {
      return V !== ue && (e = e.replace(/\/deep\//g, " ")), t[C.prefixedMatchesSelector](e);
    }
    var ze = function(t) {
      return t.parentNode || t.host;
    };
    function me(t, e) {
      for (var n, r = [], i = t; (n = ze(i)) && i !== e && n !== i.ownerDocument; )
        r.unshift(i), i = n;
      return r;
    }
    function dt(t, e, n) {
      for (; b.element(t); ) {
        if (G(t, e))
          return !0;
        if ((t = te(t)) === n)
          return G(t, e);
      }
      return !1;
    }
    function et(t) {
      return t.correspondingUseElement || t;
    }
    function _e(t) {
      var e = t instanceof q.SVGElement ? t.getBoundingClientRect() : t.getClientRects()[0];
      return e && { left: e.left, right: e.right, top: e.top, bottom: e.bottom, width: e.width || e.right - e.left, height: e.height || e.bottom - e.top };
    }
    function Xe(t) {
      var e, n = _e(t);
      if (!C.isIOS7 && n) {
        var r = { x: (e = (e = Y(t)) || V).scrollX || e.document.documentElement.scrollLeft, y: e.scrollY || e.document.documentElement.scrollTop };
        n.left += r.x, n.right += r.x, n.top += r.y, n.bottom += r.y;
      }
      return n;
    }
    function pt(t) {
      for (var e = []; t; )
        e.push(t), t = te(t);
      return e;
    }
    function ht(t) {
      return !!b.string(t) && (q.document.querySelector(t), !0);
    }
    function P(t, e) {
      for (var n in e)
        t[n] = e[n];
      return t;
    }
    function tt(t, e, n) {
      return t === "parent" ? te(n) : t === "self" ? e.getRect(n) : Re(n, t);
    }
    function Oe(t, e, n, r) {
      var i = t;
      return b.string(i) ? i = tt(i, e, n) : b.func(i) && (i = i.apply(void 0, r)), b.element(i) && (i = Xe(i)), i;
    }
    function Ye(t) {
      return t && { x: "x" in t ? t.x : t.left, y: "y" in t ? t.y : t.top };
    }
    function nt(t) {
      return !t || "x" in t && "y" in t || ((t = P({}, t)).x = t.left || 0, t.y = t.top || 0, t.width = t.width || (t.right || 0) - t.x, t.height = t.height || (t.bottom || 0) - t.y), t;
    }
    function rt(t, e, n) {
      t.left && (e.left += n.x), t.right && (e.right += n.x), t.top && (e.top += n.y), t.bottom && (e.bottom += n.y), e.width = e.right - e.left, e.height = e.bottom - e.top;
    }
    function $e(t, e, n) {
      var r = n && t.options[n];
      return Ye(Oe(r && r.origin || t.options.origin, t, e, [t && e])) || { x: 0, y: 0 };
    }
    function Pe(t, e) {
      var n = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : function(l) {
        return !0;
      }, r = arguments.length > 3 ? arguments[3] : void 0;
      if (r = r || {}, b.string(t) && t.search(" ") !== -1 && (t = gt(t)), b.array(t))
        return t.forEach(function(l) {
          return Pe(l, e, n, r);
        }), r;
      if (b.object(t) && (e = t, t = ""), b.func(e) && n(t))
        r[t] = r[t] || [], r[t].push(e);
      else if (b.array(e))
        for (var i = 0, o = e; i < o.length; i++) {
          var a = o[i];
          Pe(t, a, n, r);
        }
      else if (b.object(e))
        for (var s in e)
          Pe(gt(s).map(function(l) {
            return "".concat(t).concat(l);
          }), e[s], n, r);
      return r;
    }
    function gt(t) {
      return t.trim().split(/ +/);
    }
    var Ge = function(t, e) {
      return Math.sqrt(t * t + e * e);
    }, Dt = ["webkit", "moz"];
    function it(t, e) {
      t.__set || (t.__set = {});
      var n = function(i) {
        if (Dt.some(function(o) {
          return i.indexOf(o) === 0;
        }))
          return 1;
        typeof t[i] != "function" && i !== "__set" && Object.defineProperty(t, i, { get: function() {
          return i in t.__set ? t.__set[i] : t.__set[i] = e[i];
        }, set: function(o) {
          t.__set[i] = o;
        }, configurable: !0 });
      };
      for (var r in e)
        n(r);
      return t;
    }
    function ke(t, e) {
      t.page = t.page || {}, t.page.x = e.page.x, t.page.y = e.page.y, t.client = t.client || {}, t.client.x = e.client.x, t.client.y = e.client.y, t.timeStamp = e.timeStamp;
    }
    function S(t) {
      t.page.x = 0, t.page.y = 0, t.client.x = 0, t.client.y = 0;
    }
    function A(t) {
      return t instanceof q.Event || t instanceof q.Touch;
    }
    function N(t, e, n) {
      return t = t || "page", (n = n || {}).x = e[t + "X"], n.y = e[t + "Y"], n;
    }
    function X(t, e) {
      return e = e || { x: 0, y: 0 }, C.isOperaMobile && A(t) ? (N("screen", t, e), e.x += window.scrollX, e.y += window.scrollY) : N("page", t, e), e;
    }
    function Q(t) {
      return b.number(t.pointerId) ? t.pointerId : t.identifier;
    }
    function H(t, e, n) {
      var r = e.length > 1 ? se(e) : e[0];
      X(r, t.page), function(i, o) {
        o = o || {}, C.isOperaMobile && A(i) ? N("screen", i, o) : N("client", i, o);
      }(r, t.client), t.timeStamp = n;
    }
    function D(t) {
      var e = [];
      return b.array(t) ? (e[0] = t[0], e[1] = t[1]) : t.type === "touchend" ? t.touches.length === 1 ? (e[0] = t.touches[0], e[1] = t.changedTouches[0]) : t.touches.length === 0 && (e[0] = t.changedTouches[0], e[1] = t.changedTouches[1]) : (e[0] = t.touches[0], e[1] = t.touches[1]), e;
    }
    function se(t) {
      for (var e = { pageX: 0, pageY: 0, clientX: 0, clientY: 0, screenX: 0, screenY: 0 }, n = 0; n < t.length; n++) {
        var r = t[n];
        for (var i in e)
          e[i] += r[i];
      }
      for (var o in e)
        e[o] /= t.length;
      return e;
    }
    function fe(t) {
      if (!t.length)
        return null;
      var e = D(t), n = Math.min(e[0].pageX, e[1].pageX), r = Math.min(e[0].pageY, e[1].pageY), i = Math.max(e[0].pageX, e[1].pageX), o = Math.max(e[0].pageY, e[1].pageY);
      return { x: n, y: r, left: n, top: r, right: i, bottom: o, width: i - n, height: o - r };
    }
    function ye(t, e) {
      var n = e + "X", r = e + "Y", i = D(t), o = i[0][n] - i[1][n], a = i[0][r] - i[1][r];
      return Ge(o, a);
    }
    function Ie(t, e) {
      var n = e + "X", r = e + "Y", i = D(t), o = i[1][n] - i[0][n], a = i[1][r] - i[0][r];
      return 180 * Math.atan2(a, o) / Math.PI;
    }
    function Fe(t) {
      return b.string(t.pointerType) ? t.pointerType : b.number(t.pointerType) ? [void 0, void 0, "touch", "pen", "mouse"][t.pointerType] : /touch/.test(t.type || "") || t instanceof q.Touch ? "touch" : "mouse";
    }
    function ot(t) {
      var e = b.func(t.composedPath) ? t.composedPath() : t.path;
      return [et(e ? e[0] : t.target), et(t.currentTarget)];
    }
    var je = function() {
      function t(e) {
        w(this, t), this.immediatePropagationStopped = !1, this.propagationStopped = !1, this._interaction = e;
      }
      return g(t, [{ key: "preventDefault", value: function() {
      } }, { key: "stopPropagation", value: function() {
        this.propagationStopped = !0;
      } }, { key: "stopImmediatePropagation", value: function() {
        this.immediatePropagationStopped = this.propagationStopped = !0;
      } }]), t;
    }();
    Object.defineProperty(je.prototype, "interaction", { get: function() {
      return this._interaction._proxy;
    }, set: function() {
    } });
    var at = function(t, e) {
      for (var n = 0; n < e.length; n++) {
        var r = e[n];
        t.push(r);
      }
      return t;
    }, st = function(t) {
      return at([], t);
    }, vt = function(t, e) {
      for (var n = 0; n < t.length; n++)
        if (e(t[n], n, t))
          return n;
      return -1;
    }, mt = function(t, e) {
      return t[vt(t, e)];
    }, ct = function(t) {
      J(n, t);
      var e = ne(n);
      function n(r, i, o) {
        var a;
        w(this, n), (a = e.call(this, i._interaction)).dropzone = void 0, a.dragEvent = void 0, a.relatedTarget = void 0, a.draggable = void 0, a.propagationStopped = !1, a.immediatePropagationStopped = !1;
        var s = o === "dragleave" ? r.prev : r.cur, l = s.element, h = s.dropzone;
        return a.type = o, a.target = l, a.currentTarget = l, a.dropzone = h, a.dragEvent = i, a.relatedTarget = i.target, a.draggable = i.interactable, a.timeStamp = i.timeStamp, a;
      }
      return g(n, [{ key: "reject", value: function() {
        var r = this, i = this._interaction.dropState;
        if (this.type === "dropactivate" || this.dropzone && i.cur.dropzone === this.dropzone && i.cur.element === this.target)
          if (i.prev.dropzone = this.dropzone, i.prev.element = this.target, i.rejected = !0, i.events.enter = null, this.stopImmediatePropagation(), this.type === "dropactivate") {
            var o = i.activeDrops, a = vt(o, function(l) {
              var h = l.dropzone, u = l.element;
              return h === r.dropzone && u === r.target;
            });
            i.activeDrops.splice(a, 1);
            var s = new n(i, this.dragEvent, "dropdeactivate");
            s.dropzone = this.dropzone, s.target = this.target, this.dropzone.fire(s);
          } else
            this.dropzone.fire(new n(i, this.dragEvent, "dragleave"));
      } }, { key: "preventDefault", value: function() {
      } }, { key: "stopPropagation", value: function() {
        this.propagationStopped = !0;
      } }, { key: "stopImmediatePropagation", value: function() {
        this.immediatePropagationStopped = this.propagationStopped = !0;
      } }]), n;
    }(je);
    function Tn(t, e) {
      for (var n = 0, r = t.slice(); n < r.length; n++) {
        var i = r[n], o = i.dropzone, a = i.element;
        e.dropzone = o, e.target = a, o.fire(e), e.propagationStopped = e.immediatePropagationStopped = !1;
      }
    }
    function Vt(t, e) {
      for (var n = function(o, a) {
        for (var s = [], l = 0, h = o.interactables.list; l < h.length; l++) {
          var u = h[l];
          if (u.options.drop.enabled) {
            var m = u.options.drop.accept;
            if (!(b.element(m) && m !== a || b.string(m) && !G(a, m) || b.func(m) && !m({ dropzone: u, draggableElement: a })))
              for (var y = 0, E = u.getAllElements(); y < E.length; y++) {
                var x = E[y];
                x !== a && s.push({ dropzone: u, element: x, rect: u.getRect(x) });
              }
          }
        }
        return s;
      }(t, e), r = 0; r < n.length; r++) {
        var i = n[r];
        i.rect = i.dropzone.getRect(i.element);
      }
      return n;
    }
    function zn(t, e, n) {
      for (var r = t.dropState, i = t.interactable, o = t.element, a = [], s = 0, l = r.activeDrops; s < l.length; s++) {
        var h = l[s], u = h.dropzone, m = h.element, y = h.rect, E = u.dropCheck(e, n, i, o, m, y);
        a.push(E ? m : null);
      }
      var x = function(T) {
        for (var O, z, M, B = [], W = 0; W < T.length; W++) {
          var R = T[W], j = T[O];
          if (R && W !== O)
            if (j) {
              var pe = ze(R), Z = ze(j);
              if (pe !== R.ownerDocument)
                if (Z !== R.ownerDocument)
                  if (pe !== Z) {
                    B = B.length ? B : me(j);
                    var be = void 0;
                    if (j instanceof q.HTMLElement && R instanceof q.SVGElement && !(R instanceof q.SVGSVGElement)) {
                      if (R === Z)
                        continue;
                      be = R.ownerSVGElement;
                    } else
                      be = R;
                    for (var Se = me(be, j.ownerDocument), Be = 0; Se[Be] && Se[Be] === B[Be]; )
                      Be++;
                    var Wt = [Se[Be - 1], Se[Be], B[Be]];
                    if (Wt[0])
                      for (var kt = Wt[0].lastChild; kt; ) {
                        if (kt === Wt[1]) {
                          O = W, B = Se;
                          break;
                        }
                        if (kt === Wt[2])
                          break;
                        kt = kt.previousSibling;
                      }
                  } else
                    M = j, (parseInt(Y(z = R).getComputedStyle(z).zIndex, 10) || 0) >= (parseInt(Y(M).getComputedStyle(M).zIndex, 10) || 0) && (O = W);
                else
                  O = W;
            } else
              O = W;
        }
        return O;
      }(a);
      return r.activeDrops[x] || null;
    }
    function Kt(t, e, n) {
      var r = t.dropState, i = { enter: null, leave: null, activate: null, deactivate: null, move: null, drop: null };
      return n.type === "dragstart" && (i.activate = new ct(r, n, "dropactivate"), i.activate.target = null, i.activate.dropzone = null), n.type === "dragend" && (i.deactivate = new ct(r, n, "dropdeactivate"), i.deactivate.target = null, i.deactivate.dropzone = null), r.rejected || (r.cur.element !== r.prev.element && (r.prev.dropzone && (i.leave = new ct(r, n, "dragleave"), n.dragLeave = i.leave.target = r.prev.element, n.prevDropzone = i.leave.dropzone = r.prev.dropzone), r.cur.dropzone && (i.enter = new ct(r, n, "dragenter"), n.dragEnter = r.cur.element, n.dropzone = r.cur.dropzone)), n.type === "dragend" && r.cur.dropzone && (i.drop = new ct(r, n, "drop"), n.dropzone = r.cur.dropzone, n.relatedTarget = r.cur.element), n.type === "dragmove" && r.cur.dropzone && (i.move = new ct(r, n, "dropmove"), n.dropzone = r.cur.dropzone)), i;
    }
    function Ut(t, e) {
      var n = t.dropState, r = n.activeDrops, i = n.cur, o = n.prev;
      e.leave && o.dropzone.fire(e.leave), e.enter && i.dropzone.fire(e.enter), e.move && i.dropzone.fire(e.move), e.drop && i.dropzone.fire(e.drop), e.deactivate && Tn(r, e.deactivate), n.prev.dropzone = i.dropzone, n.prev.element = i.element;
    }
    function _n(t, e) {
      var n = t.interaction, r = t.iEvent, i = t.event;
      if (r.type === "dragmove" || r.type === "dragend") {
        var o = n.dropState;
        e.dynamicDrop && (o.activeDrops = Vt(e, n.element));
        var a = r, s = zn(n, a, i);
        o.rejected = o.rejected && !!s && s.dropzone === o.cur.dropzone && s.element === o.cur.element, o.cur.dropzone = s && s.dropzone, o.cur.element = s && s.element, o.events = Kt(n, 0, a);
      }
    }
    var On = { id: "actions/drop", install: function(t) {
      var e = t.actions, n = t.interactStatic, r = t.Interactable, i = t.defaults;
      t.usePlugin($), r.prototype.dropzone = function(o) {
        return function(a, s) {
          if (b.object(s)) {
            if (a.options.drop.enabled = s.enabled !== !1, s.listeners) {
              var l = Pe(s.listeners), h = Object.keys(l).reduce(function(m, y) {
                return m[/^(enter|leave)/.test(y) ? "drag".concat(y) : /^(activate|deactivate|move)/.test(y) ? "drop".concat(y) : y] = l[y], m;
              }, {}), u = a.options.drop.listeners;
              u && a.off(u), a.on(h), a.options.drop.listeners = h;
            }
            return b.func(s.ondrop) && a.on("drop", s.ondrop), b.func(s.ondropactivate) && a.on("dropactivate", s.ondropactivate), b.func(s.ondropdeactivate) && a.on("dropdeactivate", s.ondropdeactivate), b.func(s.ondragenter) && a.on("dragenter", s.ondragenter), b.func(s.ondragleave) && a.on("dragleave", s.ondragleave), b.func(s.ondropmove) && a.on("dropmove", s.ondropmove), /^(pointer|center)$/.test(s.overlap) ? a.options.drop.overlap = s.overlap : b.number(s.overlap) && (a.options.drop.overlap = Math.max(Math.min(1, s.overlap), 0)), "accept" in s && (a.options.drop.accept = s.accept), "checker" in s && (a.options.drop.checker = s.checker), a;
          }
          return b.bool(s) ? (a.options.drop.enabled = s, a) : a.options.drop;
        }(this, o);
      }, r.prototype.dropCheck = function(o, a, s, l, h, u) {
        return function(m, y, E, x, T, O, z) {
          var M = !1;
          if (!(z = z || m.getRect(O)))
            return !!m.options.drop.checker && m.options.drop.checker(y, E, M, m, O, x, T);
          var B = m.options.drop.overlap;
          if (B === "pointer") {
            var W = $e(x, T, "drag"), R = X(y);
            R.x += W.x, R.y += W.y;
            var j = R.x > z.left && R.x < z.right, pe = R.y > z.top && R.y < z.bottom;
            M = j && pe;
          }
          var Z = x.getRect(T);
          if (Z && B === "center") {
            var be = Z.left + Z.width / 2, Se = Z.top + Z.height / 2;
            M = be >= z.left && be <= z.right && Se >= z.top && Se <= z.bottom;
          }
          return Z && b.number(B) && (M = Math.max(0, Math.min(z.right, Z.right) - Math.max(z.left, Z.left)) * Math.max(0, Math.min(z.bottom, Z.bottom) - Math.max(z.top, Z.top)) / (Z.width * Z.height) >= B), m.options.drop.checker && (M = m.options.drop.checker(y, E, M, m, O, x, T)), M;
        }(this, o, a, s, l, h, u);
      }, n.dynamicDrop = function(o) {
        return b.bool(o) ? (t.dynamicDrop = o, n) : t.dynamicDrop;
      }, P(e.phaselessTypes, { dragenter: !0, dragleave: !0, dropactivate: !0, dropdeactivate: !0, dropmove: !0, drop: !0 }), e.methodDict.drop = "dropzone", t.dynamicDrop = !1, i.actions.drop = On.defaults;
    }, listeners: { "interactions:before-action-start": function(t) {
      var e = t.interaction;
      e.prepared.name === "drag" && (e.dropState = { cur: { dropzone: null, element: null }, prev: { dropzone: null, element: null }, rejected: null, events: null, activeDrops: [] });
    }, "interactions:after-action-start": function(t, e) {
      var n = t.interaction, r = (t.event, t.iEvent);
      if (n.prepared.name === "drag") {
        var i = n.dropState;
        i.activeDrops = [], i.events = {}, i.activeDrops = Vt(e, n.element), i.events = Kt(n, 0, r), i.events.activate && (Tn(i.activeDrops, i.events.activate), e.fire("actions/drop:start", { interaction: n, dragEvent: r }));
      }
    }, "interactions:action-move": _n, "interactions:after-action-move": function(t, e) {
      var n = t.interaction, r = t.iEvent;
      if (n.prepared.name === "drag") {
        var i = n.dropState;
        Ut(n, i.events), e.fire("actions/drop:move", { interaction: n, dragEvent: r }), i.events = {};
      }
    }, "interactions:action-end": function(t, e) {
      if (t.interaction.prepared.name === "drag") {
        var n = t.interaction, r = t.iEvent;
        _n(t, e), Ut(n, n.dropState.events), e.fire("actions/drop:end", { interaction: n, dragEvent: r });
      }
    }, "interactions:stop": function(t) {
      var e = t.interaction;
      if (e.prepared.name === "drag") {
        var n = e.dropState;
        n && (n.activeDrops = null, n.events = null, n.cur.dropzone = null, n.cur.element = null, n.prev.dropzone = null, n.prev.element = null, n.rejected = !1);
      }
    } }, getActiveDrops: Vt, getDrop: zn, getDropEvents: Kt, fireDropEvents: Ut, filterEventType: function(t) {
      return t.search("drag") === 0 || t.search("drop") === 0;
    }, defaults: { enabled: !1, accept: null, overlap: "pointer" } }, ni = On;
    function Jt(t) {
      var e = t.interaction, n = t.iEvent, r = t.phase;
      if (e.prepared.name === "gesture") {
        var i = e.pointers.map(function(h) {
          return h.pointer;
        }), o = r === "start", a = r === "end", s = e.interactable.options.deltaSource;
        if (n.touches = [i[0], i[1]], o)
          n.distance = ye(i, s), n.box = fe(i), n.scale = 1, n.ds = 0, n.angle = Ie(i, s), n.da = 0, e.gesture.startDistance = n.distance, e.gesture.startAngle = n.angle;
        else if (a || e.pointers.length < 2) {
          var l = e.prevEvent;
          n.distance = l.distance, n.box = l.box, n.scale = l.scale, n.ds = 0, n.angle = l.angle, n.da = 0;
        } else
          n.distance = ye(i, s), n.box = fe(i), n.scale = n.distance / e.gesture.startDistance, n.angle = Ie(i, s), n.ds = n.scale - e.gesture.scale, n.da = n.angle - e.gesture.angle;
        e.gesture.distance = n.distance, e.gesture.angle = n.angle, b.number(n.scale) && n.scale !== 1 / 0 && !isNaN(n.scale) && (e.gesture.scale = n.scale);
      }
    }
    var Qt = { id: "actions/gesture", before: ["actions/drag", "actions/resize"], install: function(t) {
      var e = t.actions, n = t.Interactable, r = t.defaults;
      n.prototype.gesturable = function(i) {
        return b.object(i) ? (this.options.gesture.enabled = i.enabled !== !1, this.setPerAction("gesture", i), this.setOnEvents("gesture", i), this) : b.bool(i) ? (this.options.gesture.enabled = i, this) : this.options.gesture;
      }, e.map.gesture = Qt, e.methodDict.gesture = "gesturable", r.actions.gesture = Qt.defaults;
    }, listeners: { "interactions:action-start": Jt, "interactions:action-move": Jt, "interactions:action-end": Jt, "interactions:new": function(t) {
      t.interaction.gesture = { angle: 0, distance: 0, scale: 1, startAngle: 0, startDistance: 0 };
    }, "auto-start:check": function(t) {
      if (!(t.interaction.pointers.length < 2)) {
        var e = t.interactable.options.gesture;
        if (e && e.enabled)
          return t.action = { name: "gesture" }, !1;
      }
    } }, defaults: {}, getCursor: function() {
      return "";
    }, filterEventType: function(t) {
      return t.search("gesture") === 0;
    } }, ri = Qt;
    function ii(t, e, n, r, i, o, a) {
      if (!e)
        return !1;
      if (e === !0) {
        var s = b.number(o.width) ? o.width : o.right - o.left, l = b.number(o.height) ? o.height : o.bottom - o.top;
        if (a = Math.min(a, Math.abs((t === "left" || t === "right" ? s : l) / 2)), s < 0 && (t === "left" ? t = "right" : t === "right" && (t = "left")), l < 0 && (t === "top" ? t = "bottom" : t === "bottom" && (t = "top")), t === "left") {
          var h = s >= 0 ? o.left : o.right;
          return n.x < h + a;
        }
        if (t === "top") {
          var u = l >= 0 ? o.top : o.bottom;
          return n.y < u + a;
        }
        if (t === "right")
          return n.x > (s >= 0 ? o.right : o.left) - a;
        if (t === "bottom")
          return n.y > (l >= 0 ? o.bottom : o.top) - a;
      }
      return !!b.element(r) && (b.element(e) ? e === r : dt(r, e, i));
    }
    function Pn(t) {
      var e = t.iEvent, n = t.interaction;
      if (n.prepared.name === "resize" && n.resizeAxes) {
        var r = e;
        n.interactable.options.resize.square ? (n.resizeAxes === "y" ? r.delta.x = r.delta.y : r.delta.y = r.delta.x, r.axes = "xy") : (r.axes = n.resizeAxes, n.resizeAxes === "x" ? r.delta.y = 0 : n.resizeAxes === "y" && (r.delta.x = 0));
      }
    }
    var Ce, Ve, Ae = { id: "actions/resize", before: ["actions/drag"], install: function(t) {
      var e = t.actions, n = t.browser, r = t.Interactable, i = t.defaults;
      Ae.cursors = function(o) {
        return o.isIe9 ? { x: "e-resize", y: "s-resize", xy: "se-resize", top: "n-resize", left: "w-resize", bottom: "s-resize", right: "e-resize", topleft: "se-resize", bottomright: "se-resize", topright: "ne-resize", bottomleft: "ne-resize" } : { x: "ew-resize", y: "ns-resize", xy: "nwse-resize", top: "ns-resize", left: "ew-resize", bottom: "ns-resize", right: "ew-resize", topleft: "nwse-resize", bottomright: "nwse-resize", topright: "nesw-resize", bottomleft: "nesw-resize" };
      }(n), Ae.defaultMargin = n.supportsTouch || n.supportsPointerEvent ? 20 : 10, r.prototype.resizable = function(o) {
        return function(a, s, l) {
          return b.object(s) ? (a.options.resize.enabled = s.enabled !== !1, a.setPerAction("resize", s), a.setOnEvents("resize", s), b.string(s.axis) && /^x$|^y$|^xy$/.test(s.axis) ? a.options.resize.axis = s.axis : s.axis === null && (a.options.resize.axis = l.defaults.actions.resize.axis), b.bool(s.preserveAspectRatio) ? a.options.resize.preserveAspectRatio = s.preserveAspectRatio : b.bool(s.square) && (a.options.resize.square = s.square), a) : b.bool(s) ? (a.options.resize.enabled = s, a) : a.options.resize;
        }(this, o, t);
      }, e.map.resize = Ae, e.methodDict.resize = "resizable", i.actions.resize = Ae.defaults;
    }, listeners: { "interactions:new": function(t) {
      t.interaction.resizeAxes = "xy";
    }, "interactions:action-start": function(t) {
      (function(e) {
        var n = e.iEvent, r = e.interaction;
        if (r.prepared.name === "resize" && r.prepared.edges) {
          var i = n, o = r.rect;
          r._rects = { start: P({}, o), corrected: P({}, o), previous: P({}, o), delta: { left: 0, right: 0, width: 0, top: 0, bottom: 0, height: 0 } }, i.edges = r.prepared.edges, i.rect = r._rects.corrected, i.deltaRect = r._rects.delta;
        }
      })(t), Pn(t);
    }, "interactions:action-move": function(t) {
      (function(e) {
        var n = e.iEvent, r = e.interaction;
        if (r.prepared.name === "resize" && r.prepared.edges) {
          var i = n, o = r.interactable.options.resize.invert, a = o === "reposition" || o === "negate", s = r.rect, l = r._rects, h = l.start, u = l.corrected, m = l.delta, y = l.previous;
          if (P(y, u), a) {
            if (P(u, s), o === "reposition") {
              if (u.top > u.bottom) {
                var E = u.top;
                u.top = u.bottom, u.bottom = E;
              }
              if (u.left > u.right) {
                var x = u.left;
                u.left = u.right, u.right = x;
              }
            }
          } else
            u.top = Math.min(s.top, h.bottom), u.bottom = Math.max(s.bottom, h.top), u.left = Math.min(s.left, h.right), u.right = Math.max(s.right, h.left);
          for (var T in u.width = u.right - u.left, u.height = u.bottom - u.top, u)
            m[T] = u[T] - y[T];
          i.edges = r.prepared.edges, i.rect = u, i.deltaRect = m;
        }
      })(t), Pn(t);
    }, "interactions:action-end": function(t) {
      var e = t.iEvent, n = t.interaction;
      if (n.prepared.name === "resize" && n.prepared.edges) {
        var r = e;
        r.edges = n.prepared.edges, r.rect = n._rects.corrected, r.deltaRect = n._rects.delta;
      }
    }, "auto-start:check": function(t) {
      var e = t.interaction, n = t.interactable, r = t.element, i = t.rect, o = t.buttons;
      if (i) {
        var a = P({}, e.coords.cur.page), s = n.options.resize;
        if (s && s.enabled && (!e.pointerIsDown || !/mouse|pointer/.test(e.pointerType) || o & s.mouseButtons)) {
          if (b.object(s.edges)) {
            var l = { left: !1, right: !1, top: !1, bottom: !1 };
            for (var h in l)
              l[h] = ii(h, s.edges[h], a, e._latestPointer.eventTarget, r, i, s.margin || Ae.defaultMargin);
            l.left = l.left && !l.right, l.top = l.top && !l.bottom, (l.left || l.right || l.top || l.bottom) && (t.action = { name: "resize", edges: l });
          } else {
            var u = s.axis !== "y" && a.x > i.right - Ae.defaultMargin, m = s.axis !== "x" && a.y > i.bottom - Ae.defaultMargin;
            (u || m) && (t.action = { name: "resize", axes: (u ? "x" : "") + (m ? "y" : "") });
          }
          return !t.action && void 0;
        }
      }
    } }, defaults: { square: !1, preserveAspectRatio: !1, axis: "xy", margin: NaN, edges: null, invert: "none" }, cursors: null, getCursor: function(t) {
      var e = t.edges, n = t.axis, r = t.name, i = Ae.cursors, o = null;
      if (n)
        o = i[r + n];
      else if (e) {
        for (var a = "", s = 0, l = ["top", "bottom", "left", "right"]; s < l.length; s++) {
          var h = l[s];
          e[h] && (a += h);
        }
        o = i[a];
      }
      return o;
    }, filterEventType: function(t) {
      return t.search("resize") === 0;
    }, defaultMargin: null }, oi = Ae, ai = { id: "actions", install: function(t) {
      t.usePlugin(ri), t.usePlugin(oi), t.usePlugin($), t.usePlugin(ni);
    } }, Mn = 0, Le = { request: function(t) {
      return Ce(t);
    }, cancel: function(t) {
      return Ve(t);
    }, init: function(t) {
      if (Ce = t.requestAnimationFrame, Ve = t.cancelAnimationFrame, !Ce)
        for (var e = ["ms", "moz", "webkit", "o"], n = 0; n < e.length; n++) {
          var r = e[n];
          Ce = t["".concat(r, "RequestAnimationFrame")], Ve = t["".concat(r, "CancelAnimationFrame")] || t["".concat(r, "CancelRequestAnimationFrame")];
        }
      Ce = Ce && Ce.bind(t), Ve = Ve && Ve.bind(t), Ce || (Ce = function(i) {
        var o = Date.now(), a = Math.max(0, 16 - (o - Mn)), s = t.setTimeout(function() {
          i(o + a);
        }, a);
        return Mn = o + a, s;
      }, Ve = function(i) {
        return clearTimeout(i);
      });
    } }, I = { defaults: { enabled: !1, margin: 60, container: null, speed: 300 }, now: Date.now, interaction: null, i: 0, x: 0, y: 0, isScrolling: !1, prevTime: 0, margin: 0, speed: 0, start: function(t) {
      I.isScrolling = !0, Le.cancel(I.i), t.autoScroll = I, I.interaction = t, I.prevTime = I.now(), I.i = Le.request(I.scroll);
    }, stop: function() {
      I.isScrolling = !1, I.interaction && (I.interaction.autoScroll = null), Le.cancel(I.i);
    }, scroll: function() {
      var t = I.interaction, e = t.interactable, n = t.element, r = t.prepared.name, i = e.options[r].autoScroll, o = Dn(i.container, e, n), a = I.now(), s = (a - I.prevTime) / 1e3, l = i.speed * s;
      if (l >= 1) {
        var h = { x: I.x * l, y: I.y * l };
        if (h.x || h.y) {
          var u = Rn(o);
          b.window(o) ? o.scrollBy(h.x, h.y) : o && (o.scrollLeft += h.x, o.scrollTop += h.y);
          var m = Rn(o), y = { x: m.x - u.x, y: m.y - u.y };
          (y.x || y.y) && e.fire({ type: "autoscroll", target: n, interactable: e, delta: y, interaction: t, container: o });
        }
        I.prevTime = a;
      }
      I.isScrolling && (Le.cancel(I.i), I.i = Le.request(I.scroll));
    }, check: function(t, e) {
      var n;
      return (n = t.options[e].autoScroll) == null ? void 0 : n.enabled;
    }, onInteractionMove: function(t) {
      var e = t.interaction, n = t.pointer;
      if (e.interacting() && I.check(e.interactable, e.prepared.name))
        if (e.simulation)
          I.x = I.y = 0;
        else {
          var r, i, o, a, s = e.interactable, l = e.element, h = e.prepared.name, u = s.options[h].autoScroll, m = Dn(u.container, s, l);
          if (b.window(m))
            a = n.clientX < I.margin, r = n.clientY < I.margin, i = n.clientX > m.innerWidth - I.margin, o = n.clientY > m.innerHeight - I.margin;
          else {
            var y = _e(m);
            a = n.clientX < y.left + I.margin, r = n.clientY < y.top + I.margin, i = n.clientX > y.right - I.margin, o = n.clientY > y.bottom - I.margin;
          }
          I.x = i ? 1 : a ? -1 : 0, I.y = o ? 1 : r ? -1 : 0, I.isScrolling || (I.margin = u.margin, I.speed = u.speed, I.start(e));
        }
    } };
    function Dn(t, e, n) {
      return (b.string(t) ? tt(t, e, n) : t) || Y(n);
    }
    function Rn(t) {
      return b.window(t) && (t = window.document.body), { x: t.scrollLeft, y: t.scrollTop };
    }
    var si = { id: "auto-scroll", install: function(t) {
      var e = t.defaults, n = t.actions;
      t.autoScroll = I, I.now = function() {
        return t.now();
      }, n.phaselessTypes.autoscroll = !0, e.perAction.autoScroll = I.defaults;
    }, listeners: { "interactions:new": function(t) {
      t.interaction.autoScroll = null;
    }, "interactions:destroy": function(t) {
      t.interaction.autoScroll = null, I.stop(), I.interaction && (I.interaction = null);
    }, "interactions:stop": I.stop, "interactions:action-move": function(t) {
      return I.onInteractionMove(t);
    } } }, ci = si;
    function yt(t, e) {
      var n = !1;
      return function() {
        return n || (V.console.warn(e), n = !0), t.apply(this, arguments);
      };
    }
    function Zt(t, e) {
      return t.name = e.name, t.axis = e.axis, t.edges = e.edges, t;
    }
    function li(t) {
      return b.bool(t) ? (this.options.styleCursor = t, this) : t === null ? (delete this.options.styleCursor, this) : this.options.styleCursor;
    }
    function ui(t) {
      return b.func(t) ? (this.options.actionChecker = t, this) : t === null ? (delete this.options.actionChecker, this) : this.options.actionChecker;
    }
    var fi = { id: "auto-start/interactableMethods", install: function(t) {
      var e = t.Interactable;
      e.prototype.getAction = function(n, r, i, o) {
        var a = function(s, l, h, u, m) {
          var y = s.getRect(u), E = l.buttons || { 0: 1, 1: 4, 3: 8, 4: 16 }[l.button], x = { action: null, interactable: s, interaction: h, element: u, rect: y, buttons: E };
          return m.fire("auto-start:check", x), x.action;
        }(this, r, i, o, t);
        return this.options.actionChecker ? this.options.actionChecker(n, r, a, this, o, i) : a;
      }, e.prototype.ignoreFrom = yt(function(n) {
        return this._backCompatOption("ignoreFrom", n);
      }, "Interactable.ignoreFrom() has been deprecated. Use Interactble.draggable({ignoreFrom: newValue})."), e.prototype.allowFrom = yt(function(n) {
        return this._backCompatOption("allowFrom", n);
      }, "Interactable.allowFrom() has been deprecated. Use Interactble.draggable({allowFrom: newValue})."), e.prototype.actionChecker = ui, e.prototype.styleCursor = li;
    } };
    function In(t, e, n, r, i) {
      return e.testIgnoreAllow(e.options[t.name], n, r) && e.options[t.name].enabled && Rt(e, n, t, i) ? t : null;
    }
    function di(t, e, n, r, i, o, a) {
      for (var s = 0, l = r.length; s < l; s++) {
        var h = r[s], u = i[s], m = h.getAction(e, n, t, u);
        if (m) {
          var y = In(m, h, u, o, a);
          if (y)
            return { action: y, interactable: h, element: u };
        }
      }
      return { action: null, interactable: null, element: null };
    }
    function Cn(t, e, n, r, i) {
      var o = [], a = [], s = r;
      function l(u) {
        o.push(u), a.push(s);
      }
      for (; b.element(s); ) {
        o = [], a = [], i.interactables.forEachMatch(s, l);
        var h = di(t, e, n, o, a, r, i);
        if (h.action && !h.interactable.options[h.action.name].manualStart)
          return h;
        s = te(s);
      }
      return { action: null, interactable: null, element: null };
    }
    function An(t, e, n) {
      var r = e.action, i = e.interactable, o = e.element;
      r = r || { name: null }, t.interactable = i, t.element = o, Zt(t.prepared, r), t.rect = i && r.name ? i.getRect(o) : null, Fn(t, n), n.fire("autoStart:prepared", { interaction: t });
    }
    function Rt(t, e, n, r) {
      var i = t.options, o = i[n.name].max, a = i[n.name].maxPerElement, s = r.autoStart.maxInteractions, l = 0, h = 0, u = 0;
      if (!(o && a && s))
        return !1;
      for (var m = 0, y = r.interactions.list; m < y.length; m++) {
        var E = y[m], x = E.prepared.name;
        if (E.interacting() && (++l >= s || E.interactable === t && ((h += x === n.name ? 1 : 0) >= o || E.element === e && (u++, x === n.name && u >= a))))
          return !1;
      }
      return s > 0;
    }
    function Bn(t, e) {
      return b.number(t) ? (e.autoStart.maxInteractions = t, this) : e.autoStart.maxInteractions;
    }
    function en(t, e, n) {
      var r = n.autoStart.cursorElement;
      r && r !== t && (r.style.cursor = ""), t.ownerDocument.documentElement.style.cursor = e, t.style.cursor = e, n.autoStart.cursorElement = e ? t : null;
    }
    function Fn(t, e) {
      var n = t.interactable, r = t.element, i = t.prepared;
      if (t.pointerType === "mouse" && n && n.options.styleCursor) {
        var o = "";
        if (i.name) {
          var a = n.options[i.name].cursorChecker;
          o = b.func(a) ? a(i, n, r, t._interacting) : e.actions.map[i.name].getCursor(i);
        }
        en(t.element, o || "", e);
      } else
        e.autoStart.cursorElement && en(e.autoStart.cursorElement, "", e);
    }
    var pi = { id: "auto-start/base", before: ["actions"], install: function(t) {
      var e = t.interactStatic, n = t.defaults;
      t.usePlugin(fi), n.base.actionChecker = null, n.base.styleCursor = !0, P(n.perAction, { manualStart: !1, max: 1 / 0, maxPerElement: 1, allowFrom: null, ignoreFrom: null, mouseButtons: 1 }), e.maxInteractions = function(r) {
        return Bn(r, t);
      }, t.autoStart = { maxInteractions: 1 / 0, withinInteractionLimit: Rt, cursorElement: null };
    }, listeners: { "interactions:down": function(t, e) {
      var n = t.interaction, r = t.pointer, i = t.event, o = t.eventTarget;
      n.interacting() || An(n, Cn(n, r, i, o, e), e);
    }, "interactions:move": function(t, e) {
      (function(n, r) {
        var i = n.interaction, o = n.pointer, a = n.event, s = n.eventTarget;
        i.pointerType !== "mouse" || i.pointerIsDown || i.interacting() || An(i, Cn(i, o, a, s, r), r);
      })(t, e), function(n, r) {
        var i = n.interaction;
        if (i.pointerIsDown && !i.interacting() && i.pointerWasMoved && i.prepared.name) {
          r.fire("autoStart:before-start", n);
          var o = i.interactable, a = i.prepared.name;
          a && o && (o.options[a].manualStart || !Rt(o, i.element, i.prepared, r) ? i.stop() : (i.start(i.prepared, o, i.element), Fn(i, r)));
        }
      }(t, e);
    }, "interactions:stop": function(t, e) {
      var n = t.interaction, r = n.interactable;
      r && r.options.styleCursor && en(n.element, "", e);
    } }, maxInteractions: Bn, withinInteractionLimit: Rt, validateAction: In }, tn = pi, hi = { id: "auto-start/dragAxis", listeners: { "autoStart:before-start": function(t, e) {
      var n = t.interaction, r = t.eventTarget, i = t.dx, o = t.dy;
      if (n.prepared.name === "drag") {
        var a = Math.abs(i), s = Math.abs(o), l = n.interactable.options.drag, h = l.startAxis, u = a > s ? "x" : a < s ? "y" : "xy";
        if (n.prepared.axis = l.lockAxis === "start" ? u[0] : l.lockAxis, u !== "xy" && h !== "xy" && h !== u) {
          n.prepared.name = null;
          for (var m = r, y = function(x) {
            if (x !== n.interactable) {
              var T = n.interactable.options.drag;
              if (!T.manualStart && x.testIgnoreAllow(T, m, r)) {
                var O = x.getAction(n.downPointer, n.downEvent, n, m);
                if (O && O.name === "drag" && function(z, M) {
                  if (!M)
                    return !1;
                  var B = M.options.drag.startAxis;
                  return z === "xy" || B === "xy" || B === z;
                }(u, x) && tn.validateAction(O, x, m, r, e))
                  return x;
              }
            }
          }; b.element(m); ) {
            var E = e.interactables.forEachMatch(m, y);
            if (E) {
              n.prepared.name = "drag", n.interactable = E, n.element = m;
              break;
            }
            m = te(m);
          }
        }
      }
    } } };
    function nn(t) {
      var e = t.prepared && t.prepared.name;
      if (!e)
        return null;
      var n = t.interactable.options;
      return n[e].hold || n[e].delay;
    }
    var gi = { id: "auto-start/hold", install: function(t) {
      var e = t.defaults;
      t.usePlugin(tn), e.perAction.hold = 0, e.perAction.delay = 0;
    }, listeners: { "interactions:new": function(t) {
      t.interaction.autoStartHoldTimer = null;
    }, "autoStart:prepared": function(t) {
      var e = t.interaction, n = nn(e);
      n > 0 && (e.autoStartHoldTimer = setTimeout(function() {
        e.start(e.prepared, e.interactable, e.element);
      }, n));
    }, "interactions:move": function(t) {
      var e = t.interaction, n = t.duplicate;
      e.autoStartHoldTimer && e.pointerWasMoved && !n && (clearTimeout(e.autoStartHoldTimer), e.autoStartHoldTimer = null);
    }, "autoStart:before-start": function(t) {
      var e = t.interaction;
      nn(e) > 0 && (e.prepared.name = null);
    } }, getHoldDuration: nn }, vi = gi, mi = { id: "auto-start", install: function(t) {
      t.usePlugin(tn), t.usePlugin(vi), t.usePlugin(hi);
    } }, yi = function(t) {
      return /^(always|never|auto)$/.test(t) ? (this.options.preventDefault = t, this) : b.bool(t) ? (this.options.preventDefault = t ? "always" : "never", this) : this.options.preventDefault;
    };
    function bi(t) {
      var e = t.interaction, n = t.event;
      e.interactable && e.interactable.checkAndPreventDefault(n);
    }
    var jn = { id: "core/interactablePreventDefault", install: function(t) {
      var e = t.Interactable;
      e.prototype.preventDefault = yi, e.prototype.checkAndPreventDefault = function(n) {
        return function(r, i, o) {
          var a = r.options.preventDefault;
          if (a !== "never")
            if (a !== "always") {
              if (i.events.supportsPassive && /^touch(start|move)$/.test(o.type)) {
                var s = Y(o.target).document, l = i.getDocOptions(s);
                if (!l || !l.events || l.events.passive !== !1)
                  return;
              }
              /^(mouse|pointer|touch)*(down|start)/i.test(o.type) || b.element(o.target) && G(o.target, "input,select,textarea,[contenteditable=true],[contenteditable=true] *") || o.preventDefault();
            } else
              o.preventDefault();
        }(this, t, n);
      }, t.interactions.docEvents.push({ type: "dragstart", listener: function(n) {
        for (var r = 0, i = t.interactions.list; r < i.length; r++) {
          var o = i[r];
          if (o.element && (o.element === n.target || le(o.element, n.target)))
            return void o.interactable.checkAndPreventDefault(n);
        }
      } });
    }, listeners: ["down", "move", "up", "cancel"].reduce(function(t, e) {
      return t["interactions:".concat(e)] = bi, t;
    }, {}) };
    function It(t, e) {
      if (e.phaselessTypes[t])
        return !0;
      for (var n in e.map)
        if (t.indexOf(n) === 0 && t.substr(n.length) in e.phases)
          return !0;
      return !1;
    }
    function lt(t) {
      var e = {};
      for (var n in t) {
        var r = t[n];
        b.plainObject(r) ? e[n] = lt(r) : b.array(r) ? e[n] = st(r) : e[n] = r;
      }
      return e;
    }
    var rn = function() {
      function t(e) {
        w(this, t), this.states = [], this.startOffset = { left: 0, right: 0, top: 0, bottom: 0 }, this.startDelta = void 0, this.result = void 0, this.endResult = void 0, this.startEdges = void 0, this.edges = void 0, this.interaction = void 0, this.interaction = e, this.result = Ct(), this.edges = { left: !1, right: !1, top: !1, bottom: !1 };
      }
      return g(t, [{ key: "start", value: function(e, n) {
        var r, i, o = e.phase, a = this.interaction, s = function(h) {
          var u = h.interactable.options[h.prepared.name], m = u.modifiers;
          return m && m.length ? m : ["snap", "snapSize", "snapEdges", "restrict", "restrictEdges", "restrictSize"].map(function(y) {
            var E = u[y];
            return E && E.enabled && { options: E, methods: E._methods };
          }).filter(function(y) {
            return !!y;
          });
        }(a);
        this.prepareStates(s), this.startEdges = P({}, a.edges), this.edges = P({}, this.startEdges), this.startOffset = (r = a.rect, i = n, r ? { left: i.x - r.left, top: i.y - r.top, right: r.right - i.x, bottom: r.bottom - i.y } : { left: 0, top: 0, right: 0, bottom: 0 }), this.startDelta = { x: 0, y: 0 };
        var l = this.fillArg({ phase: o, pageCoords: n, preEnd: !1 });
        return this.result = Ct(), this.startAll(l), this.result = this.setAll(l);
      } }, { key: "fillArg", value: function(e) {
        var n = this.interaction;
        return e.interaction = n, e.interactable = n.interactable, e.element = n.element, e.rect || (e.rect = n.rect), e.edges || (e.edges = this.startEdges), e.startOffset = this.startOffset, e;
      } }, { key: "startAll", value: function(e) {
        for (var n = 0, r = this.states; n < r.length; n++) {
          var i = r[n];
          i.methods.start && (e.state = i, i.methods.start(e));
        }
      } }, { key: "setAll", value: function(e) {
        var n = e.phase, r = e.preEnd, i = e.skipModifiers, o = e.rect, a = e.edges;
        e.coords = P({}, e.pageCoords), e.rect = P({}, o), e.edges = P({}, a);
        for (var s = i ? this.states.slice(i) : this.states, l = Ct(e.coords, e.rect), h = 0; h < s.length; h++) {
          var u, m = s[h], y = m.options, E = P({}, e.coords), x = null;
          (u = m.methods) != null && u.set && this.shouldDo(y, r, n) && (e.state = m, x = m.methods.set(e), rt(e.edges, e.rect, { x: e.coords.x - E.x, y: e.coords.y - E.y })), l.eventProps.push(x);
        }
        P(this.edges, e.edges), l.delta.x = e.coords.x - e.pageCoords.x, l.delta.y = e.coords.y - e.pageCoords.y, l.rectDelta.left = e.rect.left - o.left, l.rectDelta.right = e.rect.right - o.right, l.rectDelta.top = e.rect.top - o.top, l.rectDelta.bottom = e.rect.bottom - o.bottom;
        var T = this.result.coords, O = this.result.rect;
        if (T && O) {
          var z = l.rect.left !== O.left || l.rect.right !== O.right || l.rect.top !== O.top || l.rect.bottom !== O.bottom;
          l.changed = z || T.x !== l.coords.x || T.y !== l.coords.y;
        }
        return l;
      } }, { key: "applyToInteraction", value: function(e) {
        var n = this.interaction, r = e.phase, i = n.coords.cur, o = n.coords.start, a = this.result, s = this.startDelta, l = a.delta;
        r === "start" && P(this.startDelta, a.delta);
        for (var h = 0, u = [[o, s], [i, l]]; h < u.length; h++) {
          var m = u[h], y = m[0], E = m[1];
          y.page.x += E.x, y.page.y += E.y, y.client.x += E.x, y.client.y += E.y;
        }
        var x = this.result.rectDelta, T = e.rect || n.rect;
        T.left += x.left, T.right += x.right, T.top += x.top, T.bottom += x.bottom, T.width = T.right - T.left, T.height = T.bottom - T.top;
      } }, { key: "setAndApply", value: function(e) {
        var n = this.interaction, r = e.phase, i = e.preEnd, o = e.skipModifiers, a = this.setAll(this.fillArg({ preEnd: i, phase: r, pageCoords: e.modifiedCoords || n.coords.cur.page }));
        if (this.result = a, !a.changed && (!o || o < this.states.length) && n.interacting())
          return !1;
        if (e.modifiedCoords) {
          var s = n.coords.cur.page, l = { x: e.modifiedCoords.x - s.x, y: e.modifiedCoords.y - s.y };
          a.coords.x += l.x, a.coords.y += l.y, a.delta.x += l.x, a.delta.y += l.y;
        }
        this.applyToInteraction(e);
      } }, { key: "beforeEnd", value: function(e) {
        var n = e.interaction, r = e.event, i = this.states;
        if (i && i.length) {
          for (var o = !1, a = 0; a < i.length; a++) {
            var s = i[a];
            e.state = s;
            var l = s.options, h = s.methods, u = h.beforeEnd && h.beforeEnd(e);
            if (u)
              return this.endResult = u, !1;
            o = o || !o && this.shouldDo(l, !0, e.phase, !0);
          }
          o && n.move({ event: r, preEnd: !0 });
        }
      } }, { key: "stop", value: function(e) {
        var n = e.interaction;
        if (this.states && this.states.length) {
          var r = P({ states: this.states, interactable: n.interactable, element: n.element, rect: null }, e);
          this.fillArg(r);
          for (var i = 0, o = this.states; i < o.length; i++) {
            var a = o[i];
            r.state = a, a.methods.stop && a.methods.stop(r);
          }
          this.states = null, this.endResult = null;
        }
      } }, { key: "prepareStates", value: function(e) {
        this.states = [];
        for (var n = 0; n < e.length; n++) {
          var r = e[n], i = r.options, o = r.methods, a = r.name;
          this.states.push({ options: i, methods: o, index: n, name: a });
        }
        return this.states;
      } }, { key: "restoreInteractionCoords", value: function(e) {
        var n = e.interaction, r = n.coords, i = n.rect, o = n.modification;
        if (o.result) {
          for (var a = o.startDelta, s = o.result, l = s.delta, h = s.rectDelta, u = 0, m = [[r.start, a], [r.cur, l]]; u < m.length; u++) {
            var y = m[u], E = y[0], x = y[1];
            E.page.x -= x.x, E.page.y -= x.y, E.client.x -= x.x, E.client.y -= x.y;
          }
          i.left -= h.left, i.right -= h.right, i.top -= h.top, i.bottom -= h.bottom;
        }
      } }, { key: "shouldDo", value: function(e, n, r, i) {
        return !(!e || e.enabled === !1 || i && !e.endOnly || e.endOnly && !n || r === "start" && !e.setStart);
      } }, { key: "copyFrom", value: function(e) {
        this.startOffset = e.startOffset, this.startDelta = e.startDelta, this.startEdges = e.startEdges, this.edges = e.edges, this.states = e.states.map(function(n) {
          return lt(n);
        }), this.result = Ct(P({}, e.result.coords), P({}, e.result.rect));
      } }, { key: "destroy", value: function() {
        for (var e in this)
          this[e] = null;
      } }]), t;
    }();
    function Ct(t, e) {
      return { rect: e, coords: t, delta: { x: 0, y: 0 }, rectDelta: { left: 0, right: 0, top: 0, bottom: 0 }, eventProps: [], changed: !0 };
    }
    function We(t, e) {
      var n = t.defaults, r = { start: t.start, set: t.set, beforeEnd: t.beforeEnd, stop: t.stop }, i = function(o) {
        var a = o || {};
        for (var s in a.enabled = a.enabled !== !1, n)
          s in a || (a[s] = n[s]);
        var l = { options: a, methods: r, name: e, enable: function() {
          return a.enabled = !0, l;
        }, disable: function() {
          return a.enabled = !1, l;
        } };
        return l;
      };
      return e && typeof e == "string" && (i._defaults = n, i._methods = r), i;
    }
    function bt(t) {
      var e = t.iEvent, n = t.interaction.modification.result;
      n && (e.modifiers = n.eventProps);
    }
    var wi = { id: "modifiers/base", before: ["actions"], install: function(t) {
      t.defaults.perAction.modifiers = [];
    }, listeners: { "interactions:new": function(t) {
      var e = t.interaction;
      e.modification = new rn(e);
    }, "interactions:before-action-start": function(t) {
      var e = t.interaction, n = t.interaction.modification;
      n.start(t, e.coords.start.page), e.edges = n.edges, n.applyToInteraction(t);
    }, "interactions:before-action-move": function(t) {
      var e = t.interaction, n = e.modification, r = n.setAndApply(t);
      return e.edges = n.edges, r;
    }, "interactions:before-action-end": function(t) {
      var e = t.interaction, n = e.modification, r = n.beforeEnd(t);
      return e.edges = n.startEdges, r;
    }, "interactions:action-start": bt, "interactions:action-move": bt, "interactions:action-end": bt, "interactions:after-action-start": function(t) {
      return t.interaction.modification.restoreInteractionCoords(t);
    }, "interactions:after-action-move": function(t) {
      return t.interaction.modification.restoreInteractionCoords(t);
    }, "interactions:stop": function(t) {
      return t.interaction.modification.stop(t);
    } } }, Ln = wi, Wn = { base: { preventDefault: "auto", deltaSource: "page" }, perAction: { enabled: !1, origin: { x: 0, y: 0 } }, actions: {} }, on = function(t) {
      J(n, t);
      var e = ne(n);
      function n(r, i, o, a, s, l, h) {
        var u;
        w(this, n), (u = e.call(this, r)).relatedTarget = null, u.screenX = void 0, u.screenY = void 0, u.button = void 0, u.buttons = void 0, u.ctrlKey = void 0, u.shiftKey = void 0, u.altKey = void 0, u.metaKey = void 0, u.page = void 0, u.client = void 0, u.delta = void 0, u.rect = void 0, u.x0 = void 0, u.y0 = void 0, u.t0 = void 0, u.dt = void 0, u.duration = void 0, u.clientX0 = void 0, u.clientY0 = void 0, u.velocity = void 0, u.speed = void 0, u.swipe = void 0, u.axes = void 0, u.preEnd = void 0, s = s || r.element;
        var m = r.interactable, y = (m && m.options || Wn).deltaSource, E = $e(m, s, o), x = a === "start", T = a === "end", O = x ? ce(u) : r.prevEvent, z = x ? r.coords.start : T ? { page: O.page, client: O.client, timeStamp: r.coords.cur.timeStamp } : r.coords.cur;
        return u.page = P({}, z.page), u.client = P({}, z.client), u.rect = P({}, r.rect), u.timeStamp = z.timeStamp, T || (u.page.x -= E.x, u.page.y -= E.y, u.client.x -= E.x, u.client.y -= E.y), u.ctrlKey = i.ctrlKey, u.altKey = i.altKey, u.shiftKey = i.shiftKey, u.metaKey = i.metaKey, u.button = i.button, u.buttons = i.buttons, u.target = s, u.currentTarget = s, u.preEnd = l, u.type = h || o + (a || ""), u.interactable = m, u.t0 = x ? r.pointers[r.pointers.length - 1].downTime : O.t0, u.x0 = r.coords.start.page.x - E.x, u.y0 = r.coords.start.page.y - E.y, u.clientX0 = r.coords.start.client.x - E.x, u.clientY0 = r.coords.start.client.y - E.y, u.delta = x || T ? { x: 0, y: 0 } : { x: u[y].x - O[y].x, y: u[y].y - O[y].y }, u.dt = r.coords.delta.timeStamp, u.duration = u.timeStamp - u.t0, u.velocity = P({}, r.coords.velocity[y]), u.speed = Ge(u.velocity.x, u.velocity.y), u.swipe = T || a === "inertiastart" ? u.getSwipe() : null, u;
      }
      return g(n, [{ key: "getSwipe", value: function() {
        var r = this._interaction;
        if (r.prevEvent.speed < 600 || this.timeStamp - r.prevEvent.timeStamp > 150)
          return null;
        var i = 180 * Math.atan2(r.prevEvent.velocityY, r.prevEvent.velocityX) / Math.PI;
        i < 0 && (i += 360);
        var o = 112.5 <= i && i < 247.5, a = 202.5 <= i && i < 337.5;
        return { up: a, down: !a && 22.5 <= i && i < 157.5, left: o, right: !o && (292.5 <= i || i < 67.5), angle: i, speed: r.prevEvent.speed, velocity: { x: r.prevEvent.velocityX, y: r.prevEvent.velocityY } };
      } }, { key: "preventDefault", value: function() {
      } }, { key: "stopImmediatePropagation", value: function() {
        this.immediatePropagationStopped = this.propagationStopped = !0;
      } }, { key: "stopPropagation", value: function() {
        this.propagationStopped = !0;
      } }]), n;
    }(je);
    Object.defineProperties(on.prototype, { pageX: { get: function() {
      return this.page.x;
    }, set: function(t) {
      this.page.x = t;
    } }, pageY: { get: function() {
      return this.page.y;
    }, set: function(t) {
      this.page.y = t;
    } }, clientX: { get: function() {
      return this.client.x;
    }, set: function(t) {
      this.client.x = t;
    } }, clientY: { get: function() {
      return this.client.y;
    }, set: function(t) {
      this.client.y = t;
    } }, dx: { get: function() {
      return this.delta.x;
    }, set: function(t) {
      this.delta.x = t;
    } }, dy: { get: function() {
      return this.delta.y;
    }, set: function(t) {
      this.delta.y = t;
    } }, velocityX: { get: function() {
      return this.velocity.x;
    }, set: function(t) {
      this.velocity.x = t;
    } }, velocityY: { get: function() {
      return this.velocity.y;
    }, set: function(t) {
      this.velocity.y = t;
    } } });
    var xi = g(function t(e, n, r, i, o) {
      w(this, t), this.id = void 0, this.pointer = void 0, this.event = void 0, this.downTime = void 0, this.downTarget = void 0, this.id = e, this.pointer = n, this.event = r, this.downTime = i, this.downTarget = o;
    }), Ei = function(t) {
      return t.interactable = "", t.element = "", t.prepared = "", t.pointerIsDown = "", t.pointerWasMoved = "", t._proxy = "", t;
    }({}), Hn = function(t) {
      return t.start = "", t.move = "", t.end = "", t.stop = "", t.interacting = "", t;
    }({}), ki = 0, Si = function() {
      function t(e) {
        var n = this, r = e.pointerType, i = e.scopeFire;
        w(this, t), this.interactable = null, this.element = null, this.rect = null, this._rects = void 0, this.edges = null, this._scopeFire = void 0, this.prepared = { name: null, axis: null, edges: null }, this.pointerType = void 0, this.pointers = [], this.downEvent = null, this.downPointer = {}, this._latestPointer = { pointer: null, event: null, eventTarget: null }, this.prevEvent = null, this.pointerIsDown = !1, this.pointerWasMoved = !1, this._interacting = !1, this._ending = !1, this._stopped = !0, this._proxy = void 0, this.simulation = null, this.doMove = yt(function(u) {
          this.move(u);
        }, "The interaction.doMove() method has been renamed to interaction.move()"), this.coords = { start: { page: { x: 0, y: 0 }, client: { x: 0, y: 0 }, timeStamp: 0 }, prev: { page: { x: 0, y: 0 }, client: { x: 0, y: 0 }, timeStamp: 0 }, cur: { page: { x: 0, y: 0 }, client: { x: 0, y: 0 }, timeStamp: 0 }, delta: { page: { x: 0, y: 0 }, client: { x: 0, y: 0 }, timeStamp: 0 }, velocity: { page: { x: 0, y: 0 }, client: { x: 0, y: 0 }, timeStamp: 0 } }, this._id = ki++, this._scopeFire = i, this.pointerType = r;
        var o = this;
        this._proxy = {};
        var a = function(u) {
          Object.defineProperty(n._proxy, u, { get: function() {
            return o[u];
          } });
        };
        for (var s in Ei)
          a(s);
        var l = function(u) {
          Object.defineProperty(n._proxy, u, { value: function() {
            return o[u].apply(o, arguments);
          } });
        };
        for (var h in Hn)
          l(h);
        this._scopeFire("interactions:new", { interaction: this });
      }
      return g(t, [{ key: "pointerMoveTolerance", get: function() {
        return 1;
      } }, { key: "pointerDown", value: function(e, n, r) {
        var i = this.updatePointer(e, n, r, !0), o = this.pointers[i];
        this._scopeFire("interactions:down", { pointer: e, event: n, eventTarget: r, pointerIndex: i, pointerInfo: o, type: "down", interaction: this });
      } }, { key: "start", value: function(e, n, r) {
        return !(this.interacting() || !this.pointerIsDown || this.pointers.length < (e.name === "gesture" ? 2 : 1) || !n.options[e.name].enabled) && (Zt(this.prepared, e), this.interactable = n, this.element = r, this.rect = n.getRect(r), this.edges = this.prepared.edges ? P({}, this.prepared.edges) : { left: !0, right: !0, top: !0, bottom: !0 }, this._stopped = !1, this._interacting = this._doPhase({ interaction: this, event: this.downEvent, phase: "start" }) && !this._stopped, this._interacting);
      } }, { key: "pointerMove", value: function(e, n, r) {
        this.simulation || this.modification && this.modification.endResult || this.updatePointer(e, n, r, !1);
        var i, o, a = this.coords.cur.page.x === this.coords.prev.page.x && this.coords.cur.page.y === this.coords.prev.page.y && this.coords.cur.client.x === this.coords.prev.client.x && this.coords.cur.client.y === this.coords.prev.client.y;
        this.pointerIsDown && !this.pointerWasMoved && (i = this.coords.cur.client.x - this.coords.start.client.x, o = this.coords.cur.client.y - this.coords.start.client.y, this.pointerWasMoved = Ge(i, o) > this.pointerMoveTolerance);
        var s, l, h, u = this.getPointerIndex(e), m = { pointer: e, pointerIndex: u, pointerInfo: this.pointers[u], event: n, type: "move", eventTarget: r, dx: i, dy: o, duplicate: a, interaction: this };
        a || (s = this.coords.velocity, l = this.coords.delta, h = Math.max(l.timeStamp / 1e3, 1e-3), s.page.x = l.page.x / h, s.page.y = l.page.y / h, s.client.x = l.client.x / h, s.client.y = l.client.y / h, s.timeStamp = h), this._scopeFire("interactions:move", m), a || this.simulation || (this.interacting() && (m.type = null, this.move(m)), this.pointerWasMoved && ke(this.coords.prev, this.coords.cur));
      } }, { key: "move", value: function(e) {
        e && e.event || S(this.coords.delta), (e = P({ pointer: this._latestPointer.pointer, event: this._latestPointer.event, eventTarget: this._latestPointer.eventTarget, interaction: this }, e || {})).phase = "move", this._doPhase(e);
      } }, { key: "pointerUp", value: function(e, n, r, i) {
        var o = this.getPointerIndex(e);
        o === -1 && (o = this.updatePointer(e, n, r, !1));
        var a = /cancel$/i.test(n.type) ? "cancel" : "up";
        this._scopeFire("interactions:".concat(a), { pointer: e, pointerIndex: o, pointerInfo: this.pointers[o], event: n, eventTarget: r, type: a, curEventTarget: i, interaction: this }), this.simulation || this.end(n), this.removePointer(e, n);
      } }, { key: "documentBlur", value: function(e) {
        this.end(e), this._scopeFire("interactions:blur", { event: e, type: "blur", interaction: this });
      } }, { key: "end", value: function(e) {
        var n;
        this._ending = !0, e = e || this._latestPointer.event, this.interacting() && (n = this._doPhase({ event: e, interaction: this, phase: "end" })), this._ending = !1, n === !0 && this.stop();
      } }, { key: "currentAction", value: function() {
        return this._interacting ? this.prepared.name : null;
      } }, { key: "interacting", value: function() {
        return this._interacting;
      } }, { key: "stop", value: function() {
        this._scopeFire("interactions:stop", { interaction: this }), this.interactable = this.element = null, this._interacting = !1, this._stopped = !0, this.prepared.name = this.prevEvent = null;
      } }, { key: "getPointerIndex", value: function(e) {
        var n = Q(e);
        return this.pointerType === "mouse" || this.pointerType === "pen" ? this.pointers.length - 1 : vt(this.pointers, function(r) {
          return r.id === n;
        });
      } }, { key: "getPointerInfo", value: function(e) {
        return this.pointers[this.getPointerIndex(e)];
      } }, { key: "updatePointer", value: function(e, n, r, i) {
        var o, a, s, l = Q(e), h = this.getPointerIndex(e), u = this.pointers[h];
        return i = i !== !1 && (i || /(down|start)$/i.test(n.type)), u ? u.pointer = e : (u = new xi(l, e, n, null, null), h = this.pointers.length, this.pointers.push(u)), H(this.coords.cur, this.pointers.map(function(m) {
          return m.pointer;
        }), this._now()), o = this.coords.delta, a = this.coords.prev, s = this.coords.cur, o.page.x = s.page.x - a.page.x, o.page.y = s.page.y - a.page.y, o.client.x = s.client.x - a.client.x, o.client.y = s.client.y - a.client.y, o.timeStamp = s.timeStamp - a.timeStamp, i && (this.pointerIsDown = !0, u.downTime = this.coords.cur.timeStamp, u.downTarget = r, it(this.downPointer, e), this.interacting() || (ke(this.coords.start, this.coords.cur), ke(this.coords.prev, this.coords.cur), this.downEvent = n, this.pointerWasMoved = !1)), this._updateLatestPointer(e, n, r), this._scopeFire("interactions:update-pointer", { pointer: e, event: n, eventTarget: r, down: i, pointerInfo: u, pointerIndex: h, interaction: this }), h;
      } }, { key: "removePointer", value: function(e, n) {
        var r = this.getPointerIndex(e);
        if (r !== -1) {
          var i = this.pointers[r];
          this._scopeFire("interactions:remove-pointer", { pointer: e, event: n, eventTarget: null, pointerIndex: r, pointerInfo: i, interaction: this }), this.pointers.splice(r, 1), this.pointerIsDown = !1;
        }
      } }, { key: "_updateLatestPointer", value: function(e, n, r) {
        this._latestPointer.pointer = e, this._latestPointer.event = n, this._latestPointer.eventTarget = r;
      } }, { key: "destroy", value: function() {
        this._latestPointer.pointer = null, this._latestPointer.event = null, this._latestPointer.eventTarget = null;
      } }, { key: "_createPreparedEvent", value: function(e, n, r, i) {
        return new on(this, e, this.prepared.name, n, this.element, r, i);
      } }, { key: "_fireEvent", value: function(e) {
        var n;
        (n = this.interactable) == null || n.fire(e), (!this.prevEvent || e.timeStamp >= this.prevEvent.timeStamp) && (this.prevEvent = e);
      } }, { key: "_doPhase", value: function(e) {
        var n = e.event, r = e.phase, i = e.preEnd, o = e.type, a = this.rect;
        if (a && r === "move" && (rt(this.edges, a, this.coords.delta[this.interactable.options.deltaSource]), a.width = a.right - a.left, a.height = a.bottom - a.top), this._scopeFire("interactions:before-action-".concat(r), e) === !1)
          return !1;
        var s = e.iEvent = this._createPreparedEvent(n, r, i, o);
        return this._scopeFire("interactions:action-".concat(r), e), r === "start" && (this.prevEvent = s), this._fireEvent(s), this._scopeFire("interactions:after-action-".concat(r), e), !0;
      } }, { key: "_now", value: function() {
        return Date.now();
      } }]), t;
    }();
    function Nn(t) {
      qn(t.interaction);
    }
    function qn(t) {
      if (!function(n) {
        return !(!n.offset.pending.x && !n.offset.pending.y);
      }(t))
        return !1;
      var e = t.offset.pending;
      return an(t.coords.cur, e), an(t.coords.delta, e), rt(t.edges, t.rect, e), e.x = 0, e.y = 0, !0;
    }
    function Ti(t) {
      var e = t.x, n = t.y;
      this.offset.pending.x += e, this.offset.pending.y += n, this.offset.total.x += e, this.offset.total.y += n;
    }
    function an(t, e) {
      var n = t.page, r = t.client, i = e.x, o = e.y;
      n.x += i, n.y += o, r.x += i, r.y += o;
    }
    Hn.offsetBy = "";
    var zi = { id: "offset", before: ["modifiers", "pointer-events", "actions", "inertia"], install: function(t) {
      t.Interaction.prototype.offsetBy = Ti;
    }, listeners: { "interactions:new": function(t) {
      t.interaction.offset = { total: { x: 0, y: 0 }, pending: { x: 0, y: 0 } };
    }, "interactions:update-pointer": function(t) {
      return function(e) {
        e.pointerIsDown && (an(e.coords.cur, e.offset.total), e.offset.pending.x = 0, e.offset.pending.y = 0);
      }(t.interaction);
    }, "interactions:before-action-start": Nn, "interactions:before-action-move": Nn, "interactions:before-action-end": function(t) {
      var e = t.interaction;
      if (qn(e))
        return e.move({ offset: !0 }), e.end(), !1;
    }, "interactions:stop": function(t) {
      var e = t.interaction;
      e.offset.total.x = 0, e.offset.total.y = 0, e.offset.pending.x = 0, e.offset.pending.y = 0;
    } } }, Xn = zi, _i = function() {
      function t(e) {
        w(this, t), this.active = !1, this.isModified = !1, this.smoothEnd = !1, this.allowResume = !1, this.modification = void 0, this.modifierCount = 0, this.modifierArg = void 0, this.startCoords = void 0, this.t0 = 0, this.v0 = 0, this.te = 0, this.targetOffset = void 0, this.modifiedOffset = void 0, this.currentOffset = void 0, this.lambda_v0 = 0, this.one_ve_v0 = 0, this.timeout = void 0, this.interaction = void 0, this.interaction = e;
      }
      return g(t, [{ key: "start", value: function(e) {
        var n = this.interaction, r = At(n);
        if (!r || !r.enabled)
          return !1;
        var i = n.coords.velocity.client, o = Ge(i.x, i.y), a = this.modification || (this.modification = new rn(n));
        if (a.copyFrom(n.modification), this.t0 = n._now(), this.allowResume = r.allowResume, this.v0 = o, this.currentOffset = { x: 0, y: 0 }, this.startCoords = n.coords.cur.page, this.modifierArg = a.fillArg({ pageCoords: this.startCoords, preEnd: !0, phase: "inertiastart" }), this.t0 - n.coords.cur.timeStamp < 50 && o > r.minSpeed && o > r.endSpeed)
          this.startInertia();
        else {
          if (a.result = a.setAll(this.modifierArg), !a.result.changed)
            return !1;
          this.startSmoothEnd();
        }
        return n.modification.result.rect = null, n.offsetBy(this.targetOffset), n._doPhase({ interaction: n, event: e, phase: "inertiastart" }), n.offsetBy({ x: -this.targetOffset.x, y: -this.targetOffset.y }), n.modification.result.rect = null, this.active = !0, n.simulation = this, !0;
      } }, { key: "startInertia", value: function() {
        var e = this, n = this.interaction.coords.velocity.client, r = At(this.interaction), i = r.resistance, o = -Math.log(r.endSpeed / this.v0) / i;
        this.targetOffset = { x: (n.x - o) / i, y: (n.y - o) / i }, this.te = o, this.lambda_v0 = i / this.v0, this.one_ve_v0 = 1 - r.endSpeed / this.v0;
        var a = this.modification, s = this.modifierArg;
        s.pageCoords = { x: this.startCoords.x + this.targetOffset.x, y: this.startCoords.y + this.targetOffset.y }, a.result = a.setAll(s), a.result.changed && (this.isModified = !0, this.modifiedOffset = { x: this.targetOffset.x + a.result.delta.x, y: this.targetOffset.y + a.result.delta.y }), this.onNextFrame(function() {
          return e.inertiaTick();
        });
      } }, { key: "startSmoothEnd", value: function() {
        var e = this;
        this.smoothEnd = !0, this.isModified = !0, this.targetOffset = { x: this.modification.result.delta.x, y: this.modification.result.delta.y }, this.onNextFrame(function() {
          return e.smoothEndTick();
        });
      } }, { key: "onNextFrame", value: function(e) {
        var n = this;
        this.timeout = Le.request(function() {
          n.active && e();
        });
      } }, { key: "inertiaTick", value: function() {
        var e, n, r, i, o, a, s, l = this, h = this.interaction, u = At(h).resistance, m = (h._now() - this.t0) / 1e3;
        if (m < this.te) {
          var y, E = 1 - (Math.exp(-u * m) - this.lambda_v0) / this.one_ve_v0;
          this.isModified ? (e = 0, n = 0, r = this.targetOffset.x, i = this.targetOffset.y, o = this.modifiedOffset.x, a = this.modifiedOffset.y, y = { x: Yn(s = E, e, r, o), y: Yn(s, n, i, a) }) : y = { x: this.targetOffset.x * E, y: this.targetOffset.y * E };
          var x = { x: y.x - this.currentOffset.x, y: y.y - this.currentOffset.y };
          this.currentOffset.x += x.x, this.currentOffset.y += x.y, h.offsetBy(x), h.move(), this.onNextFrame(function() {
            return l.inertiaTick();
          });
        } else
          h.offsetBy({ x: this.modifiedOffset.x - this.currentOffset.x, y: this.modifiedOffset.y - this.currentOffset.y }), this.end();
      } }, { key: "smoothEndTick", value: function() {
        var e = this, n = this.interaction, r = n._now() - this.t0, i = At(n).smoothEndDuration;
        if (r < i) {
          var o = { x: $n(r, 0, this.targetOffset.x, i), y: $n(r, 0, this.targetOffset.y, i) }, a = { x: o.x - this.currentOffset.x, y: o.y - this.currentOffset.y };
          this.currentOffset.x += a.x, this.currentOffset.y += a.y, n.offsetBy(a), n.move({ skipModifiers: this.modifierCount }), this.onNextFrame(function() {
            return e.smoothEndTick();
          });
        } else
          n.offsetBy({ x: this.targetOffset.x - this.currentOffset.x, y: this.targetOffset.y - this.currentOffset.y }), this.end();
      } }, { key: "resume", value: function(e) {
        var n = e.pointer, r = e.event, i = e.eventTarget, o = this.interaction;
        o.offsetBy({ x: -this.currentOffset.x, y: -this.currentOffset.y }), o.updatePointer(n, r, i, !0), o._doPhase({ interaction: o, event: r, phase: "resume" }), ke(o.coords.prev, o.coords.cur), this.stop();
      } }, { key: "end", value: function() {
        this.interaction.move(), this.interaction.end(), this.stop();
      } }, { key: "stop", value: function() {
        this.active = this.smoothEnd = !1, this.interaction.simulation = null, Le.cancel(this.timeout);
      } }]), t;
    }();
    function At(t) {
      var e = t.interactable, n = t.prepared;
      return e && e.options && n.name && e.options[n.name].inertia;
    }
    var Oi = { id: "inertia", before: ["modifiers", "actions"], install: function(t) {
      var e = t.defaults;
      t.usePlugin(Xn), t.usePlugin(Ln), t.actions.phases.inertiastart = !0, t.actions.phases.resume = !0, e.perAction.inertia = { enabled: !1, resistance: 10, minSpeed: 100, endSpeed: 10, allowResume: !0, smoothEndDuration: 300 };
    }, listeners: { "interactions:new": function(t) {
      var e = t.interaction;
      e.inertia = new _i(e);
    }, "interactions:before-action-end": function(t) {
      var e = t.interaction, n = t.event;
      return (!e._interacting || e.simulation || !e.inertia.start(n)) && null;
    }, "interactions:down": function(t) {
      var e = t.interaction, n = t.eventTarget, r = e.inertia;
      if (r.active)
        for (var i = n; b.element(i); ) {
          if (i === e.element) {
            r.resume(t);
            break;
          }
          i = te(i);
        }
    }, "interactions:stop": function(t) {
      var e = t.interaction.inertia;
      e.active && e.stop();
    }, "interactions:before-action-resume": function(t) {
      var e = t.interaction.modification;
      e.stop(t), e.start(t, t.interaction.coords.cur.page), e.applyToInteraction(t);
    }, "interactions:before-action-inertiastart": function(t) {
      return t.interaction.modification.setAndApply(t);
    }, "interactions:action-resume": bt, "interactions:action-inertiastart": bt, "interactions:after-action-inertiastart": function(t) {
      return t.interaction.modification.restoreInteractionCoords(t);
    }, "interactions:after-action-resume": function(t) {
      return t.interaction.modification.restoreInteractionCoords(t);
    } } };
    function Yn(t, e, n, r) {
      var i = 1 - t;
      return i * i * e + 2 * i * t * n + t * t * r;
    }
    function $n(t, e, n, r) {
      return -n * (t /= r) * (t - 2) + e;
    }
    var Pi = Oi;
    function Gn(t, e) {
      for (var n = 0; n < e.length; n++) {
        var r = e[n];
        if (t.immediatePropagationStopped)
          break;
        r(t);
      }
    }
    var Vn = function() {
      function t(e) {
        w(this, t), this.options = void 0, this.types = {}, this.propagationStopped = !1, this.immediatePropagationStopped = !1, this.global = void 0, this.options = P({}, e || {});
      }
      return g(t, [{ key: "fire", value: function(e) {
        var n, r = this.global;
        (n = this.types[e.type]) && Gn(e, n), !e.propagationStopped && r && (n = r[e.type]) && Gn(e, n);
      } }, { key: "on", value: function(e, n) {
        var r = Pe(e, n);
        for (e in r)
          this.types[e] = at(this.types[e] || [], r[e]);
      } }, { key: "off", value: function(e, n) {
        var r = Pe(e, n);
        for (e in r) {
          var i = this.types[e];
          if (i && i.length)
            for (var o = 0, a = r[e]; o < a.length; o++) {
              var s = a[o], l = i.indexOf(s);
              l !== -1 && i.splice(l, 1);
            }
        }
      } }, { key: "getRect", value: function(e) {
        return null;
      } }]), t;
    }(), Mi = function() {
      function t(e) {
        w(this, t), this.currentTarget = void 0, this.originalEvent = void 0, this.type = void 0, this.originalEvent = e, it(this, e);
      }
      return g(t, [{ key: "preventOriginalDefault", value: function() {
        this.originalEvent.preventDefault();
      } }, { key: "stopPropagation", value: function() {
        this.originalEvent.stopPropagation();
      } }, { key: "stopImmediatePropagation", value: function() {
        this.originalEvent.stopImmediatePropagation();
      } }]), t;
    }();
    function wt(t) {
      return b.object(t) ? { capture: !!t.capture, passive: !!t.passive } : { capture: !!t, passive: !1 };
    }
    function Bt(t, e) {
      return t === e || (typeof t == "boolean" ? !!e.capture === t && !e.passive : !!t.capture == !!e.capture && !!t.passive == !!e.passive);
    }
    var Di = { id: "events", install: function(t) {
      var e, n = [], r = {}, i = [], o = { add: a, remove: s, addDelegate: function(u, m, y, E, x) {
        var T = wt(x);
        if (!r[y]) {
          r[y] = [];
          for (var O = 0; O < i.length; O++) {
            var z = i[O];
            a(z, y, l), a(z, y, h, !0);
          }
        }
        var M = r[y], B = mt(M, function(W) {
          return W.selector === u && W.context === m;
        });
        B || (B = { selector: u, context: m, listeners: [] }, M.push(B)), B.listeners.push({ func: E, options: T });
      }, removeDelegate: function(u, m, y, E, x) {
        var T, O = wt(x), z = r[y], M = !1;
        if (z)
          for (T = z.length - 1; T >= 0; T--) {
            var B = z[T];
            if (B.selector === u && B.context === m) {
              for (var W = B.listeners, R = W.length - 1; R >= 0; R--) {
                var j = W[R];
                if (j.func === E && Bt(j.options, O)) {
                  W.splice(R, 1), W.length || (z.splice(T, 1), s(m, y, l), s(m, y, h, !0)), M = !0;
                  break;
                }
              }
              if (M)
                break;
            }
          }
      }, delegateListener: l, delegateUseCapture: h, delegatedEvents: r, documents: i, targets: n, supportsOptions: !1, supportsPassive: !1 };
      function a(u, m, y, E) {
        if (u.addEventListener) {
          var x = wt(E), T = mt(n, function(O) {
            return O.eventTarget === u;
          });
          T || (T = { eventTarget: u, events: {} }, n.push(T)), T.events[m] || (T.events[m] = []), mt(T.events[m], function(O) {
            return O.func === y && Bt(O.options, x);
          }) || (u.addEventListener(m, y, o.supportsOptions ? x : x.capture), T.events[m].push({ func: y, options: x }));
        }
      }
      function s(u, m, y, E) {
        if (u.addEventListener && u.removeEventListener) {
          var x = vt(n, function(pe) {
            return pe.eventTarget === u;
          }), T = n[x];
          if (T && T.events)
            if (m !== "all") {
              var O = !1, z = T.events[m];
              if (z) {
                if (y === "all") {
                  for (var M = z.length - 1; M >= 0; M--) {
                    var B = z[M];
                    s(u, m, B.func, B.options);
                  }
                  return;
                }
                for (var W = wt(E), R = 0; R < z.length; R++) {
                  var j = z[R];
                  if (j.func === y && Bt(j.options, W)) {
                    u.removeEventListener(m, y, o.supportsOptions ? W : W.capture), z.splice(R, 1), z.length === 0 && (delete T.events[m], O = !0);
                    break;
                  }
                }
              }
              O && !Object.keys(T.events).length && n.splice(x, 1);
            } else
              for (m in T.events)
                T.events.hasOwnProperty(m) && s(u, m, "all");
        }
      }
      function l(u, m) {
        for (var y = wt(m), E = new Mi(u), x = r[u.type], T = ot(u)[0], O = T; b.element(O); ) {
          for (var z = 0; z < x.length; z++) {
            var M = x[z], B = M.selector, W = M.context;
            if (G(O, B) && le(W, T) && le(W, O)) {
              var R = M.listeners;
              E.currentTarget = O;
              for (var j = 0; j < R.length; j++) {
                var pe = R[j];
                Bt(pe.options, y) && pe.func(E);
              }
            }
          }
          O = te(O);
        }
      }
      function h(u) {
        return l(u, !0);
      }
      return (e = t.document) == null || e.createElement("div").addEventListener("test", null, { get capture() {
        return o.supportsOptions = !0;
      }, get passive() {
        return o.supportsPassive = !0;
      } }), t.events = o, o;
    } }, sn = { methodOrder: ["simulationResume", "mouseOrPen", "hasPointer", "idle"], search: function(t) {
      for (var e = 0, n = sn.methodOrder; e < n.length; e++) {
        var r = n[e], i = sn[r](t);
        if (i)
          return i;
      }
      return null;
    }, simulationResume: function(t) {
      var e = t.pointerType, n = t.eventType, r = t.eventTarget, i = t.scope;
      if (!/down|start/i.test(n))
        return null;
      for (var o = 0, a = i.interactions.list; o < a.length; o++) {
        var s = a[o], l = r;
        if (s.simulation && s.simulation.allowResume && s.pointerType === e)
          for (; l; ) {
            if (l === s.element)
              return s;
            l = te(l);
          }
      }
      return null;
    }, mouseOrPen: function(t) {
      var e, n = t.pointerId, r = t.pointerType, i = t.eventType, o = t.scope;
      if (r !== "mouse" && r !== "pen")
        return null;
      for (var a = 0, s = o.interactions.list; a < s.length; a++) {
        var l = s[a];
        if (l.pointerType === r) {
          if (l.simulation && !Kn(l, n))
            continue;
          if (l.interacting())
            return l;
          e || (e = l);
        }
      }
      if (e)
        return e;
      for (var h = 0, u = o.interactions.list; h < u.length; h++) {
        var m = u[h];
        if (!(m.pointerType !== r || /down/i.test(i) && m.simulation))
          return m;
      }
      return null;
    }, hasPointer: function(t) {
      for (var e = t.pointerId, n = 0, r = t.scope.interactions.list; n < r.length; n++) {
        var i = r[n];
        if (Kn(i, e))
          return i;
      }
      return null;
    }, idle: function(t) {
      for (var e = t.pointerType, n = 0, r = t.scope.interactions.list; n < r.length; n++) {
        var i = r[n];
        if (i.pointers.length === 1) {
          var o = i.interactable;
          if (o && (!o.options.gesture || !o.options.gesture.enabled))
            continue;
        } else if (i.pointers.length >= 2)
          continue;
        if (!i.interacting() && e === i.pointerType)
          return i;
      }
      return null;
    } };
    function Kn(t, e) {
      return t.pointers.some(function(n) {
        return n.id === e;
      });
    }
    var Ri = sn, cn = ["pointerDown", "pointerMove", "pointerUp", "updatePointer", "removePointer", "windowBlur"];
    function Un(t, e) {
      return function(n) {
        var r = e.interactions.list, i = Fe(n), o = ot(n), a = o[0], s = o[1], l = [];
        if (/^touch/.test(n.type)) {
          e.prevTouchTime = e.now();
          for (var h = 0, u = n.changedTouches; h < u.length; h++) {
            var m = u[h], y = { pointer: m, pointerId: Q(m), pointerType: i, eventType: n.type, eventTarget: a, curEventTarget: s, scope: e }, E = Jn(y);
            l.push([y.pointer, y.eventTarget, y.curEventTarget, E]);
          }
        } else {
          var x = !1;
          if (!C.supportsPointerEvent && /mouse/.test(n.type)) {
            for (var T = 0; T < r.length && !x; T++)
              x = r[T].pointerType !== "mouse" && r[T].pointerIsDown;
            x = x || e.now() - e.prevTouchTime < 500 || n.timeStamp === 0;
          }
          if (!x) {
            var O = { pointer: n, pointerId: Q(n), pointerType: i, eventType: n.type, curEventTarget: s, eventTarget: a, scope: e }, z = Jn(O);
            l.push([O.pointer, O.eventTarget, O.curEventTarget, z]);
          }
        }
        for (var M = 0; M < l.length; M++) {
          var B = l[M], W = B[0], R = B[1], j = B[2];
          B[3][t](W, n, R, j);
        }
      };
    }
    function Jn(t) {
      var e = t.pointerType, n = t.scope, r = { interaction: Ri.search(t), searchDetails: t };
      return n.fire("interactions:find", r), r.interaction || n.interactions.new({ pointerType: e });
    }
    function ln(t, e) {
      var n = t.doc, r = t.scope, i = t.options, o = r.interactions.docEvents, a = r.events, s = a[e];
      for (var l in r.browser.isIOS && !i.events && (i.events = { passive: !1 }), a.delegatedEvents)
        s(n, l, a.delegateListener), s(n, l, a.delegateUseCapture, !0);
      for (var h = i && i.events, u = 0; u < o.length; u++) {
        var m = o[u];
        s(n, m.type, m.listener, h);
      }
    }
    var Ii = { id: "core/interactions", install: function(t) {
      for (var e = {}, n = 0; n < cn.length; n++) {
        var r = cn[n];
        e[r] = Un(r, t);
      }
      var i, o = C.pEventTypes;
      function a() {
        for (var s = 0, l = t.interactions.list; s < l.length; s++) {
          var h = l[s];
          if (h.pointerIsDown && h.pointerType === "touch" && !h._interacting)
            for (var u = function() {
              var E = y[m];
              t.documents.some(function(x) {
                return le(x.doc, E.downTarget);
              }) || h.removePointer(E.pointer, E.event);
            }, m = 0, y = h.pointers; m < y.length; m++)
              u();
        }
      }
      (i = q.PointerEvent ? [{ type: o.down, listener: a }, { type: o.down, listener: e.pointerDown }, { type: o.move, listener: e.pointerMove }, { type: o.up, listener: e.pointerUp }, { type: o.cancel, listener: e.pointerUp }] : [{ type: "mousedown", listener: e.pointerDown }, { type: "mousemove", listener: e.pointerMove }, { type: "mouseup", listener: e.pointerUp }, { type: "touchstart", listener: a }, { type: "touchstart", listener: e.pointerDown }, { type: "touchmove", listener: e.pointerMove }, { type: "touchend", listener: e.pointerUp }, { type: "touchcancel", listener: e.pointerUp }]).push({ type: "blur", listener: function(s) {
        for (var l = 0, h = t.interactions.list; l < h.length; l++)
          h[l].documentBlur(s);
      } }), t.prevTouchTime = 0, t.Interaction = function(s) {
        J(h, s);
        var l = ne(h);
        function h() {
          return w(this, h), l.apply(this, arguments);
        }
        return g(h, [{ key: "pointerMoveTolerance", get: function() {
          return t.interactions.pointerMoveTolerance;
        }, set: function(u) {
          t.interactions.pointerMoveTolerance = u;
        } }, { key: "_now", value: function() {
          return t.now();
        } }]), h;
      }(Si), t.interactions = { list: [], new: function(s) {
        s.scopeFire = function(h, u) {
          return t.fire(h, u);
        };
        var l = new t.Interaction(s);
        return t.interactions.list.push(l), l;
      }, listeners: e, docEvents: i, pointerMoveTolerance: 1 }, t.usePlugin(jn);
    }, listeners: { "scope:add-document": function(t) {
      return ln(t, "add");
    }, "scope:remove-document": function(t) {
      return ln(t, "remove");
    }, "interactable:unset": function(t, e) {
      for (var n = t.interactable, r = e.interactions.list.length - 1; r >= 0; r--) {
        var i = e.interactions.list[r];
        i.interactable === n && (i.stop(), e.fire("interactions:destroy", { interaction: i }), i.destroy(), e.interactions.list.length > 2 && e.interactions.list.splice(r, 1));
      }
    } }, onDocSignal: ln, doOnInteractions: Un, methodNames: cn }, Ci = Ii, He = function(t) {
      return t[t.On = 0] = "On", t[t.Off = 1] = "Off", t;
    }(He || {}), Ai = function() {
      function t(e, n, r, i) {
        w(this, t), this.target = void 0, this.options = void 0, this._actions = void 0, this.events = new Vn(), this._context = void 0, this._win = void 0, this._doc = void 0, this._scopeEvents = void 0, this._actions = n.actions, this.target = e, this._context = n.context || r, this._win = Y(ht(e) ? this._context : e), this._doc = this._win.document, this._scopeEvents = i, this.set(n);
      }
      return g(t, [{ key: "_defaults", get: function() {
        return { base: {}, perAction: {}, actions: {} };
      } }, { key: "setOnEvents", value: function(e, n) {
        return b.func(n.onstart) && this.on("".concat(e, "start"), n.onstart), b.func(n.onmove) && this.on("".concat(e, "move"), n.onmove), b.func(n.onend) && this.on("".concat(e, "end"), n.onend), b.func(n.oninertiastart) && this.on("".concat(e, "inertiastart"), n.oninertiastart), this;
      } }, { key: "updatePerActionListeners", value: function(e, n, r) {
        var i, o = this, a = (i = this._actions.map[e]) == null ? void 0 : i.filterEventType, s = function(l) {
          return (a == null || a(l)) && It(l, o._actions);
        };
        (b.array(n) || b.object(n)) && this._onOff(He.Off, e, n, void 0, s), (b.array(r) || b.object(r)) && this._onOff(He.On, e, r, void 0, s);
      } }, { key: "setPerAction", value: function(e, n) {
        var r = this._defaults;
        for (var i in n) {
          var o = i, a = this.options[e], s = n[o];
          o === "listeners" && this.updatePerActionListeners(e, a.listeners, s), b.array(s) ? a[o] = st(s) : b.plainObject(s) ? (a[o] = P(a[o] || {}, lt(s)), b.object(r.perAction[o]) && "enabled" in r.perAction[o] && (a[o].enabled = s.enabled !== !1)) : b.bool(s) && b.object(r.perAction[o]) ? a[o].enabled = s : a[o] = s;
        }
      } }, { key: "getRect", value: function(e) {
        return e = e || (b.element(this.target) ? this.target : null), b.string(this.target) && (e = e || this._context.querySelector(this.target)), Xe(e);
      } }, { key: "rectChecker", value: function(e) {
        var n = this;
        return b.func(e) ? (this.getRect = function(r) {
          var i = P({}, e.apply(n, r));
          return "width" in i || (i.width = i.right - i.left, i.height = i.bottom - i.top), i;
        }, this) : e === null ? (delete this.getRect, this) : this.getRect;
      } }, { key: "_backCompatOption", value: function(e, n) {
        if (ht(n) || b.object(n)) {
          for (var r in this.options[e] = n, this._actions.map)
            this.options[r][e] = n;
          return this;
        }
        return this.options[e];
      } }, { key: "origin", value: function(e) {
        return this._backCompatOption("origin", e);
      } }, { key: "deltaSource", value: function(e) {
        return e === "page" || e === "client" ? (this.options.deltaSource = e, this) : this.options.deltaSource;
      } }, { key: "getAllElements", value: function() {
        var e = this.target;
        return b.string(e) ? Array.from(this._context.querySelectorAll(e)) : b.func(e) && e.getAllElements ? e.getAllElements() : b.element(e) ? [e] : [];
      } }, { key: "context", value: function() {
        return this._context;
      } }, { key: "inContext", value: function(e) {
        return this._context === e.ownerDocument || le(this._context, e);
      } }, { key: "testIgnoreAllow", value: function(e, n, r) {
        return !this.testIgnore(e.ignoreFrom, n, r) && this.testAllow(e.allowFrom, n, r);
      } }, { key: "testAllow", value: function(e, n, r) {
        return !e || !!b.element(r) && (b.string(e) ? dt(r, e, n) : !!b.element(e) && le(e, r));
      } }, { key: "testIgnore", value: function(e, n, r) {
        return !(!e || !b.element(r)) && (b.string(e) ? dt(r, e, n) : !!b.element(e) && le(e, r));
      } }, { key: "fire", value: function(e) {
        return this.events.fire(e), this;
      } }, { key: "_onOff", value: function(e, n, r, i, o) {
        b.object(n) && !b.array(n) && (i = r, r = null);
        var a = Pe(n, r, o);
        for (var s in a) {
          s === "wheel" && (s = C.wheelEvent);
          for (var l = 0, h = a[s]; l < h.length; l++) {
            var u = h[l];
            It(s, this._actions) ? this.events[e === He.On ? "on" : "off"](s, u) : b.string(this.target) ? this._scopeEvents[e === He.On ? "addDelegate" : "removeDelegate"](this.target, this._context, s, u, i) : this._scopeEvents[e === He.On ? "add" : "remove"](this.target, s, u, i);
          }
        }
        return this;
      } }, { key: "on", value: function(e, n, r) {
        return this._onOff(He.On, e, n, r);
      } }, { key: "off", value: function(e, n, r) {
        return this._onOff(He.Off, e, n, r);
      } }, { key: "set", value: function(e) {
        var n = this._defaults;
        for (var r in b.object(e) || (e = {}), this.options = lt(n.base), this._actions.methodDict) {
          var i = r, o = this._actions.methodDict[i];
          this.options[i] = {}, this.setPerAction(i, P(P({}, n.perAction), n.actions[i])), this[o](e[i]);
        }
        for (var a in e)
          a !== "getRect" ? b.func(this[a]) && this[a](e[a]) : this.rectChecker(e.getRect);
        return this;
      } }, { key: "unset", value: function() {
        if (b.string(this.target))
          for (var e in this._scopeEvents.delegatedEvents)
            for (var n = this._scopeEvents.delegatedEvents[e], r = n.length - 1; r >= 0; r--) {
              var i = n[r], o = i.selector, a = i.context, s = i.listeners;
              o === this.target && a === this._context && n.splice(r, 1);
              for (var l = s.length - 1; l >= 0; l--)
                this._scopeEvents.removeDelegate(this.target, this._context, e, s[l][0], s[l][1]);
            }
        else
          this._scopeEvents.remove(this.target, "all");
      } }]), t;
    }(), Bi = function() {
      function t(e) {
        var n = this;
        w(this, t), this.list = [], this.selectorMap = {}, this.scope = void 0, this.scope = e, e.addListeners({ "interactable:unset": function(r) {
          var i = r.interactable, o = i.target, a = b.string(o) ? n.selectorMap[o] : o[n.scope.id], s = vt(a, function(l) {
            return l === i;
          });
          a.splice(s, 1);
        } });
      }
      return g(t, [{ key: "new", value: function(e, n) {
        n = P(n || {}, { actions: this.scope.actions });
        var r = new this.scope.Interactable(e, n, this.scope.document, this.scope.events);
        return this.scope.addDocument(r._doc), this.list.push(r), b.string(e) ? (this.selectorMap[e] || (this.selectorMap[e] = []), this.selectorMap[e].push(r)) : (r.target[this.scope.id] || Object.defineProperty(e, this.scope.id, { value: [], configurable: !0 }), e[this.scope.id].push(r)), this.scope.fire("interactable:new", { target: e, options: n, interactable: r, win: this.scope._win }), r;
      } }, { key: "getExisting", value: function(e, n) {
        var r = n && n.context || this.scope.document, i = b.string(e), o = i ? this.selectorMap[e] : e[this.scope.id];
        if (o)
          return mt(o, function(a) {
            return a._context === r && (i || a.inContext(e));
          });
      } }, { key: "forEachMatch", value: function(e, n) {
        for (var r = 0, i = this.list; r < i.length; r++) {
          var o = i[r], a = void 0;
          if ((b.string(o.target) ? b.element(e) && G(e, o.target) : e === o.target) && o.inContext(e) && (a = n(o)), a !== void 0)
            return a;
        }
      } }]), t;
    }(), Fi = function() {
      function t() {
        var e = this;
        w(this, t), this.id = "__interact_scope_".concat(Math.floor(100 * Math.random())), this.isInitialized = !1, this.listenerMaps = [], this.browser = C, this.defaults = lt(Wn), this.Eventable = Vn, this.actions = { map: {}, phases: { start: !0, move: !0, end: !0 }, methodDict: {}, phaselessTypes: {} }, this.interactStatic = function(r) {
          var i = function o(a, s) {
            var l = r.interactables.getExisting(a, s);
            return l || ((l = r.interactables.new(a, s)).events.global = o.globalEvents), l;
          };
          return i.getPointerAverage = se, i.getTouchBBox = fe, i.getTouchDistance = ye, i.getTouchAngle = Ie, i.getElementRect = Xe, i.getElementClientRect = _e, i.matchesSelector = G, i.closest = Re, i.globalEvents = {}, i.version = "1.10.26", i.scope = r, i.use = function(o, a) {
            return this.scope.usePlugin(o, a), this;
          }, i.isSet = function(o, a) {
            return !!this.scope.interactables.get(o, a && a.context);
          }, i.on = yt(function(o, a, s) {
            if (b.string(o) && o.search(" ") !== -1 && (o = o.trim().split(/ +/)), b.array(o)) {
              for (var l = 0, h = o; l < h.length; l++) {
                var u = h[l];
                this.on(u, a, s);
              }
              return this;
            }
            if (b.object(o)) {
              for (var m in o)
                this.on(m, o[m], a);
              return this;
            }
            return It(o, this.scope.actions) ? this.globalEvents[o] ? this.globalEvents[o].push(a) : this.globalEvents[o] = [a] : this.scope.events.add(this.scope.document, o, a, { options: s }), this;
          }, "The interact.on() method is being deprecated"), i.off = yt(function(o, a, s) {
            if (b.string(o) && o.search(" ") !== -1 && (o = o.trim().split(/ +/)), b.array(o)) {
              for (var l = 0, h = o; l < h.length; l++) {
                var u = h[l];
                this.off(u, a, s);
              }
              return this;
            }
            if (b.object(o)) {
              for (var m in o)
                this.off(m, o[m], a);
              return this;
            }
            var y;
            return It(o, this.scope.actions) ? o in this.globalEvents && (y = this.globalEvents[o].indexOf(a)) !== -1 && this.globalEvents[o].splice(y, 1) : this.scope.events.remove(this.scope.document, o, a, s), this;
          }, "The interact.off() method is being deprecated"), i.debug = function() {
            return this.scope;
          }, i.supportsTouch = function() {
            return C.supportsTouch;
          }, i.supportsPointerEvent = function() {
            return C.supportsPointerEvent;
          }, i.stop = function() {
            for (var o = 0, a = this.scope.interactions.list; o < a.length; o++)
              a[o].stop();
            return this;
          }, i.pointerMoveTolerance = function(o) {
            return b.number(o) ? (this.scope.interactions.pointerMoveTolerance = o, this) : this.scope.interactions.pointerMoveTolerance;
          }, i.addDocument = function(o, a) {
            this.scope.addDocument(o, a);
          }, i.removeDocument = function(o) {
            this.scope.removeDocument(o);
          }, i;
        }(this), this.InteractEvent = on, this.Interactable = void 0, this.interactables = new Bi(this), this._win = void 0, this.document = void 0, this.window = void 0, this.documents = [], this._plugins = { list: [], map: {} }, this.onWindowUnload = function(r) {
          return e.removeDocument(r.target);
        };
        var n = this;
        this.Interactable = function(r) {
          J(o, r);
          var i = ne(o);
          function o() {
            return w(this, o), i.apply(this, arguments);
          }
          return g(o, [{ key: "_defaults", get: function() {
            return n.defaults;
          } }, { key: "set", value: function(a) {
            return Ee(F(o.prototype), "set", this).call(this, a), n.fire("interactable:set", { options: a, interactable: this }), this;
          } }, { key: "unset", value: function() {
            Ee(F(o.prototype), "unset", this).call(this);
            var a = n.interactables.list.indexOf(this);
            a < 0 || (n.interactables.list.splice(a, 1), n.fire("interactable:unset", { interactable: this }));
          } }]), o;
        }(Ai);
      }
      return g(t, [{ key: "addListeners", value: function(e, n) {
        this.listenerMaps.push({ id: n, map: e });
      } }, { key: "fire", value: function(e, n) {
        for (var r = 0, i = this.listenerMaps; r < i.length; r++) {
          var o = i[r].map[e];
          if (o && o(n, this, e) === !1)
            return !1;
        }
      } }, { key: "init", value: function(e) {
        return this.isInitialized ? this : function(n, r) {
          return n.isInitialized = !0, b.window(r) && oe(r), q.init(r), C.init(r), Le.init(r), n.window = r, n.document = r.document, n.usePlugin(Ci), n.usePlugin(Di), n;
        }(this, e);
      } }, { key: "pluginIsInstalled", value: function(e) {
        var n = e.id;
        return n ? !!this._plugins.map[n] : this._plugins.list.indexOf(e) !== -1;
      } }, { key: "usePlugin", value: function(e, n) {
        if (!this.isInitialized)
          return this;
        if (this.pluginIsInstalled(e))
          return this;
        if (e.id && (this._plugins.map[e.id] = e), this._plugins.list.push(e), e.install && e.install(this, n), e.listeners && e.before) {
          for (var r = 0, i = this.listenerMaps.length, o = e.before.reduce(function(s, l) {
            return s[l] = !0, s[Qn(l)] = !0, s;
          }, {}); r < i; r++) {
            var a = this.listenerMaps[r].id;
            if (a && (o[a] || o[Qn(a)]))
              break;
          }
          this.listenerMaps.splice(r, 0, { id: e.id, map: e.listeners });
        } else
          e.listeners && this.listenerMaps.push({ id: e.id, map: e.listeners });
        return this;
      } }, { key: "addDocument", value: function(e, n) {
        if (this.getDocIndex(e) !== -1)
          return !1;
        var r = Y(e);
        n = n ? P({}, n) : {}, this.documents.push({ doc: e, options: n }), this.events.documents.push(e), e !== this.document && this.events.add(r, "unload", this.onWindowUnload), this.fire("scope:add-document", { doc: e, window: r, scope: this, options: n });
      } }, { key: "removeDocument", value: function(e) {
        var n = this.getDocIndex(e), r = Y(e), i = this.documents[n].options;
        this.events.remove(r, "unload", this.onWindowUnload), this.documents.splice(n, 1), this.events.documents.splice(n, 1), this.fire("scope:remove-document", { doc: e, window: r, scope: this, options: i });
      } }, { key: "getDocIndex", value: function(e) {
        for (var n = 0; n < this.documents.length; n++)
          if (this.documents[n].doc === e)
            return n;
        return -1;
      } }, { key: "getDocOptions", value: function(e) {
        var n = this.getDocIndex(e);
        return n === -1 ? null : this.documents[n].options;
      } }, { key: "now", value: function() {
        return (this.window.Date || Date).now();
      } }]), t;
    }();
    function Qn(t) {
      return t && t.replace(/\/.*$/, "");
    }
    var Zn = new Fi(), de = Zn.interactStatic, ji = typeof globalThis < "u" ? globalThis : window;
    Zn.init(ji);
    var Li = Object.freeze({ __proto__: null, edgeTarget: function() {
    }, elements: function() {
    }, grid: function(t) {
      var e = [["x", "y"], ["left", "top"], ["right", "bottom"], ["width", "height"]].filter(function(r) {
        var i = r[0], o = r[1];
        return i in t || o in t;
      }), n = function(r, i) {
        for (var o = t.range, a = t.limits, s = a === void 0 ? { left: -1 / 0, right: 1 / 0, top: -1 / 0, bottom: 1 / 0 } : a, l = t.offset, h = l === void 0 ? { x: 0, y: 0 } : l, u = { range: o, grid: t, x: null, y: null }, m = 0; m < e.length; m++) {
          var y = e[m], E = y[0], x = y[1], T = Math.round((r - h.x) / t[E]), O = Math.round((i - h.y) / t[x]);
          u[E] = Math.max(s.left, Math.min(s.right, T * t[E] + h.x)), u[x] = Math.max(s.top, Math.min(s.bottom, O * t[x] + h.y));
        }
        return u;
      };
      return n.grid = t, n.coordFields = e, n;
    } }), Wi = { id: "snappers", install: function(t) {
      var e = t.interactStatic;
      e.snappers = P(e.snappers || {}, Li), e.createSnapGrid = e.snappers.grid;
    } }, Hi = Wi, Ni = { start: function(t) {
      var e = t.state, n = t.rect, r = t.edges, i = t.pageCoords, o = e.options, a = o.ratio, s = o.enabled, l = e.options, h = l.equalDelta, u = l.modifiers;
      a === "preserve" && (a = n.width / n.height), e.startCoords = P({}, i), e.startRect = P({}, n), e.ratio = a, e.equalDelta = h;
      var m = e.linkedEdges = { top: r.top || r.left && !r.bottom, left: r.left || r.top && !r.right, bottom: r.bottom || r.right && !r.top, right: r.right || r.bottom && !r.left };
      if (e.xIsPrimaryAxis = !(!r.left && !r.right), e.equalDelta) {
        var y = (m.left ? 1 : -1) * (m.top ? 1 : -1);
        e.edgeSign = { x: y, y };
      } else
        e.edgeSign = { x: m.left ? -1 : 1, y: m.top ? -1 : 1 };
      if (s !== !1 && P(r, m), u != null && u.length) {
        var E = new rn(t.interaction);
        E.copyFrom(t.interaction.modification), E.prepareStates(u), e.subModification = E, E.startAll(f({}, t));
      }
    }, set: function(t) {
      var e = t.state, n = t.rect, r = t.coords, i = e.linkedEdges, o = P({}, r), a = e.equalDelta ? qi : Xi;
      if (P(t.edges, i), a(e, e.xIsPrimaryAxis, r, n), !e.subModification)
        return null;
      var s = P({}, n);
      rt(i, s, { x: r.x - o.x, y: r.y - o.y });
      var l = e.subModification.setAll(f(f({}, t), {}, { rect: s, edges: i, pageCoords: r, prevCoords: r, prevRect: s })), h = l.delta;
      return l.changed && (a(e, Math.abs(h.x) > Math.abs(h.y), l.coords, l.rect), P(r, l.coords)), l.eventProps;
    }, defaults: { ratio: "preserve", equalDelta: !1, modifiers: [], enabled: !1 } };
    function qi(t, e, n) {
      var r = t.startCoords, i = t.edgeSign;
      e ? n.y = r.y + (n.x - r.x) * i.y : n.x = r.x + (n.y - r.y) * i.x;
    }
    function Xi(t, e, n, r) {
      var i = t.startRect, o = t.startCoords, a = t.ratio, s = t.edgeSign;
      if (e) {
        var l = r.width / a;
        n.y = o.y + (l - i.height) * s.y;
      } else {
        var h = r.height * a;
        n.x = o.x + (h - i.width) * s.x;
      }
    }
    var Yi = We(Ni, "aspectRatio"), er = function() {
    };
    er._defaults = {};
    var Ft = er;
    function Ke(t, e, n) {
      return b.func(t) ? Oe(t, e.interactable, e.element, [n.x, n.y, e]) : Oe(t, e.interactable, e.element);
    }
    var jt = { start: function(t) {
      var e = t.rect, n = t.startOffset, r = t.state, i = t.interaction, o = t.pageCoords, a = r.options, s = a.elementRect, l = P({ left: 0, top: 0, right: 0, bottom: 0 }, a.offset || {});
      if (e && s) {
        var h = Ke(a.restriction, i, o);
        if (h) {
          var u = h.right - h.left - e.width, m = h.bottom - h.top - e.height;
          u < 0 && (l.left += u, l.right += u), m < 0 && (l.top += m, l.bottom += m);
        }
        l.left += n.left - e.width * s.left, l.top += n.top - e.height * s.top, l.right += n.right - e.width * (1 - s.right), l.bottom += n.bottom - e.height * (1 - s.bottom);
      }
      r.offset = l;
    }, set: function(t) {
      var e = t.coords, n = t.interaction, r = t.state, i = r.options, o = r.offset, a = Ke(i.restriction, n, e);
      if (a) {
        var s = function(l) {
          return !l || "left" in l && "top" in l || ((l = P({}, l)).left = l.x || 0, l.top = l.y || 0, l.right = l.right || l.left + l.width, l.bottom = l.bottom || l.top + l.height), l;
        }(a);
        e.x = Math.max(Math.min(s.right - o.right, e.x), s.left + o.left), e.y = Math.max(Math.min(s.bottom - o.bottom, e.y), s.top + o.top);
      }
    }, defaults: { restriction: null, elementRect: null, offset: null, endOnly: !1, enabled: !1 } }, $i = We(jt, "restrict"), tr = { top: 1 / 0, left: 1 / 0, bottom: -1 / 0, right: -1 / 0 }, nr = { top: -1 / 0, left: -1 / 0, bottom: 1 / 0, right: 1 / 0 };
    function rr(t, e) {
      for (var n = 0, r = ["top", "left", "bottom", "right"]; n < r.length; n++) {
        var i = r[n];
        i in t || (t[i] = e[i]);
      }
      return t;
    }
    var xt = { noInner: tr, noOuter: nr, start: function(t) {
      var e, n = t.interaction, r = t.startOffset, i = t.state, o = i.options;
      o && (e = Ye(Ke(o.offset, n, n.coords.start.page))), e = e || { x: 0, y: 0 }, i.offset = { top: e.y + r.top, left: e.x + r.left, bottom: e.y - r.bottom, right: e.x - r.right };
    }, set: function(t) {
      var e = t.coords, n = t.edges, r = t.interaction, i = t.state, o = i.offset, a = i.options;
      if (n) {
        var s = P({}, e), l = Ke(a.inner, r, s) || {}, h = Ke(a.outer, r, s) || {};
        rr(l, tr), rr(h, nr), n.top ? e.y = Math.min(Math.max(h.top + o.top, s.y), l.top + o.top) : n.bottom && (e.y = Math.max(Math.min(h.bottom + o.bottom, s.y), l.bottom + o.bottom)), n.left ? e.x = Math.min(Math.max(h.left + o.left, s.x), l.left + o.left) : n.right && (e.x = Math.max(Math.min(h.right + o.right, s.x), l.right + o.right));
      }
    }, defaults: { inner: null, outer: null, offset: null, endOnly: !1, enabled: !1 } }, Gi = We(xt, "restrictEdges"), Vi = P({ get elementRect() {
      return { top: 0, left: 0, bottom: 1, right: 1 };
    }, set elementRect(t) {
    } }, jt.defaults), Ki = We({ start: jt.start, set: jt.set, defaults: Vi }, "restrictRect"), Ui = { width: -1 / 0, height: -1 / 0 }, Ji = { width: 1 / 0, height: 1 / 0 }, Qi = We({ start: function(t) {
      return xt.start(t);
    }, set: function(t) {
      var e = t.interaction, n = t.state, r = t.rect, i = t.edges, o = n.options;
      if (i) {
        var a = nt(Ke(o.min, e, t.coords)) || Ui, s = nt(Ke(o.max, e, t.coords)) || Ji;
        n.options = { endOnly: o.endOnly, inner: P({}, xt.noInner), outer: P({}, xt.noOuter) }, i.top ? (n.options.inner.top = r.bottom - a.height, n.options.outer.top = r.bottom - s.height) : i.bottom && (n.options.inner.bottom = r.top + a.height, n.options.outer.bottom = r.top + s.height), i.left ? (n.options.inner.left = r.right - a.width, n.options.outer.left = r.right - s.width) : i.right && (n.options.inner.right = r.left + a.width, n.options.outer.right = r.left + s.width), xt.set(t), n.options = o;
      }
    }, defaults: { min: null, max: null, endOnly: !1, enabled: !1 } }, "restrictSize"), un = { start: function(t) {
      var e, n = t.interaction, r = t.interactable, i = t.element, o = t.rect, a = t.state, s = t.startOffset, l = a.options, h = l.offsetWithOrigin ? function(y) {
        var E = y.interaction.element, x = Ye(Oe(y.state.options.origin, null, null, [E])), T = x || $e(y.interactable, E, y.interaction.prepared.name);
        return T;
      }(t) : { x: 0, y: 0 };
      if (l.offset === "startCoords")
        e = { x: n.coords.start.page.x, y: n.coords.start.page.y };
      else {
        var u = Oe(l.offset, r, i, [n]);
        (e = Ye(u) || { x: 0, y: 0 }).x += h.x, e.y += h.y;
      }
      var m = l.relativePoints;
      a.offsets = o && m && m.length ? m.map(function(y, E) {
        return { index: E, relativePoint: y, x: s.left - o.width * y.x + e.x, y: s.top - o.height * y.y + e.y };
      }) : [{ index: 0, relativePoint: null, x: e.x, y: e.y }];
    }, set: function(t) {
      var e = t.interaction, n = t.coords, r = t.state, i = r.options, o = r.offsets, a = $e(e.interactable, e.element, e.prepared.name), s = P({}, n), l = [];
      i.offsetWithOrigin || (s.x -= a.x, s.y -= a.y);
      for (var h = 0, u = o; h < u.length; h++)
        for (var m = u[h], y = s.x - m.x, E = s.y - m.y, x = 0, T = i.targets.length; x < T; x++) {
          var O = i.targets[x], z = void 0;
          (z = b.func(O) ? O(y, E, e._proxy, m, x) : O) && l.push({ x: (b.number(z.x) ? z.x : y) + m.x, y: (b.number(z.y) ? z.y : E) + m.y, range: b.number(z.range) ? z.range : i.range, source: O, index: x, offset: m });
        }
      for (var M = { target: null, inRange: !1, distance: 0, range: 0, delta: { x: 0, y: 0 } }, B = 0; B < l.length; B++) {
        var W = l[B], R = W.range, j = W.x - s.x, pe = W.y - s.y, Z = Ge(j, pe), be = Z <= R;
        R === 1 / 0 && M.inRange && M.range !== 1 / 0 && (be = !1), M.target && !(be ? M.inRange && R !== 1 / 0 ? Z / R < M.distance / M.range : R === 1 / 0 && M.range !== 1 / 0 || Z < M.distance : !M.inRange && Z < M.distance) || (M.target = W, M.distance = Z, M.range = R, M.inRange = be, M.delta.x = j, M.delta.y = pe);
      }
      return M.inRange && (n.x = M.target.x, n.y = M.target.y), r.closest = M, M;
    }, defaults: { range: 1 / 0, targets: null, offset: null, offsetWithOrigin: !0, origin: null, relativePoints: null, endOnly: !1, enabled: !1 } }, Zi = We(un, "snap"), Lt = { start: function(t) {
      var e = t.state, n = t.edges, r = e.options;
      if (!n)
        return null;
      t.state = { options: { targets: null, relativePoints: [{ x: n.left ? 0 : 1, y: n.top ? 0 : 1 }], offset: r.offset || "self", origin: { x: 0, y: 0 }, range: r.range } }, e.targetFields = e.targetFields || [["width", "height"], ["x", "y"]], un.start(t), e.offsets = t.state.offsets, t.state = e;
    }, set: function(t) {
      var e = t.interaction, n = t.state, r = t.coords, i = n.options, o = n.offsets, a = { x: r.x - o[0].x, y: r.y - o[0].y };
      n.options = P({}, i), n.options.targets = [];
      for (var s = 0, l = i.targets || []; s < l.length; s++) {
        var h = l[s], u = void 0;
        if (u = b.func(h) ? h(a.x, a.y, e) : h) {
          for (var m = 0, y = n.targetFields; m < y.length; m++) {
            var E = y[m], x = E[0], T = E[1];
            if (x in u || T in u) {
              u.x = u[x], u.y = u[T];
              break;
            }
          }
          n.options.targets.push(u);
        }
      }
      var O = un.set(t);
      return n.options = i, O;
    }, defaults: { range: 1 / 0, targets: null, offset: null, endOnly: !1, enabled: !1 } }, eo = We(Lt, "snapSize"), fn = { aspectRatio: Yi, restrictEdges: Gi, restrict: $i, restrictRect: Ki, restrictSize: Qi, snapEdges: We({ start: function(t) {
      var e = t.edges;
      return e ? (t.state.targetFields = t.state.targetFields || [[e.left ? "left" : "right", e.top ? "top" : "bottom"]], Lt.start(t)) : null;
    }, set: Lt.set, defaults: P(lt(Lt.defaults), { targets: void 0, range: void 0, offset: { x: 0, y: 0 } }) }, "snapEdges"), snap: Zi, snapSize: eo, spring: Ft, avoid: Ft, transform: Ft, rubberband: Ft }, to = { id: "modifiers", install: function(t) {
      var e = t.interactStatic;
      for (var n in t.usePlugin(Ln), t.usePlugin(Hi), e.modifiers = fn, fn) {
        var r = fn[n], i = r._defaults, o = r._methods;
        i._methods = o, t.defaults.perAction[n] = i;
      }
    } }, no = to, ir = function(t) {
      J(n, t);
      var e = ne(n);
      function n(r, i, o, a, s, l) {
        var h;
        if (w(this, n), it(ce(h = e.call(this, s)), o), o !== i && it(ce(h), i), h.timeStamp = l, h.originalEvent = o, h.type = r, h.pointerId = Q(i), h.pointerType = Fe(i), h.target = a, h.currentTarget = null, r === "tap") {
          var u = s.getPointerIndex(i);
          h.dt = h.timeStamp - s.pointers[u].downTime;
          var m = h.timeStamp - s.tapTime;
          h.double = !!s.prevTap && s.prevTap.type !== "doubletap" && s.prevTap.target === h.target && m < 500;
        } else
          r === "doubletap" && (h.dt = i.timeStamp - s.tapTime, h.double = !0);
        return h;
      }
      return g(n, [{ key: "_subtractOrigin", value: function(r) {
        var i = r.x, o = r.y;
        return this.pageX -= i, this.pageY -= o, this.clientX -= i, this.clientY -= o, this;
      } }, { key: "_addOrigin", value: function(r) {
        var i = r.x, o = r.y;
        return this.pageX += i, this.pageY += o, this.clientX += i, this.clientY += o, this;
      } }, { key: "preventDefault", value: function() {
        this.originalEvent.preventDefault();
      } }]), n;
    }(je), Et = { id: "pointer-events/base", before: ["inertia", "modifiers", "auto-start", "actions"], install: function(t) {
      t.pointerEvents = Et, t.defaults.actions.pointerEvents = Et.defaults, P(t.actions.phaselessTypes, Et.types);
    }, listeners: { "interactions:new": function(t) {
      var e = t.interaction;
      e.prevTap = null, e.tapTime = 0;
    }, "interactions:update-pointer": function(t) {
      var e = t.down, n = t.pointerInfo;
      !e && n.hold || (n.hold = { duration: 1 / 0, timeout: null });
    }, "interactions:move": function(t, e) {
      var n = t.interaction, r = t.pointer, i = t.event, o = t.eventTarget;
      t.duplicate || n.pointerIsDown && !n.pointerWasMoved || (n.pointerIsDown && dn(t), Ne({ interaction: n, pointer: r, event: i, eventTarget: o, type: "move" }, e));
    }, "interactions:down": function(t, e) {
      (function(n, r) {
        for (var i = n.interaction, o = n.pointer, a = n.event, s = n.eventTarget, l = n.pointerIndex, h = i.pointers[l].hold, u = pt(s), m = { interaction: i, pointer: o, event: a, eventTarget: s, type: "hold", targets: [], path: u, node: null }, y = 0; y < u.length; y++) {
          var E = u[y];
          m.node = E, r.fire("pointerEvents:collect-targets", m);
        }
        if (m.targets.length) {
          for (var x = 1 / 0, T = 0, O = m.targets; T < O.length; T++) {
            var z = O[T].eventable.options.holdDuration;
            z < x && (x = z);
          }
          h.duration = x, h.timeout = setTimeout(function() {
            Ne({ interaction: i, eventTarget: s, pointer: o, event: a, type: "hold" }, r);
          }, x);
        }
      })(t, e), Ne(t, e);
    }, "interactions:up": function(t, e) {
      dn(t), Ne(t, e), function(n, r) {
        var i = n.interaction, o = n.pointer, a = n.event, s = n.eventTarget;
        i.pointerWasMoved || Ne({ interaction: i, eventTarget: s, pointer: o, event: a, type: "tap" }, r);
      }(t, e);
    }, "interactions:cancel": function(t, e) {
      dn(t), Ne(t, e);
    } }, PointerEvent: ir, fire: Ne, collectEventTargets: or, defaults: { holdDuration: 600, ignoreFrom: null, allowFrom: null, origin: { x: 0, y: 0 } }, types: { down: !0, move: !0, up: !0, cancel: !0, tap: !0, doubletap: !0, hold: !0 } };
    function Ne(t, e) {
      var n = t.interaction, r = t.pointer, i = t.event, o = t.eventTarget, a = t.type, s = t.targets, l = s === void 0 ? or(t, e) : s, h = new ir(a, r, i, o, n, e.now());
      e.fire("pointerEvents:new", { pointerEvent: h });
      for (var u = { interaction: n, pointer: r, event: i, eventTarget: o, targets: l, type: a, pointerEvent: h }, m = 0; m < l.length; m++) {
        var y = l[m];
        for (var E in y.props || {})
          h[E] = y.props[E];
        var x = $e(y.eventable, y.node);
        if (h._subtractOrigin(x), h.eventable = y.eventable, h.currentTarget = y.node, y.eventable.fire(h), h._addOrigin(x), h.immediatePropagationStopped || h.propagationStopped && m + 1 < l.length && l[m + 1].node !== h.currentTarget)
          break;
      }
      if (e.fire("pointerEvents:fired", u), a === "tap") {
        var T = h.double ? Ne({ interaction: n, pointer: r, event: i, eventTarget: o, type: "doubletap" }, e) : h;
        n.prevTap = T, n.tapTime = T.timeStamp;
      }
      return h;
    }
    function or(t, e) {
      var n = t.interaction, r = t.pointer, i = t.event, o = t.eventTarget, a = t.type, s = n.getPointerIndex(r), l = n.pointers[s];
      if (a === "tap" && (n.pointerWasMoved || !l || l.downTarget !== o))
        return [];
      for (var h = pt(o), u = { interaction: n, pointer: r, event: i, eventTarget: o, type: a, path: h, targets: [], node: null }, m = 0; m < h.length; m++) {
        var y = h[m];
        u.node = y, e.fire("pointerEvents:collect-targets", u);
      }
      return a === "hold" && (u.targets = u.targets.filter(function(E) {
        var x, T;
        return E.eventable.options.holdDuration === ((x = n.pointers[s]) == null || (T = x.hold) == null ? void 0 : T.duration);
      })), u.targets;
    }
    function dn(t) {
      var e = t.interaction, n = t.pointerIndex, r = e.pointers[n].hold;
      r && r.timeout && (clearTimeout(r.timeout), r.timeout = null);
    }
    var ro = Object.freeze({ __proto__: null, default: Et });
    function io(t) {
      var e = t.interaction;
      e.holdIntervalHandle && (clearInterval(e.holdIntervalHandle), e.holdIntervalHandle = null);
    }
    var oo = { id: "pointer-events/holdRepeat", install: function(t) {
      t.usePlugin(Et);
      var e = t.pointerEvents;
      e.defaults.holdRepeatInterval = 0, e.types.holdrepeat = t.actions.phaselessTypes.holdrepeat = !0;
    }, listeners: ["move", "up", "cancel", "endall"].reduce(function(t, e) {
      return t["pointerEvents:".concat(e)] = io, t;
    }, { "pointerEvents:new": function(t) {
      var e = t.pointerEvent;
      e.type === "hold" && (e.count = (e.count || 0) + 1);
    }, "pointerEvents:fired": function(t, e) {
      var n = t.interaction, r = t.pointerEvent, i = t.eventTarget, o = t.targets;
      if (r.type === "hold" && o.length) {
        var a = o[0].eventable.options.holdRepeatInterval;
        a <= 0 || (n.holdIntervalHandle = setTimeout(function() {
          e.pointerEvents.fire({ interaction: n, eventTarget: i, type: "hold", pointer: r, event: r }, e);
        }, a));
      }
    } }) }, ao = oo, so = { id: "pointer-events/interactableTargets", install: function(t) {
      var e = t.Interactable;
      e.prototype.pointerEvents = function(r) {
        return P(this.events.options, r), this;
      };
      var n = e.prototype._backCompatOption;
      e.prototype._backCompatOption = function(r, i) {
        var o = n.call(this, r, i);
        return o === this && (this.events.options[r] = i), o;
      };
    }, listeners: { "pointerEvents:collect-targets": function(t, e) {
      var n = t.targets, r = t.node, i = t.type, o = t.eventTarget;
      e.interactables.forEachMatch(r, function(a) {
        var s = a.events, l = s.options;
        s.types[i] && s.types[i].length && a.testIgnoreAllow(l, r, o) && n.push({ node: r, eventable: s, props: { interactable: a } });
      });
    }, "interactable:new": function(t) {
      var e = t.interactable;
      e.events.getRect = function(n) {
        return e.getRect(n);
      };
    }, "interactable:set": function(t, e) {
      var n = t.interactable, r = t.options;
      P(n.events.options, e.pointerEvents.defaults), P(n.events.options, r.pointerEvents || {});
    } } }, co = so, lo = { id: "pointer-events", install: function(t) {
      t.usePlugin(ro), t.usePlugin(ao), t.usePlugin(co);
    } }, uo = lo, fo = { id: "reflow", install: function(t) {
      var e = t.Interactable;
      t.actions.phases.reflow = !0, e.prototype.reflow = function(n) {
        return function(r, i, o) {
          for (var a = r.getAllElements(), s = o.window.Promise, l = s ? [] : null, h = function() {
            var m = a[u], y = r.getRect(m);
            if (!y)
              return 1;
            var E, x = mt(o.interactions.list, function(z) {
              return z.interacting() && z.interactable === r && z.element === m && z.prepared.name === i.name;
            });
            if (x)
              x.move(), l && (E = x._reflowPromise || new s(function(z) {
                x._reflowResolve = z;
              }));
            else {
              var T = nt(y), O = function(z) {
                return { coords: z, get page() {
                  return this.coords.page;
                }, get client() {
                  return this.coords.client;
                }, get timeStamp() {
                  return this.coords.timeStamp;
                }, get pageX() {
                  return this.coords.page.x;
                }, get pageY() {
                  return this.coords.page.y;
                }, get clientX() {
                  return this.coords.client.x;
                }, get clientY() {
                  return this.coords.client.y;
                }, get pointerId() {
                  return this.coords.pointerId;
                }, get target() {
                  return this.coords.target;
                }, get type() {
                  return this.coords.type;
                }, get pointerType() {
                  return this.coords.pointerType;
                }, get buttons() {
                  return this.coords.buttons;
                }, preventDefault: function() {
                } };
              }({ page: { x: T.x, y: T.y }, client: { x: T.x, y: T.y }, timeStamp: o.now() });
              E = function(z, M, B, W, R) {
                var j = z.interactions.new({ pointerType: "reflow" }), pe = { interaction: j, event: R, pointer: R, eventTarget: B, phase: "reflow" };
                j.interactable = M, j.element = B, j.prevEvent = R, j.updatePointer(R, R, B, !0), S(j.coords.delta), Zt(j.prepared, W), j._doPhase(pe);
                var Z = z.window, be = Z.Promise, Se = be ? new be(function(Be) {
                  j._reflowResolve = Be;
                }) : void 0;
                return j._reflowPromise = Se, j.start(W, M, B), j._interacting ? (j.move(pe), j.end(R)) : (j.stop(), j._reflowResolve()), j.removePointer(R, R), Se;
              }(o, r, m, i, O);
            }
            l && l.push(E);
          }, u = 0; u < a.length && !h(); u++)
            ;
          return l && s.all(l).then(function() {
            return r;
          });
        }(this, n, t);
      };
    }, listeners: { "interactions:stop": function(t, e) {
      var n = t.interaction;
      n.pointerType === "reflow" && (n._reflowResolve && n._reflowResolve(), function(r, i) {
        r.splice(r.indexOf(i), 1);
      }(e.interactions.list, n));
    } } }, po = fo;
    if (de.use(jn), de.use(Xn), de.use(uo), de.use(Pi), de.use(no), de.use(mi), de.use(ai), de.use(ci), de.use(po), de.default = de, v(c) === "object" && c)
      try {
        c.exports = de;
      } catch {
      }
    return de.default = de, de;
  });
})(Go);
const mr = En, yr = /* @__PURE__ */ Ir({
  __name: "grid-item",
  props: {
    isDraggable: {
      type: Boolean,
      default: null
    },
    isResizable: {
      type: Boolean,
      default: null
    },
    isBounded: {
      type: Boolean,
      default: null
    },
    static: {
      type: Boolean,
      default: !1
    },
    minH: {
      type: Number,
      default: 1
    },
    minW: {
      type: Number,
      default: 1
    },
    maxH: {
      type: Number,
      default: 1 / 0
    },
    maxW: {
      type: Number,
      default: 1 / 0
    },
    x: {
      type: Number,
      required: !0
    },
    y: {
      type: Number,
      required: !0
    },
    w: {
      type: Number,
      required: !0
    },
    h: {
      type: Number,
      required: !0
    },
    i: {
      type: [Number, String],
      required: !0
    },
    dragIgnoreFrom: {
      type: String,
      default: "a, button"
    },
    dragAllowFrom: {
      type: String,
      default: null
    },
    resizeIgnoreFrom: {
      type: String,
      default: "a, button"
    },
    preserveAspectRatio: {
      type: Boolean,
      default: !1
    },
    dragOption: {
      type: Object,
      default: () => ({})
    },
    resizeOption: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["container-resized", "resize", "resized", "move", "moved"],
  setup(c, { expose: d, emit: p }) {
    const f = c, v = ar(Hr), w = ar(Nr);
    if (!v)
      throw new Error("[grid-layout-plus]: missing layout store, GridItem must under a GridLayout.");
    const k = Ot(null), g = Pt({
      cols: 1,
      containerWidth: 100,
      rowHeight: 30,
      margin: [10, 10],
      maxRows: 1 / 0,
      draggable: null,
      resizable: null,
      bounded: null,
      transformScale: 1,
      useCssTransforms: !0,
      useStyleCursor: !0,
      isDragging: !1,
      dragging: {
        top: -1,
        left: -1
      },
      isResizing: !1,
      resizing: {
        width: -1,
        height: -1
      },
      style: {},
      rtl: !1
    });
    let K = !1, J = !1, F = NaN, ae = NaN, ce = NaN, ne = NaN, Ee = -1, De = -1, Te = -1, ue = -1, V = f.x, oe = f.y, Y = f.w, re = f.h;
    const ve = Ot(), b = Pt({
      i: ho(f, "i"),
      state: g,
      wrapper: ve,
      calcXY: tt
    });
    function ge(S) {
      $e(S);
    }
    function xe() {
      Pe();
    }
    function _(S) {
      we(f.isDraggable) && (g.draggable = S);
    }
    function $(S) {
      we(f.isResizable) && (g.resizable = S);
    }
    function L(S) {
      we(f.isBounded) && (g.bounded = S);
    }
    function ie(S) {
      g.transformScale = S;
    }
    function q(S) {
      g.rowHeight = S;
    }
    function U(S) {
      g.maxRows = S;
    }
    function C() {
      g.rtl = vr() === "rtl", Pe();
    }
    function le(S) {
      g.cols = Math.floor(S);
    }
    v.increaseItem(b), Cr(() => {
      g.rtl = vr() === "rtl";
    }), Ar(() => {
      v.responsive && v.lastBreakpoint ? g.cols = xn(v.lastBreakpoint, v.cols) : g.cols = v.colNum, g.rowHeight = v.rowHeight, g.containerWidth = v.width !== null ? v.width : 100, g.margin = v.margin !== void 0 ? v.margin.map(Number) : [10, 10], g.maxRows = v.maxRows, we(f.isDraggable) ? g.draggable = v.isDraggable : g.draggable = f.isDraggable, we(f.isResizable) ? g.resizable = v.isResizable : g.resizable = f.isResizable, we(f.isBounded) ? g.bounded = v.isBounded : g.bounded = f.isBounded, g.transformScale = v.transformScale, g.useCssTransforms = v.useCssTransforms, g.useStyleCursor = v.useStyleCursor, go(() => {
        V = f.x, oe = f.y, re = f.h, Y = f.w, he(_e);
      }), w.on("updateWidth", ge), w.on("compact", xe), w.on("setDraggable", _), w.on("setResizable", $), w.on("setBounded", L), w.on("setTransformScale", ie), w.on("setRowHeight", q), w.on("setMaxRows", U), w.on("directionchange", C), w.on("setColNum", le);
    }), Br(() => {
      w.off("updateWidth", ge), w.off("compact", xe), w.off("setDraggable", _), w.off("setResizable", $), w.off("setBounded", L), w.off("setTransformScale", ie), w.off("setRowHeight", q), w.off("setMaxRows", U), w.off("directionchange", C), w.off("setColNum", le), k.value && (k.value.unset(), k.value = null), v.decreaseItem(b);
    }), d({ state: g, wrapper: ve });
    const Re = navigator.userAgent.toLowerCase().includes("android"), te = Ue(() => g.resizable && !f.static), G = Ue(() => v.isMirrored ? !g.rtl : g.rtl), ze = Ue(() => (g.draggable || g.resizable) && !f.static), me = Lo("item"), dt = Ue(() => ({
      [me.b()]: !0,
      [me.bm("resizable")]: te.value,
      [me.bm("static")]: f.static,
      [me.bm("resizing")]: g.isResizing,
      [me.bm("dragging")]: g.isDragging,
      [me.bm("transform")]: g.useCssTransforms,
      [me.bm("rtl")]: G.value,
      [me.bm("no-touch")]: Re && ze.value
    })), et = Ue(() => [me.be("resizer"), G.value && me.bem("resizer", "rtl")].filter(Boolean));
    ee(
      () => f.isDraggable,
      (S) => {
        g.draggable = S;
      }
    ), ee(
      () => f.static,
      () => {
        he(Dt), he(ke);
      }
    ), ee(
      () => g.draggable,
      () => {
        he(Dt);
      }
    ), ee(
      () => f.isResizable,
      (S) => {
        g.resizable = S;
      }
    ), ee(
      () => f.isBounded,
      (S) => {
        g.bounded = S;
      }
    ), ee(
      () => g.resizable,
      () => {
        he(ke);
      }
    ), ee(
      () => g.rowHeight,
      () => {
        he(_e), he(Xe);
      }
    ), ee([() => g.cols, () => g.containerWidth], () => {
      he(ke), he(_e), he(Xe);
    }), ee([() => f.minH, () => f.maxH, () => f.minW, () => f.maxW], () => {
      he(ke);
    }), ee(G, () => {
      he(ke), he(_e);
    }), ee([() => v.margin, () => v.margin[0], () => v.margin[1]], () => {
      const S = v.margin;
      !S || S[0] === g.margin[0] && S[1] === g.margin[1] || (g.margin = S.map(Number), he(_e), he(Xe));
    });
    function _e() {
      f.x + f.w > g.cols ? (V = 0, Y = f.w > g.cols ? g.cols : f.w) : (V = f.x, Y = f.w);
      const S = P(V, oe, Y, re);
      g.isDragging && (S.top = g.dragging.top, G.value ? S.right = g.dragging.left : S.left = g.dragging.left), g.isResizing && (S.width = g.resizing.width, S.height = g.resizing.height);
      let A;
      g.useCssTransforms ? G.value ? A = Ao(S.top, S.right, S.width, S.height) : A = Co(S.top, S.left, S.width, S.height) : G.value ? A = Fo(S.top, S.right, S.width, S.height) : A = Bo(S.top, S.left, S.width, S.height), g.style = A;
    }
    function Xe() {
      const S = {};
      for (const A of ["width", "height"]) {
        const X = g.style[A].match(/^(\d+)px$/);
        if (!X)
          return;
        S[A] = X[1];
      }
      p("container-resized", f.i, f.h, f.w, S.height, S.width);
    }
    function pt(S) {
      if (f.static)
        return;
      const A = S.type;
      if (A === "resizestart" && g.isResizing || A !== "resizestart" && !g.isResizing)
        return;
      const N = hr(S);
      if (we(N))
        return;
      const { x: X, y: Q } = N, H = { width: 0, height: 0 };
      let D;
      switch (A) {
        case "resizestart": {
          ke(), Ee = Y, De = re, D = P(V, oe, Y, re), H.width = D.width, H.height = D.height, g.resizing = H, g.isResizing = !0;
          break;
        }
        case "resizemove": {
          const se = gr(ce, ne, X, Q);
          G.value ? H.width = g.resizing.width - se.deltaX / g.transformScale : H.width = g.resizing.width + se.deltaX / g.transformScale, H.height = g.resizing.height + se.deltaY / g.transformScale, g.resizing = H;
          break;
        }
        case "resizeend": {
          D = P(V, oe, Y, re), H.width = D.width, H.height = D.height, g.resizing = { width: -1, height: -1 }, g.isResizing = !1;
          break;
        }
      }
      D = rt(H.height, H.width), D.w < f.minW && (D.w = f.minW), D.w > f.maxW && (D.w = f.maxW), D.h < f.minH && (D.h = f.minH), D.h > f.maxH && (D.h = f.maxH), D.h < 1 && (D.h = 1), D.w < 1 && (D.w = 1), ce = X, ne = Q, (Y !== D.w || re !== D.h) && p("resize", f.i, D.h, D.w, H.height, H.width), S.type === "resizeend" && (Ee !== Y || De !== re) && p("resized", f.i, D.h, D.w, H.height, H.width), w.emit("resizeEvent", S.type, f.i, V, oe, D.h, D.w);
    }
    function ht(S) {
      if (f.static || g.isResizing)
        return;
      const A = S.type;
      if (A === "dragstart" && g.isDragging || A !== "dragstart" && !g.isDragging)
        return;
      const N = hr(S);
      if (we(N))
        return;
      const { x: X, y: Q } = N, H = S.target;
      if (!H.offsetParent)
        return;
      const D = { top: 0, left: 0 };
      switch (A) {
        case "dragstart": {
          Te = V, ue = oe;
          const fe = H.offsetParent.getBoundingClientRect(), ye = H.getBoundingClientRect(), Ie = ye.left / g.transformScale, Fe = fe.left / g.transformScale, ot = ye.right / g.transformScale, je = fe.right / g.transformScale, at = ye.top / g.transformScale, st = fe.top / g.transformScale;
          G.value ? D.left = (ot - je) * -1 : D.left = Ie - Fe, D.top = at - st, g.dragging = D, g.isDragging = !0;
          break;
        }
        case "dragmove": {
          const fe = gr(F, ae, X, Q);
          if (G.value ? D.left = g.dragging.left - fe.deltaX / g.transformScale : D.left = g.dragging.left + fe.deltaX / g.transformScale, D.top = g.dragging.top + fe.deltaY / g.transformScale, g.bounded) {
            const ye = H.offsetParent.clientHeight - Ye(f.h, g.rowHeight, g.margin[1]);
            D.top = nt(D.top, 0, ye);
            const Ie = Oe(), Fe = g.containerWidth - Ye(f.w, Ie, g.margin[0]);
            D.left = nt(D.left, 0, Fe);
          }
          g.dragging = D;
          break;
        }
        case "dragend": {
          const fe = H.offsetParent.getBoundingClientRect(), ye = H.getBoundingClientRect(), Ie = ye.left / g.transformScale, Fe = fe.left / g.transformScale, ot = ye.right / g.transformScale, je = fe.right / g.transformScale, at = ye.top / g.transformScale, st = fe.top / g.transformScale;
          G.value ? D.left = (ot - je) * -1 : D.left = Ie - Fe, D.top = at - st, g.dragging = { top: -1, left: -1 }, g.isDragging = !1;
          break;
        }
      }
      let se;
      G.value, se = tt(D.top, D.left), F = X, ae = Q, (V !== se.x || oe !== se.y) && p("move", f.i, se.x, se.y), S.type === "dragend" && (Te !== V || ue !== oe) && p("moved", f.i, se.x, se.y), w.emit("dragEvent", S.type, f.i, se.x, se.y, re, Y);
    }
    function P(S, A, N, X) {
      const Q = Oe();
      let H;
      return G.value ? H = {
        right: Math.round(Q * S + (S + 1) * g.margin[0]),
        top: Math.round(g.rowHeight * A + (A + 1) * g.margin[1]),
        // 0 * Infinity === NaN, which causes problems with resize constraints;
        // Fix this if it occurs.
        // Note we do it here rather than later because Math.round(Infinity) causes depot
        width: N === 1 / 0 ? N : Math.round(Q * N + Math.max(0, N - 1) * g.margin[0]),
        height: X === 1 / 0 ? X : Math.round(g.rowHeight * X + Math.max(0, X - 1) * g.margin[1])
      } : H = {
        left: Math.round(Q * S + (S + 1) * g.margin[0]),
        top: Math.round(g.rowHeight * A + (A + 1) * g.margin[1]),
        // 0 * Infinity === NaN, which causes problems with resize constraints;
        // Fix this if it occurs.
        // Note we do it here rather than later because Math.round(Infinity) causes depot
        width: N === 1 / 0 ? N : Math.round(Q * N + Math.max(0, N - 1) * g.margin[0]),
        height: X === 1 / 0 ? X : Math.round(g.rowHeight * X + Math.max(0, X - 1) * g.margin[1])
      }, H;
    }
    function tt(S, A) {
      const N = Oe();
      let X = Math.round((A - g.margin[0]) / (N + g.margin[0])), Q = Math.round((S - g.margin[1]) / (g.rowHeight + g.margin[1]));
      return X = Math.max(Math.min(X, g.cols - Y), 0), Q = Math.max(Math.min(Q, g.maxRows - re), 0), { x: X, y: Q };
    }
    function Oe() {
      return (g.containerWidth - g.margin[0] * (g.cols + 1)) / g.cols;
    }
    function Ye(S, A, N) {
      return Number.isFinite(S) ? Math.round(A * S + Math.max(0, S - 1) * N) : S;
    }
    function nt(S, A, N) {
      return Math.max(Math.min(S, N), A);
    }
    function rt(S, A, N = !1) {
      const X = Oe();
      let Q = Math.round((A + g.margin[0]) / (X + g.margin[0])), H = 0;
      return N ? H = Math.ceil((S + g.margin[1]) / (g.rowHeight + g.margin[1])) : H = Math.round((S + g.margin[1]) / (g.rowHeight + g.margin[1])), Q = Math.max(Math.min(Q, g.cols - V), 0), H = Math.max(Math.min(H, g.maxRows - oe), 0), { w: Q, h: H };
    }
    function $e(S, A) {
      g.containerWidth = S, A != null && (g.cols = A);
    }
    function Pe() {
      _e();
    }
    function gt() {
      !k.value && ve.value && (k.value = mr(ve.value), g.useStyleCursor || k.value.styleCursor(!1));
    }
    const Ge = fr(ht);
    function Dt() {
      if (gt(), !!k.value)
        if (g.draggable && !f.static) {
          const S = {
            ignoreFrom: f.dragIgnoreFrom,
            allowFrom: f.dragAllowFrom,
            ...f.dragOption
          };
          k.value.draggable(S), K || (K = !0, k.value.on("dragstart dragmove dragend", (A) => {
            A.type === "dragmove" ? Ge(A) : ht(A);
          }));
        } else
          k.value.draggable({ enabled: !1 });
    }
    const it = fr(pt);
    function ke() {
      if (gt(), !!k.value)
        if (g.resizable && !f.static) {
          const S = P(0, 0, f.maxW, f.maxH), A = P(0, 0, f.minW, f.minH), N = {
            edges: {
              left: !1,
              right: `.${et.value[0]}`,
              bottom: `.${et.value[0]}`,
              top: !1
            },
            ignoreFrom: f.resizeIgnoreFrom,
            restrictSize: {
              min: {
                height: A.height * g.transformScale,
                width: A.width * g.transformScale
              },
              max: {
                height: S.height * g.transformScale,
                width: S.width * g.transformScale
              }
            },
            ...f.resizeOption
          };
          f.preserveAspectRatio && (N.modifiers = [mr.modifiers.aspectRatio({ ratio: "preserve" })]), k.value.resizable(N), J || (J = !0, k.value.on("resizestart resizemove resizeend", (X) => {
            X.type === "resizemove" ? it(X) : pt(X);
          }));
        } else
          k.value.resizable({ enabled: !1 });
    }
    return (S, A) => (St(), Yt("section", {
      ref_key: "wrapper",
      ref: ve,
      class: sr(pn(dt)),
      style: Fr(g.style)
    }, [
      yn(S.$slots, "default"),
      pn(te) ? (St(), Yt("span", {
        key: 0,
        class: sr(pn(et))
      }, null, 2)) : vo("", !0)
    ], 6));
  }
}), Ze = typeof window < "u";
var br;
Ze && (br = window == null ? void 0 : window.navigator) != null && br.userAgent && /iP(ad|hone|od)/.test(window.navigator.userAgent);
function wr(c) {
  return c != null;
}
function hn() {
}
const Vo = Object.freeze({
  aliceblue: "f0f8ff",
  antiquewhite: "faebd7",
  aqua: "0ff",
  aquamarine: "7fffd4",
  azure: "f0ffff",
  beige: "f5f5dc",
  bisque: "ffe4c4",
  black: "000",
  blanchedalmond: "ffebcd",
  blue: "00f",
  blueviolet: "8a2be2",
  brown: "a52a2a",
  burlywood: "deb887",
  burntsienna: "ea7e5d",
  cadetblue: "5f9ea0",
  chartreuse: "7fff00",
  chocolate: "d2691e",
  coral: "ff7f50",
  cornflowerblue: "6495ed",
  cornsilk: "fff8dc",
  crimson: "dc143c",
  cyan: "0ff",
  darkblue: "00008b",
  darkcyan: "008b8b",
  darkgoldenrod: "b8860b",
  darkgray: "a9a9a9",
  darkgreen: "006400",
  darkgrey: "a9a9a9",
  darkkhaki: "bdb76b",
  darkmagenta: "8b008b",
  darkolivegreen: "556b2f",
  darkorange: "ff8c00",
  darkorchid: "9932cc",
  darkred: "8b0000",
  darksalmon: "e9967a",
  darkseagreen: "8fbc8f",
  darkslateblue: "483d8b",
  darkslategray: "2f4f4f",
  darkslategrey: "2f4f4f",
  darkturquoise: "00ced1",
  darkviolet: "9400d3",
  deeppink: "ff1493",
  deepskyblue: "00bfff",
  dimgray: "696969",
  dimgrey: "696969",
  dodgerblue: "1e90ff",
  firebrick: "b22222",
  floralwhite: "fffaf0",
  forestgreen: "228b22",
  fuchsia: "f0f",
  gainsboro: "dcdcdc",
  ghostwhite: "f8f8ff",
  gold: "ffd700",
  goldenrod: "daa520",
  gray: "808080",
  green: "008000",
  greenyellow: "adff2f",
  grey: "808080",
  honeydew: "f0fff0",
  hotpink: "ff69b4",
  indianred: "cd5c5c",
  indigo: "4b0082",
  ivory: "fffff0",
  khaki: "f0e68c",
  lavender: "e6e6fa",
  lavenderblush: "fff0f5",
  lawngreen: "7cfc00",
  lemonchiffon: "fffacd",
  lightblue: "add8e6",
  lightcoral: "f08080",
  lightcyan: "e0ffff",
  lightgoldenrodyellow: "fafad2",
  lightgray: "d3d3d3",
  lightgreen: "90ee90",
  lightgrey: "d3d3d3",
  lightpink: "ffb6c1",
  lightsalmon: "ffa07a",
  lightseagreen: "20b2aa",
  lightskyblue: "87cefa",
  lightslategray: "789",
  lightslategrey: "789",
  lightsteelblue: "b0c4de",
  lightyellow: "ffffe0",
  lime: "0f0",
  limegreen: "32cd32",
  linen: "faf0e6",
  magenta: "f0f",
  maroon: "800000",
  mediumaquamarine: "66cdaa",
  mediumblue: "0000cd",
  mediumorchid: "ba55d3",
  mediumpurple: "9370db",
  mediumseagreen: "3cb371",
  mediumslateblue: "7b68ee",
  mediumspringgreen: "00fa9a",
  mediumturquoise: "48d1cc",
  mediumvioletred: "c71585",
  midnightblue: "191970",
  mintcream: "f5fffa",
  mistyrose: "ffe4e1",
  moccasin: "ffe4b5",
  navajowhite: "ffdead",
  navy: "000080",
  oldlace: "fdf5e6",
  olive: "808000",
  olivedrab: "6b8e23",
  orange: "ffa500",
  orangered: "ff4500",
  orchid: "da70d6",
  palegoldenrod: "eee8aa",
  palegreen: "98fb98",
  paleturquoise: "afeeee",
  palevioletred: "db7093",
  papayawhip: "ffefd5",
  peachpuff: "ffdab9",
  peru: "cd853f",
  pink: "ffc0cb",
  plum: "dda0dd",
  powderblue: "b0e0e6",
  purple: "800080",
  rebeccapurple: "663399",
  red: "f00",
  rosybrown: "bc8f8f",
  royalblue: "4169e1",
  saddlebrown: "8b4513",
  salmon: "fa8072",
  sandybrown: "f4a460",
  seagreen: "2e8b57",
  seashell: "fff5ee",
  sienna: "a0522d",
  silver: "c0c0c0",
  skyblue: "87ceeb",
  slateblue: "6a5acd",
  slategray: "708090",
  slategrey: "708090",
  snow: "fffafa",
  springgreen: "00ff7f",
  steelblue: "4682b4",
  tan: "d2b48c",
  teal: "008080",
  thistle: "d8bfd8",
  tomato: "ff6347",
  turquoise: "40e0d0",
  violet: "ee82ee",
  wheat: "f5deb3",
  white: "fff",
  whitesmoke: "f5f5f5",
  yellow: "ff0",
  yellowgreen: "9acd32"
});
Object.freeze(new Set(Object.keys(Vo)));
const Ko = Ze && ("ontouchstart" in window || Jo() > 0), Uo = Ko ? "pointerdown" : "click";
function Jo() {
  return typeof navigator < "u" && (navigator.maxTouchPoints || navigator.msMaxTouchPoints) || 0;
}
function Qo(c, d, p = window.Event) {
  const { type: f, bubbles: v = !1, cancelable: w = !1, ...k } = d;
  if (!wr(f) || f === "")
    return !1;
  let g;
  return wr(p) ? g = new p(f, { bubbles: v, cancelable: w }) : (g = document.createEvent("HTMLEvents"), g.initEvent(f, v, w)), Object.assign(g, k), c.dispatchEvent(g);
}
const Zo = "clickoutside", ea = /* @__PURE__ */ new Set();
Ze && document.addEventListener(
  Uo,
  (c) => {
    const d = c.target, p = c.composedPath && c.composedPath();
    ea.forEach((f) => {
      f !== d && (p ? !p.includes(f) : !f.contains(d)) && (!f.__transferElement || f.__transferElement !== d && !f.__transferElement.contains(d)) && Qo(f, { type: Zo });
    });
  },
  !0
);
const ta = [
  [
    "requestFullscreen",
    "exitFullscreen",
    "fullscreenElement",
    "fullscreenEnabled",
    "fullscreenchange",
    "fullscreenerror"
  ],
  // New WebKit
  [
    "webkitRequestFullscreen",
    "webkitExitFullscreen",
    "webkitFullscreenElement",
    "webkitFullscreenEnabled",
    "webkitfullscreenchange",
    "webkitfullscreenerror"
  ],
  // Old WebKit
  [
    "webkitRequestFullScreen",
    "webkitCancelFullScreen",
    "webkitCurrentFullScreenElement",
    "webkitCancelFullScreen",
    "webkitfullscreenchange",
    "webkitfullscreenerror"
  ],
  [
    "mozRequestFullScreen",
    "mozCancelFullScreen",
    "mozFullScreenElement",
    "mozFullScreenEnabled",
    "mozfullscreenchange",
    "mozfullscreenerror"
  ],
  [
    "msRequestFullscreen",
    "msExitFullscreen",
    "msFullscreenElement",
    "msFullscreenEnabled",
    "MSFullscreenChange",
    "MSFullscreenError"
  ]
];
let $t;
if (Ze) {
  for (const c of ta)
    if (c[1] in document) {
      $t = c;
      break;
    }
}
const na = !!$t;
Ue(() => !1);
const ra = /* @__PURE__ */ new Set(), ia = /* @__PURE__ */ new WeakMap();
if (Ze && $t) {
  const c = $t[4];
  document.addEventListener(
    c,
    (d) => {
      if (ra.forEach((p) => {
        p.value = !1;
      }), d.target) {
        const p = ia.get(d.target);
        p && (p.value = !0);
      }
    },
    !1
  );
}
const Vr = /* @__PURE__ */ new Map();
Vr.set("x", 0);
Vr.set("y", 0);
var Je = [], oa = function() {
  return Je.some(function(c) {
    return c.activeTargets.length > 0;
  });
}, aa = function() {
  return Je.some(function(c) {
    return c.skippedTargets.length > 0;
  });
}, xr = "ResizeObserver loop completed with undelivered notifications.", sa = function() {
  var c;
  typeof ErrorEvent == "function" ? c = new ErrorEvent("error", {
    message: xr
  }) : (c = document.createEvent("Event"), c.initEvent("error", !1, !1), c.message = xr), window.dispatchEvent(c);
}, Mt;
(function(c) {
  c.BORDER_BOX = "border-box", c.CONTENT_BOX = "content-box", c.DEVICE_PIXEL_CONTENT_BOX = "device-pixel-content-box";
})(Mt || (Mt = {}));
var Qe = function(c) {
  return Object.freeze(c);
}, ca = /* @__PURE__ */ function() {
  function c(d, p) {
    this.inlineSize = d, this.blockSize = p, Qe(this);
  }
  return c;
}(), Kr = function() {
  function c(d, p, f, v) {
    return this.x = d, this.y = p, this.width = f, this.height = v, this.top = this.y, this.left = this.x, this.bottom = this.top + this.height, this.right = this.left + this.width, Qe(this);
  }
  return c.prototype.toJSON = function() {
    var d = this, p = d.x, f = d.y, v = d.top, w = d.right, k = d.bottom, g = d.left, K = d.width, J = d.height;
    return { x: p, y: f, top: v, right: w, bottom: k, left: g, width: K, height: J };
  }, c.fromRect = function(d) {
    return new c(d.x, d.y, d.width, d.height);
  }, c;
}(), Sn = function(c) {
  return c instanceof SVGElement && "getBBox" in c;
}, Ur = function(c) {
  if (Sn(c)) {
    var d = c.getBBox(), p = d.width, f = d.height;
    return !p && !f;
  }
  var v = c, w = v.offsetWidth, k = v.offsetHeight;
  return !(w || k || c.getClientRects().length);
}, Er = function(c) {
  var d;
  if (c instanceof Element)
    return !0;
  var p = (d = c == null ? void 0 : c.ownerDocument) === null || d === void 0 ? void 0 : d.defaultView;
  return !!(p && c instanceof p.Element);
}, la = function(c) {
  switch (c.tagName) {
    case "INPUT":
      if (c.type !== "image")
        break;
    case "VIDEO":
    case "AUDIO":
    case "EMBED":
    case "OBJECT":
    case "CANVAS":
    case "IFRAME":
    case "IMG":
      return !0;
  }
  return !1;
}, _t = typeof window < "u" ? window : {}, Ht = /* @__PURE__ */ new WeakMap(), kr = /auto|scroll/, ua = /^tb|vertical/, fa = /msie|trident/i.test(_t.navigator && _t.navigator.userAgent), Me = function(c) {
  return parseFloat(c || "0");
}, ft = function(c, d, p) {
  return c === void 0 && (c = 0), d === void 0 && (d = 0), p === void 0 && (p = !1), new ca((p ? d : c) || 0, (p ? c : d) || 0);
}, Sr = Qe({
  devicePixelContentBoxSize: ft(),
  borderBoxSize: ft(),
  contentBoxSize: ft(),
  contentRect: new Kr(0, 0, 0, 0)
}), Jr = function(c, d) {
  if (d === void 0 && (d = !1), Ht.has(c) && !d)
    return Ht.get(c);
  if (Ur(c))
    return Ht.set(c, Sr), Sr;
  var p = getComputedStyle(c), f = Sn(c) && c.ownerSVGElement && c.getBBox(), v = !fa && p.boxSizing === "border-box", w = ua.test(p.writingMode || ""), k = !f && kr.test(p.overflowY || ""), g = !f && kr.test(p.overflowX || ""), K = f ? 0 : Me(p.paddingTop), J = f ? 0 : Me(p.paddingRight), F = f ? 0 : Me(p.paddingBottom), ae = f ? 0 : Me(p.paddingLeft), ce = f ? 0 : Me(p.borderTopWidth), ne = f ? 0 : Me(p.borderRightWidth), Ee = f ? 0 : Me(p.borderBottomWidth), De = f ? 0 : Me(p.borderLeftWidth), Te = ae + J, ue = K + F, V = De + ne, oe = ce + Ee, Y = g ? c.offsetHeight - oe - c.clientHeight : 0, re = k ? c.offsetWidth - V - c.clientWidth : 0, ve = v ? Te + V : 0, b = v ? ue + oe : 0, ge = f ? f.width : Me(p.width) - ve - re, xe = f ? f.height : Me(p.height) - b - Y, _ = ge + Te + re + V, $ = xe + ue + Y + oe, L = Qe({
    devicePixelContentBoxSize: ft(Math.round(ge * devicePixelRatio), Math.round(xe * devicePixelRatio), w),
    borderBoxSize: ft(_, $, w),
    contentBoxSize: ft(ge, xe, w),
    contentRect: new Kr(ae, K, ge, xe)
  });
  return Ht.set(c, L), L;
}, Qr = function(c, d, p) {
  var f = Jr(c, p), v = f.borderBoxSize, w = f.contentBoxSize, k = f.devicePixelContentBoxSize;
  switch (d) {
    case Mt.DEVICE_PIXEL_CONTENT_BOX:
      return k;
    case Mt.BORDER_BOX:
      return v;
    default:
      return w;
  }
}, da = /* @__PURE__ */ function() {
  function c(d) {
    var p = Jr(d);
    this.target = d, this.contentRect = p.contentRect, this.borderBoxSize = Qe([p.borderBoxSize]), this.contentBoxSize = Qe([p.contentBoxSize]), this.devicePixelContentBoxSize = Qe([p.devicePixelContentBoxSize]);
  }
  return c;
}(), Zr = function(c) {
  if (Ur(c))
    return 1 / 0;
  for (var d = 0, p = c.parentNode; p; )
    d += 1, p = p.parentNode;
  return d;
}, pa = function() {
  var c = 1 / 0, d = [];
  Je.forEach(function(w) {
    if (w.activeTargets.length !== 0) {
      var k = [];
      w.activeTargets.forEach(function(g) {
        var K = new da(g.target), J = Zr(g.target);
        k.push(K), g.lastReportedSize = Qr(g.target, g.observedBox), J < c && (c = J);
      }), d.push(function() {
        w.callback.call(w.observer, k, w.observer);
      }), w.activeTargets.splice(0, w.activeTargets.length);
    }
  });
  for (var p = 0, f = d; p < f.length; p++) {
    var v = f[p];
    v();
  }
  return c;
}, Tr = function(c) {
  Je.forEach(function(d) {
    d.activeTargets.splice(0, d.activeTargets.length), d.skippedTargets.splice(0, d.skippedTargets.length), d.observationTargets.forEach(function(p) {
      p.isActive() && (Zr(p.target) > c ? d.activeTargets.push(p) : d.skippedTargets.push(p));
    });
  });
}, ha = function() {
  var c = 0;
  for (Tr(c); oa(); )
    c = pa(), Tr(c);
  return aa() && sa(), c > 0;
}, gn, ei = [], ga = function() {
  return ei.splice(0).forEach(function(c) {
    return c();
  });
}, va = function(c) {
  if (!gn) {
    var d = 0, p = document.createTextNode(""), f = { characterData: !0 };
    new MutationObserver(function() {
      return ga();
    }).observe(p, f), gn = function() {
      p.textContent = "".concat(d ? d-- : d++);
    };
  }
  ei.push(c), gn();
}, ma = function(c) {
  va(function() {
    requestAnimationFrame(c);
  });
}, Xt = 0, ya = function() {
  return !!Xt;
}, ba = 250, wa = { attributes: !0, characterData: !0, childList: !0, subtree: !0 }, zr = [
  "resize",
  "load",
  "transitionend",
  "animationend",
  "animationstart",
  "animationiteration",
  "keyup",
  "keydown",
  "mouseup",
  "mousedown",
  "mouseover",
  "mouseout",
  "blur",
  "focus"
], _r = function(c) {
  return c === void 0 && (c = 0), Date.now() + c;
}, vn = !1, xa = function() {
  function c() {
    var d = this;
    this.stopped = !0, this.listener = function() {
      return d.schedule();
    };
  }
  return c.prototype.run = function(d) {
    var p = this;
    if (d === void 0 && (d = ba), !vn) {
      vn = !0;
      var f = _r(d);
      ma(function() {
        var v = !1;
        try {
          v = ha();
        } finally {
          if (vn = !1, d = f - _r(), !ya())
            return;
          v ? p.run(1e3) : d > 0 ? p.run(d) : p.start();
        }
      });
    }
  }, c.prototype.schedule = function() {
    this.stop(), this.run();
  }, c.prototype.observe = function() {
    var d = this, p = function() {
      return d.observer && d.observer.observe(document.body, wa);
    };
    document.body ? p() : _t.addEventListener("DOMContentLoaded", p);
  }, c.prototype.start = function() {
    var d = this;
    this.stopped && (this.stopped = !1, this.observer = new MutationObserver(this.listener), this.observe(), zr.forEach(function(p) {
      return _t.addEventListener(p, d.listener, !0);
    }));
  }, c.prototype.stop = function() {
    var d = this;
    this.stopped || (this.observer && this.observer.disconnect(), zr.forEach(function(p) {
      return _t.removeEventListener(p, d.listener, !0);
    }), this.stopped = !0);
  }, c;
}(), kn = new xa(), Or = function(c) {
  !Xt && c > 0 && kn.start(), Xt += c, !Xt && kn.stop();
}, Ea = function(c) {
  return !Sn(c) && !la(c) && getComputedStyle(c).display === "inline";
}, ka = function() {
  function c(d, p) {
    this.target = d, this.observedBox = p || Mt.CONTENT_BOX, this.lastReportedSize = {
      inlineSize: 0,
      blockSize: 0
    };
  }
  return c.prototype.isActive = function() {
    var d = Qr(this.target, this.observedBox, !0);
    return Ea(this.target) && (this.lastReportedSize = d), this.lastReportedSize.inlineSize !== d.inlineSize || this.lastReportedSize.blockSize !== d.blockSize;
  }, c;
}(), Sa = /* @__PURE__ */ function() {
  function c(d, p) {
    this.activeTargets = [], this.skippedTargets = [], this.observationTargets = [], this.observer = d, this.callback = p;
  }
  return c;
}(), Nt = /* @__PURE__ */ new WeakMap(), Pr = function(c, d) {
  for (var p = 0; p < c.length; p += 1)
    if (c[p].target === d)
      return p;
  return -1;
}, qt = function() {
  function c() {
  }
  return c.connect = function(d, p) {
    var f = new Sa(d, p);
    Nt.set(d, f);
  }, c.observe = function(d, p, f) {
    var v = Nt.get(d), w = v.observationTargets.length === 0;
    Pr(v.observationTargets, p) < 0 && (w && Je.push(v), v.observationTargets.push(new ka(p, f && f.box)), Or(1), kn.schedule());
  }, c.unobserve = function(d, p) {
    var f = Nt.get(d), v = Pr(f.observationTargets, p), w = f.observationTargets.length === 1;
    v >= 0 && (w && Je.splice(Je.indexOf(f), 1), f.observationTargets.splice(v, 1), Or(-1));
  }, c.disconnect = function(d) {
    var p = this, f = Nt.get(d);
    f.observationTargets.slice().forEach(function(v) {
      return p.unobserve(d, v.target);
    }), f.activeTargets.splice(0, f.activeTargets.length);
  }, c;
}(), Ta = function() {
  function c(d) {
    if (arguments.length === 0)
      throw new TypeError("Failed to construct 'ResizeObserver': 1 argument required, but only 0 present.");
    if (typeof d != "function")
      throw new TypeError("Failed to construct 'ResizeObserver': The callback provided as parameter 1 is not a function.");
    qt.connect(this, d);
  }
  return c.prototype.observe = function(d, p) {
    if (arguments.length === 0)
      throw new TypeError("Failed to execute 'observe' on 'ResizeObserver': 1 argument required, but only 0 present.");
    if (!Er(d))
      throw new TypeError("Failed to execute 'observe' on 'ResizeObserver': parameter 1 is not of type 'Element");
    qt.observe(this, d, p);
  }, c.prototype.unobserve = function(d) {
    if (arguments.length === 0)
      throw new TypeError("Failed to execute 'unobserve' on 'ResizeObserver': 1 argument required, but only 0 present.");
    if (!Er(d))
      throw new TypeError("Failed to execute 'unobserve' on 'ResizeObserver': parameter 1 is not of type 'Element");
    qt.unobserve(this, d);
  }, c.prototype.disconnect = function() {
    qt.disconnect(this);
  }, c.toString = function() {
    return "function ResizeObserver () { [polyfill code] }";
  }, c;
}();
const Gt = /* @__PURE__ */ new WeakMap();
function za(c) {
  for (let d = 0, p = c.length; d < p; ++d) {
    const f = c[d], v = Gt.get(f.target);
    typeof v == "function" && v(f);
  }
}
const ti = new (Ze && window.ResizeObserver || Ta)(
  za
);
function _a(c, d) {
  Gt.set(c, d), ti.observe(c);
}
function Oa(c) {
  Gt.has(c) && (ti.unobserve(c), Gt.delete(c));
}
function Pa() {
  return {
    observeResize: _a,
    unobserveResize: Oa
  };
}
const Ma = Ot(!1);
Ue(() => Ma.value);
const Mr = "__theme_style__", mn = "__theme_observer__", Dr = Pt(/* @__PURE__ */ new Map()), Rr = /* @__PURE__ */ new Map();
ee(Dr, () => {
  if (!Ze)
    return;
  Rr.clear();
  const c = document.head.querySelector(`#${Mr}`);
  c && document.head.removeChild(c);
  const d = document.createElement("style");
  let p = `.${mn} { width: 1px }`, f = 1;
  for (const [v, [w, k]] of Dr.entries())
    p += ` html.${w} .${mn}, .${k} .${mn} { width: ${++f}px }`, Rr.set(f, v);
  d.textContent = p, d.id = Mr, document.head.appendChild(d);
});
const Ra = /* @__PURE__ */ Ir({
  __name: "grid-layout",
  props: {
    autoSize: {
      type: Boolean,
      default: !0
    },
    colNum: {
      type: Number,
      default: 12
    },
    rowHeight: {
      type: Number,
      default: 150
    },
    maxRows: {
      type: Number,
      default: 1 / 0
    },
    margin: {
      type: Array,
      default: () => [10, 10]
    },
    isDraggable: {
      type: Boolean,
      default: !0
    },
    isResizable: {
      type: Boolean,
      default: !0
    },
    isMirrored: {
      type: Boolean,
      default: !1
    },
    isBounded: {
      type: Boolean,
      default: !1
    },
    useCssTransforms: {
      type: Boolean,
      default: !0
    },
    verticalCompact: {
      type: Boolean,
      default: !0
    },
    restoreOnDrag: {
      type: Boolean,
      default: !1
    },
    layout: {
      type: Array,
      required: !0
    },
    responsive: {
      type: Boolean,
      default: !1
    },
    responsiveLayouts: {
      type: Object,
      default: () => ({})
    },
    transformScale: {
      type: Number,
      default: 1
    },
    breakpoints: {
      type: Object,
      default: () => ({ lg: 1200, md: 996, sm: 768, xs: 480, xxs: 0 })
    },
    cols: {
      type: Object,
      default: () => ({ lg: 12, md: 10, sm: 6, xs: 4, xxs: 2 })
    },
    preventCollision: {
      type: Boolean,
      default: !1
    },
    useStyleCursor: {
      type: Boolean,
      default: !0
    }
  },
  emits: [
    "layout-before-mount",
    "layout-mounted",
    "layout-updated",
    "breakpoint-changed",
    "update:layout",
    "layout-ready"
  ],
  setup(c, { expose: d, emit: p }) {
    const f = c, v = Pt({
      width: -1,
      mergedStyle: {},
      lastLayoutLength: 0,
      isDragging: !1,
      placeholder: {
        x: 0,
        y: 0,
        w: 0,
        h: 0,
        i: ""
      },
      layouts: {},
      // array to store all layouts from different breakpoints
      lastBreakpoint: null,
      // store last active breakpoint
      originalLayout: null
      // store original Layout
    }), w = /* @__PURE__ */ new Map(), k = Ot(f.layout), g = Ot(), { observeResize: K, unobserveResize: J } = Pa(), F = _o();
    F.on("resizeEvent", ae), F.on("dragEvent", ce), Cr(() => {
      p("layout-before-mount", k.value);
    }), Ar(() => {
      p("layout-mounted", k.value), qe(() => {
        jo(k.value), v.originalLayout = k.value, qe(() => {
          ge(), g.value && K(g.value, Oo(V, 16)), ut(k.value, f.verticalCompact), p("layout-updated", k.value), ue(), V();
        });
      });
    }), Br(() => {
      F.clearAll(), g.value && J(g.value);
    });
    function ae(_, $, L, ie, q, U) {
      ve(_, $, L, ie, q, U);
    }
    function ce(_, $, L, ie, q, U) {
      re(_, $, L, ie, q, U);
    }
    ee(
      () => v.width,
      (_, $) => {
        qe(() => {
          F.emit("updateWidth", _), $ === -1 && qe(() => {
            p("layout-ready", k.value);
          }), ue();
        });
      }
    ), ee(
      () => [f.layout, f.layout.length],
      () => {
        k.value = f.layout, Te();
      }
    ), ee(
      () => f.colNum,
      (_) => {
        F.emit("setColNum", _);
      }
    ), ee(
      () => f.rowHeight,
      (_) => {
        F.emit("setRowHeight", _);
      }
    ), ee(
      () => f.isDraggable,
      (_) => {
        F.emit("setDraggable", _);
      }
    ), ee(
      () => f.isResizable,
      (_) => {
        F.emit("setResizable", _);
      }
    ), ee(
      () => f.isBounded,
      (_) => {
        F.emit("setBounded", _);
      }
    ), ee(
      () => f.transformScale,
      (_) => {
        F.emit("setTransformScale", _);
      }
    ), ee(
      () => f.responsive,
      (_) => {
        _ || (p("update:layout", v.originalLayout), F.emit("setColNum", f.colNum)), V();
      }
    ), ee(
      () => f.maxRows,
      (_) => {
        F.emit("setMaxRows", _);
      }
    ), ee([() => f.margin, () => f.margin[1]], ue), cr(
      Hr,
      Pt({
        ...lr(f),
        ...lr(v),
        increaseItem: ne,
        decreaseItem: Ee
      })
    ), cr(Nr, F), d({ state: v, getItem: De, resizeEvent: ve, dragEvent: re });
    function ne(_) {
      w.set(_.i, _);
    }
    function Ee(_) {
      w.delete(_.i);
    }
    function De(_) {
      return w.get(_);
    }
    function Te() {
      if (!we(k.value) && !we(v.originalLayout)) {
        if (k.value.length !== v.originalLayout.length) {
          const _ = xe(k.value, v.originalLayout);
          if (_.length > 0)
            if (k.value.length > v.originalLayout.length)
              v.originalLayout = v.originalLayout.concat(_);
            else {
              const $ = new Set(_.map((L) => L.i));
              v.originalLayout = v.originalLayout.filter((L) => !$.has(L.i));
            }
          v.lastLayoutLength = k.value.length, ge();
        }
        ut(k.value, f.verticalCompact), F.emit("updateWidth", v.width), ue(), p("layout-updated", k.value);
      }
    }
    function ue() {
      v.mergedStyle = {
        height: oe()
      };
    }
    function V() {
      g.value && (v.width = g.value.offsetWidth), F.emit("resizeEvent");
    }
    function oe() {
      if (!f.autoSize)
        return;
      const _ = parseFloat(f.margin[1]);
      return Mo(k.value) * (f.rowHeight + _) + _ + "px";
    }
    let Y;
    function re(_, $, L, ie, q, U) {
      let C = dr(k.value, $);
      we(C) && (C = { h: 0, w: 0, x: 0, y: 0, i: "" }), _ === "dragstart" && !f.verticalCompact && (Y = k.value.reduce(
        (le, { i: Re, x: te, y: G }) => ({
          ...le,
          [Re]: { x: te, y: G }
        }),
        {}
      )), _ === "dragmove" || _ === "dragstart" ? (v.placeholder.i = $, v.placeholder.x = C.x, v.placeholder.y = C.y, v.placeholder.w = U, v.placeholder.h = q, qe(() => {
        v.isDragging = !0;
      }), F.emit("updateWidth", v.width)) : qe(() => {
        v.isDragging = !1;
      }), k.value = wn(k.value, C, L, ie, !0, f.preventCollision), f.restoreOnDrag ? (C.static = !0, ut(k.value, f.verticalCompact, Y), C.static = !1) : ut(k.value, f.verticalCompact), F.emit("compact"), ue(), _ === "dragend" && (Y = void 0, p("layout-updated", k.value));
    }
    function ve(_, $, L, ie, q, U) {
      let C = dr(k.value, $);
      we(C) && (C = { h: 0, w: 0, x: 0, y: 0, i: "" });
      let le;
      if (f.preventCollision) {
        const Re = Xr(k.value, { ...C, w: U, h: q }).filter(
          (te) => te.i !== C.i
        );
        if (le = Re.length > 0, le) {
          let te = 1 / 0, G = 1 / 0;
          Re.forEach((ze) => {
            ze.x > C.x && (te = Math.min(te, ze.x)), ze.y > C.y && (G = Math.min(G, ze.y));
          }), Number.isFinite(te) && (C.w = te - C.x), Number.isFinite(G) && (C.h = G - C.y);
        }
      }
      le || (C.w = U, C.h = q), _ === "resizestart" || _ === "resizemove" ? (v.placeholder.i = $, v.placeholder.x = L, v.placeholder.y = ie, v.placeholder.w = C.w, v.placeholder.h = C.h, qe(() => {
        v.isDragging = !0;
      }), F.emit("updateWidth", v.width)) : qe(() => {
        v.isDragging = !1;
      }), f.responsive && b(), ut(k.value, f.verticalCompact), F.emit("compact"), ue(), _ === "resizeend" && p("layout-updated", k.value);
    }
    function b() {
      const _ = No(f.breakpoints, v.width), $ = xn(_, f.cols);
      !we(v.lastBreakpoint) && !v.layouts[v.lastBreakpoint] && (v.layouts[v.lastBreakpoint] = bn(k.value));
      const L = qo(
        v.originalLayout,
        v.layouts,
        f.breakpoints,
        _,
        v.lastBreakpoint,
        $,
        f.verticalCompact
      );
      v.layouts[_] = L, v.lastBreakpoint !== _ && p("breakpoint-changed", _, L), p("update:layout", L), v.lastBreakpoint = _, F.emit("setColNum", xn(_, f.cols));
    }
    function ge() {
      v.layouts = Object.assign({}, f.responsiveLayouts);
    }
    function xe(_, $) {
      const L = new Set($.map((C) => C.i)), ie = new Set(_.map((C) => C.i)), q = _.filter((C) => !L.has(C.i)), U = $.filter((C) => !ie.has(C.i));
      return q.concat(U);
    }
    return (_, $) => (St(), Yt("div", {
      ref_key: "wrapper",
      ref: g,
      class: "vgl-layout",
      style: Fr(v.mergedStyle)
    }, [
      _.$slots.default ? yn(_.$slots, "default", { key: 0 }) : (St(!0), Yt(mo, { key: 1 }, yo(k.value, (L) => (St(), bo(yr, wo({
        key: L.i
      }, L), {
        default: xo(() => [
          yn(_.$slots, "item", { item: L })
        ]),
        _: 2
      }, 1040))), 128)),
      Eo(ko(yr, {
        class: "vgl-item--placeholder",
        x: v.placeholder.x,
        y: v.placeholder.y,
        w: v.placeholder.w,
        h: v.placeholder.h,
        i: v.placeholder.i
      }, null, 8, ["x", "y", "w", "h", "i"]), [
        [So, v.isDragging]
      ])
    ], 4));
  }
});
export {
  yr as GridItem,
  Ra as GridLayout
};
