// vue-print-next/dist/vue-print-next.js
var B = Object.defineProperty, C = Object.defineProperties;
var E = Object.getOwnPropertyDescriptors;
var v = Object.getOwnPropertySymbols;
var g = Object.prototype.hasOwnProperty, x = Object.prototype.propertyIsEnumerable;
var y = (s, e, t) => e in s ? B(s, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : s[e] = t, f = (s, e) => {
  for (var t in e || (e = {}))
    g.call(e, t) && y(s, t, e[t]);
  if (v)
    for (var t of v(e))
      x.call(e, t) && y(s, t, e[t]);
  return s;
}, m = (s, e) => C(s, E(e));
var b = (s, e) => {
  var t = {};
  for (var r in s)
    g.call(s, r) && e.indexOf(r) < 0 && (t[r] = s[r]);
  if (s != null && v)
    for (var r of v(s))
      e.indexOf(r) < 0 && x.call(s, r) && (t[r] = s[r]);
  return t;
};
var h = (s, e, t) => y(s, typeof e != "symbol" ? e + "" : e, t);
const u = "[VuePrintNext]";
class A {
  constructor(e) {
    // html 文档标准
    h(this, "standards", {
      strict: "strict",
      loose: "loose",
      html5: "html5"
    });
    // 打印窗口的 iframe id
    h(this, "iframeId", "");
    // 预览窗口的 body
    h(this, "previewBody", null);
    // 预览窗口的 关闭按钮
    h(this, "close", null);
    // 调用次数，用于生成唯一 Id
    h(this, "counter", 0);
    // 用户设置
    h(this, "settings");
    const t = e.vue, c = e, { standard: r, zIndex: i, previewTitle: n, previewPrintBtnLabel: o, preview: l, popTitle: d } = c, p = b(c, ["standard", "zIndex", "previewTitle", "previewPrintBtnLabel", "preview", "popTitle"]);
    this.settings = m(f({
      standard: r || "html5",
      zIndex: i || 20002,
      previewTitle: n || "打印预览",
      previewPrintBtnLabel: o || "打印",
      preview: l || !1,
      popTitle: d || document.title
    }, p), {
      previewBeforeOpenCallback() {
        var a;
        (a = e.previewBeforeOpenCallback) == null || a.call(e, t);
      },
      previewOpenCallback() {
        var a;
        (a = e.previewOpenCallback) == null || a.call(e, t);
      },
      openCallback() {
        var a;
        (a = e.openCallback) == null || a.call(e, t);
      },
      closeCallback() {
        var a;
        (a = e.closeCallback) == null || a.call(e, t);
      },
      beforeOpenCallback() {
        var a;
        (a = e.beforeOpenCallback) == null || a.call(e, t);
      }
    }), this.init();
  }
  init() {
    this.counter++, this.iframeId = `printArea_${this.counter}`;
    const { el: e, url: t } = this.settings;
    if (e || t) {
      const r = e ? "" : t || "", i = this.getPrintWindow(r);
      e && this.write(i.doc), this.settings.preview ? this.previewIframeLoad() : this.print(i);
      return;
    }
    if (this.settings.asyncUrl) {
      this.settings.asyncUrl((r) => {
        const i = this.getPrintWindow(r);
        this.settings.preview ? this.previewIframeLoad() : this.print(i);
      }, this.settings.vue);
      return;
    }
    throw new Error(`${u}: Either "el"、"url" or "asyncUrl" parameter must be provided in the settings.`);
  }
  addEvent(e, t, r) {
    e && (e.addEventListener ? e.addEventListener(t, r, !1) : e.attachEvent ? e.attachEvent("on" + t, r) : e["on" + t] = r);
  }
  previewIframeLoad() {
    var r, i;
    const e = document.getElementById("vue-pirnt-next-previewBox");
    if (!e) return;
    const t = e.querySelector("iframe");
    (i = (r = this.settings).previewBeforeOpenCallback) == null || i.call(r), this.addEvent(t, "load", () => {
      var n, o;
      this.previewBoxShow(), (o = (n = this.settings).previewOpenCallback) == null || o.call(n);
    }), this.addEvent(e.querySelector(".previewBodyUtilPrintBtn"), "click", () => {
      var n, o, l, d, p, c, a;
      (o = (n = this.settings).beforeOpenCallback) == null || o.call(n), (d = (l = this.settings).openCallback) == null || d.call(l), (p = t == null ? void 0 : t.contentWindow) == null || p.print(), (a = (c = this.settings).closeCallback) == null || a.call(c);
    });
  }
  print(e) {
    var n, o;
    const t = document.getElementById(this.iframeId) || e.f, r = t == null ? void 0 : t.contentWindow;
    if (!r) return;
    const i = () => {
      const l = setTimeout(() => {
        var d, p, c, a;
        r.focus(), (p = (d = this.settings).openCallback) == null || p.call(d), r.print(), t.remove(), (a = (c = this.settings).closeCallback) == null || a.call(c), clearTimeout(l);
      });
    };
    (o = (n = this.settings).beforeOpenCallback) == null || o.call(n), this.addEvent(t, "load", i);
  }
  /**
   * 获取打印需要隐藏的 css
   * @private
   */
  getNoPrintMediaStyle() {
    const e = this.settings.noPrintSelector;
    return e ? !Array.isArray(e) && !(typeof e == "string") ? (console.error(new TypeError(`${u}: The "noPrintSelector" must be either a string or an array of strings. Please check your settings.`)), "") : `${(Array.isArray(e) ? e : [e]).filter((o) => o.trim()).join(",")} { display: none; }` : "";
  }
  write(e) {
    e.open(), e.write(
      `${this.docType()}<html lang='zh'>${this.getHead()}${this.getBody()}</html>`
    ), e.close();
  }
  docType() {
    if (this.settings.standard === this.standards.html5)
      return "<!DOCTYPE html>";
    const e = this.settings.standard === this.standards.loose ? " Transitional" : "", t = this.settings.standard === this.standards.loose ? "loose" : "strict";
    return `<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01${e}//EN" "http://www.w3.org/TR/html4/${t}.dtd">`;
  }
  /**
   * 获取媒体打印独有样式
   * @private
   */
  getPrintMediaStyle() {
    return "body {-webkit-print-color-adjust: exact; -moz-print-color-adjust: exact; -ms-print-color-adjust: exact; print-color-adjust: exact;}";
  }
  getHead() {
    const e = (this.settings.extraHead || "").split(",").filter((l) => l.length > 0).join(""), t = Array.from(document.querySelectorAll("link")).filter((l) => l.href.includes(".css")).map((l) => `<link type="text/css" rel="stylesheet" href='${l.href}'>`).join(""), r = Array.from(document.querySelectorAll("style")).map((l) => l.outerHTML).join(""), i = (this.settings.extraCss || "").split(",").filter((l) => l.trim().length > 0).map((l) => `<link type="text/css" rel="stylesheet" href='${l.trim()}'>`).join(""), n = this.getPrintMediaStyle(), o = this.getNoPrintMediaStyle();
    return `<head>
							<title>${this.settings.popTitle}</title>
							${e}${t}
							${r}
							<style type="text/css">${o}${n}</style>
							${i}
						</head>`;
  }
  /**
   * 获取打印区的 DOM
   */
  getPrintAreaDom() {
    const e = this.settings.el, t = typeof e == "string";
    if (e instanceof HTMLElement) return [e];
    if (!t)
      throw new TypeError(`${u}: The "el" property should be either a string (CSS selector) or an HTMLElement, but received type "${typeof e}".`);
    let r = Array.from(document.querySelectorAll(e));
    if (!(r != null && r.length))
      throw new Error(`${u}: No elements found matching the selector: "${e}".`);
    return r;
  }
  getBody() {
    const e = this.getPrintAreaDom(), t = document.createElement("div");
    return e.forEach((r) => {
      const i = r.cloneNode(!0);
      this.removeScriptHandler(i), this.canvasToImgHandler(r, i), this.formDataHandler(r, i), t.appendChild(i);
    }), `<body>${t.innerHTML}</body>`;
  }
  /**
   * 移除 script 标签
   * @param clonedElement
   */
  removeScriptHandler(e) {
    e.querySelectorAll("script").forEach((r) => {
      r.remove();
    });
  }
  /**
   * 复制 Canvas 内容到新的 Canvas 元素
   * @param originalElement
   * @param clonedElement
   * @private
   */
  canvasToImgHandler(e, t) {
    const r = e.querySelectorAll("canvas");
    t.querySelectorAll("canvas").forEach((n, o) => {
      const l = r[o], d = n.parentNode, p = l.toDataURL("image/png"), c = new Image();
      c.className = "canvasImg", c.style.display = "block", c.src = p, d == null || d.appendChild(c), n.remove();
    });
  }
  /**
   * 根据type去处理form表单
   * @param originalElement
   * @param clonedElement
   * @private
   */
  formDataHandler(e, t) {
    const r = t.querySelectorAll("input,select,textarea");
    let i = -1;
    r.forEach((n) => {
      var d;
      const o = n.value;
      switch ((d = n.getAttribute("type")) != null ? d : n.tagName.toLowerCase()) {
        case "select":
          i++;
          const p = e.querySelectorAll("select")[i];
          if (p) {
            const c = p.selectedIndex;
            n.options[c].setAttribute("selected", "selected");
          }
          break;
        case "textarea":
          n.innerHTML = o, n.setAttribute("html", o);
          break;
        case "radio":
        case "checkbox":
          n.checked && n.setAttribute("checked", "checked");
          break;
        default:
          n.value = o, n.setAttribute("value", o);
          break;
      }
    });
  }
  // 生成并返回打印窗口的 iframe 和文档对象
  getPrintWindow(e) {
    var i;
    const t = this.createIframe(e), r = t.contentDocument || ((i = t.contentWindow) == null ? void 0 : i.document) || t.document;
    if (!r)
      throw new Error(`${u}: Unable to find the document object within the created iframe. Please ensure the iframe is correctly created and loaded.`);
    return { f: t, win: t.contentWindow || t, doc: r };
  }
  // 显示预览窗口
  previewBoxShow() {
    var t;
    const e = document.getElementById("vue-pirnt-next-previewBox");
    e && ((t = document.querySelector("html")) == null || t.setAttribute("style", "overflow: hidden"), e.style.display = "block");
  }
  // 隐藏预览窗口
  previewBoxHide() {
    var t, r;
    const e = document.getElementById("vue-pirnt-next-previewBox");
    e && ((t = document.querySelector("html")) == null || t.setAttribute("style", "overflow: visible;"), (r = e.querySelector("iframe")) == null || r.remove(), e.style.display = "none");
  }
  // 创建或获取打印预览的框架
  previewBox() {
    var r;
    let e = document.getElementById("vue-pirnt-next-previewBox");
    if (e)
      return (r = e.querySelector("iframe")) == null || r.remove(), { close: e.querySelector(".previewClose"), previewBody: e.querySelector(".previewBody") };
    e = document.createElement("div"), e.setAttribute("id", "vue-pirnt-next-previewBox"), e.setAttribute(
      "style",
      "position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background: white; display: none; z-index: " + this.settings.zIndex
    );
    const t = document.createElement("div");
    return t.setAttribute("class", "previewHeader"), t.setAttribute("style", "padding: 5px 20px;"), t.innerHTML = this.settings.previewTitle || "", this.close = this.createCloseButton(), t.appendChild(this.close), e.appendChild(t), this.previewBody = this.createPreviewBody(), e.appendChild(this.previewBody), document.body.appendChild(e), { close: this.close, previewBody: this.previewBody };
  }
  // 创建iframe元素
  createIframe(e) {
    const t = document.createElement("iframe");
    if (t.id = this.iframeId, t.src = e || (/* @__PURE__ */ new Date()).getTime().toString(), t.style.display = "none", !this.settings.preview)
      document.body.appendChild(t);
    else {
      t.setAttribute("style", "border: 0px; flex: 1;");
      const { close: r, previewBody: i } = this.previewBox();
      i && i.appendChild(t), this.addEvent(r, "click", this.previewBoxHide.bind(this));
    }
    return t;
  }
  // 创建关闭按钮
  createCloseButton() {
    const e = document.createElement("div");
    e.setAttribute("class", "previewClose"), e.setAttribute("style", "position: absolute; top: 5px; right: 20px; width: 25px; height: 20px; cursor: pointer;");
    const t = document.createElement("div"), r = document.createElement("div"), i = "position: absolute; width: 3px; height: 100%; background: #040404; top: 0px; left: 50%;";
    return t.setAttribute("class", "closeBefore"), t.setAttribute("style", `${i} transform: rotate(45deg);`), r.setAttribute("class", "closeAfter"), r.setAttribute("style", `${i} transform: rotate(-45deg);`), e.appendChild(t), e.appendChild(r), e;
  }
  // 创建预览主体
  createPreviewBody() {
    const e = document.createElement("div");
    e.className = "previewBody", e.style.cssText = "display: flex; flex-direction: column; height: 100%;";
    const t = document.createElement("div");
    t.className = "previewBodyUtil", t.style.cssText = "height: 32px; background: #474747; position: relative;";
    const r = document.createElement("div");
    return r.className = "previewBodyUtilPrintBtn", r.style.cssText = "position: absolute; padding: 2px 10px; margin-top: 3px; left: 24px; font-size: 14px; color: white; cursor: pointer; background: rgba(0,0,0,.12); border: 1px solid rgba(0,0,0,.35); box-shadow: inset 0 1px 0 hsla(0,0%,100%,.05), inset 0 0 1px hsla(0,0%,100%,.15);", r.innerHTML = this.settings.previewPrintBtnLabel || "", t.appendChild(r), e.appendChild(t), e;
  }
}
const S = (s, e, t) => {
  s.addEventListener ? s.addEventListener(e, t, !1) : s.attachEvent ? s.attachEvent("on" + e, t) : s["on" + e] = t;
}, w = {
  directiveName: "print",
  // vue3 指定挂载
  mounted(s, e) {
    let t, r = {};
    S(s, "click", () => {
      e.value ? typeof e.value == "string" ? t = e.value : typeof e.value == "object" && (t = e.value.el, r = e.value) : t = "body", new A(m(f({}, r), { el: t, vue: e.instance }));
    });
  },
  // 兼容 Vue2 指令挂载
  bind(s, e, t) {
    e.instance = t.context, w.mounted(s, e);
  }
}, T = {
  install(s) {
    s.directive(w.directiveName, w);
  }
};
export {
  A as VuePrintNext,
  T as printPlugin,
  w as vPrint
};
